<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Katharina Hembach" />

<meta name="date" content="2020-06-26" />

<title>Filtering</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">neural_scRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/khembach/neural_scRNAseq">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Filtering</h1>
<h4 class="author">Katharina Hembach</h4>
<h4 class="date">June 26, 2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-06-26
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>neural_scRNAseq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200522code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200522)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200522code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200522)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtreeafe57ccb03f02d4568586b075c77cd26bc1fd24ctargetblankafe57cca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/khembach/neural_scRNAseq/tree/afe57ccb03f02d4568586b075c77cd26bc1fd24c" target="_blank">afe57cc</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtreeafe57ccb03f02d4568586b075c77cd26bc1fd24ctargetblankafe57cca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/khembach/neural_scRNAseq/tree/afe57ccb03f02d4568586b075c77cd26bc1fd24c" target="_blank">afe57cc</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    ._MA.pdf
    Ignored:    ._MA2.pdf
    Ignored:    ._MA_plots.pdf
    Ignored:    ._Rplots.pdf
    Ignored:    .__workflowr.yml
    Ignored:    ._hm.pdf
    Ignored:    ._neural_scRNAseq.Rproj
    Ignored:    ._sample5_MA_2nd_pop.pdf
    Ignored:    ._sample5_QC_2nd_pop.pdf
    Ignored:    ._tmp.pdf
    Ignored:    ._tmp_detected.pdf
    Ignored:    ._tmp_manual_discard.pdf
    Ignored:    ._tmp_manual_discard1.pdf
    Ignored:    ._tmp_manual_discard_all.pdf
    Ignored:    ._tmp_manual_discard_all1.pdf
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/._.DS_Store
    Ignored:    analysis/._01-preprocessing.Rmd
    Ignored:    analysis/._01-preprocessing.html
    Ignored:    analysis/._02.1-SampleQC.Rmd
    Ignored:    analysis/._03-filtering.Rmd
    Ignored:    analysis/._04-clustering.Rmd
    Ignored:    analysis/._04-clustering.knit.md
    Ignored:    analysis/._04.1-cell_cycle.Rmd
    Ignored:    analysis/._05-annotation.Rmd
    Ignored:    analysis/.__site.yml
    Ignored:    analysis/._additional_filtering.Rmd
    Ignored:    analysis/._additional_filtering_clustering.Rmd
    Ignored:    analysis/._index.Rmd
    Ignored:    analysis/01-preprocessing_cache/
    Ignored:    analysis/02-1-SampleQC_cache/
    Ignored:    analysis/02-quality_control_cache/
    Ignored:    analysis/02.1-SampleQC_cache/
    Ignored:    analysis/04-clustering_cache/
    Ignored:    analysis/04.1-cell_cycle_cache/
    Ignored:    analysis/05-annotation_cache/
    Ignored:    analysis/additional_filtering_cache/
    Ignored:    analysis/additional_filtering_clustering_cache/
    Ignored:    analysis/figure/
    Ignored:    analysis/sample5_QC_cache/
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._.smbdeleteAAA17ed8b4b
    Ignored:    data/._metadata.csv
    Ignored:    data/data_sushi/
    Ignored:    data/filtered_feature_matrices/
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/additional_filtering.rds
    Ignored:    output/figures/
    Ignored:    output/sce_01_preprocessing.rds
    Ignored:    output/sce_02_quality_control.rds
    Ignored:    output/sce_03_filtering.rds
    Ignored:    output/sce_preprocessing.rds
    Ignored:    output/so_04_1_cell_cycle.rds
    Ignored:    output/so_04_clustering.rds
    Ignored:    output/so_additional_filtering_clustering.rds

Untracked files:
    Untracked:  MA.pdf
    Untracked:  MA2.pdf
    Untracked:  MA_plots.pdf
    Untracked:  Rplots.pdf
    Untracked:  analysis/additional_filtering.Rmd
    Untracked:  analysis/additional_filtering_clustering.Rmd
    Untracked:  analysis/sample5_QC.Rmd
    Untracked:  analysis/tabsets.Rmd
    Untracked:  hm.pdf
    Untracked:  sample5_MA_2nd_pop.pdf
    Untracked:  sample5_QC_2nd_pop.pdf
    Untracked:  scripts/
    Untracked:  tmp.pdf
    Untracked:  tmp_detected.pdf
    Untracked:  tmp_manual_discard.pdf
    Untracked:  tmp_manual_discard1.pdf
    Untracked:  tmp_manual_discard_all.pdf
    Untracked:  tmp_manual_discard_all1.pdf

Unstaged changes:
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/03-filtering.Rmd</code>) and HTML (<code>docs/03-filtering.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/afe57ccb03f02d4568586b075c77cd26bc1fd24c/analysis/03-filtering.Rmd" target="_blank">afe57cc</a>
</td>
<td>
khembach
</td>
<td>
2020-06-26
</td>
<td>
additional filtering of sample 5 and 6
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/khembach/neural_scRNAseq/1f41de5ce67730f63c60c255c20a4850bb92a257/docs/03-filtering.html" target="_blank">1f41de5</a>
</td>
<td>
khembach
</td>
<td>
2020-06-08
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/khembach/neural_scRNAseq/6d822af9ff45e71aaae0b22865306f8d3e6b49ee/docs/03-filtering.html" target="_blank">6d822af</a>
</td>
<td>
khembach
</td>
<td>
2020-05-28
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/a7ced5960ae11be3e8967081abc689d3b9b21849/analysis/03-filtering.Rmd" target="_blank">a7ced59</a>
</td>
<td>
khembach
</td>
<td>
2020-05-28
</td>
<td>
cell and gene filtering
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>library(scater)
library(LSD)
library(dplyr)
library(edgeR)
library(ggrepel)</code></pre>
</div>
<div id="load-data" class="section level2">
<h2>Load data</h2>
<pre class="r"><code>sce &lt;- readRDS(file.path(&quot;output&quot;, &quot;sce_02_quality_control.rds&quot;))</code></pre>
</div>
<div id="identification-of-outlier-cells" class="section level2">
<h2>Identification of outlier cells</h2>
<p>Based on the QC metrics, we now identify outlier cells:</p>
<pre class="r"><code>cols &lt;- c(&quot;sum&quot;, &quot;detected&quot;, &quot;subsets_Mt_percent&quot;)
log &lt;- c(TRUE, TRUE, FALSE)
type &lt;- c(&quot;both&quot;, &quot;both&quot;, &quot;higher&quot;)

drop_cols &lt;- paste0(cols, &quot;_drop&quot;)
for (i in seq_along(cols))
    colData(sce)[[drop_cols[i]]] &lt;- isOutlier(sce[[cols[i]]], 
        nmads = 3, type = type[i], log = log[i], batch = sce$sample_id)

# Overlap of outlier cells from two metrics
sapply(drop_cols, function(i) 
    sapply(drop_cols, function(j)
        sum(sce[[i]] &amp; sce[[j]])))</code></pre>
<pre><code>                        sum_drop detected_drop subsets_Mt_percent_drop
sum_drop                    2499          2289                     614
detected_drop               2289          2740                     809
subsets_Mt_percent_drop      614           809                    2611</code></pre>
<pre class="r"><code>colData(sce)$discard &lt;- rowSums(data.frame(colData(sce)[,drop_cols])) &gt; 0
table(colData(sce)$discard)</code></pre>
<pre><code>
FALSE  TRUE 
43846  4735 </code></pre>
<pre class="r"><code>## Plot the metrics and highlight the discarded cells
plotColData(sce, x = &quot;sample_id&quot;, y = &quot;sum&quot;, colour_by = &quot;discard&quot;) + 
  scale_y_log10()</code></pre>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/1f41de5ce67730f63c60c255c20a4850bb92a257/docs/figure/03-filtering.Rmd/unnamed-chunk-3-1.png" target="_blank">1f41de5</a>
</td>
<td>
khembach
</td>
<td>
2020-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/6d822af9ff45e71aaae0b22865306f8d3e6b49ee/docs/figure/03-filtering.Rmd/unnamed-chunk-3-1.png" target="_blank">6d822af</a>
</td>
<td>
khembach
</td>
<td>
2020-05-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;detected&quot;, colour_by = &quot;discard&quot;) + 
  scale_y_log10()</code></pre>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-2">
Past versions of unnamed-chunk-3-2.png
</button>
</p>
<div id="fig-unnamed-chunk-3-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/1f41de5ce67730f63c60c255c20a4850bb92a257/docs/figure/03-filtering.Rmd/unnamed-chunk-3-2.png" target="_blank">1f41de5</a>
</td>
<td>
khembach
</td>
<td>
2020-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/6d822af9ff45e71aaae0b22865306f8d3e6b49ee/docs/figure/03-filtering.Rmd/unnamed-chunk-3-2.png" target="_blank">6d822af</a>
</td>
<td>
khembach
</td>
<td>
2020-05-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;subsets_Mt_percent&quot;, 
            colour_by = &quot;discard&quot;)</code></pre>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-3">
Past versions of unnamed-chunk-3-3.png
</button>
</p>
<div id="fig-unnamed-chunk-3-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/1f41de5ce67730f63c60c255c20a4850bb92a257/docs/figure/03-filtering.Rmd/unnamed-chunk-3-3.png" target="_blank">1f41de5</a>
</td>
<td>
khembach
</td>
<td>
2020-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/6d822af9ff45e71aaae0b22865306f8d3e6b49ee/docs/figure/03-filtering.Rmd/unnamed-chunk-3-3.png" target="_blank">6d822af</a>
</td>
<td>
khembach
</td>
<td>
2020-05-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ma-plot-of-the-two-populations" class="section level2">
<h2>MA plot of the two populations</h2>
<p>We think that the second cell population in sample 5 with the lower number of UMIs and detected genes consists of damaged cells caused by the dissociation. Sample 5 and 6 are technical replicates and we thus expect similar distributions of the QC metrics. We want to know if there are any genes enriched in the second cell population or if we can safely remove the cells?</p>
<p>To check if we would discarded an specific cell identity with out stringent filtering, we compare the gene expression in the two cell populations (excluding the outlier cells as defined above). If the second cell population is enriched for a specific cell identity, the corresponding marker genes will have high log2FC in the MA plot. Mitochondrial genes are in blue.</p>
<pre class="r"><code>for (i in c(&quot;5NC96&quot;, &quot;6NC96&quot;)) {
cat(&quot;### &quot;, i, &quot;\n&quot;)
## all retained cells
part &lt;- sce[,colData(sce)$sample_id == i &amp; colData(sce)$discard == FALSE]
## second population: cells with less than 7000 detected genes in sample 5 
colData(part)$second_pop &lt;- (colData(part)$sum &lt; 7000 | 
  colData(part)$detected &lt; 3000) 
plotColData(part, x = &quot;sample_id&quot;, y = &quot;sum&quot;, colour_by = &quot;second_pop&quot;) + 
  scale_y_log10()
plotColData(part, x = &quot;sample_id&quot;, y = &quot;detected&quot;, colour_by = &quot;second_pop&quot;) + 
  scale_y_log10()

lost &lt;- calculateAverage(counts(part)[,colData(part)$second_pop])
kept &lt;- calculateAverage(counts(part)[,!colData(part)$second_pop])
logged &lt;- cpm(cbind(lost, kept), log=TRUE, prior.count=2)
logFC &lt;- logged[,1] - logged[,2]
abundance &lt;- rowMeans(logged)

label &lt;- rep(&quot;&quot;, length(logFC))
top &lt;- order(logFC, decreasing = TRUE)[1:20]
label[top]  &lt;- rowData(sce)$symbol[top]
df &lt;- data.frame(abundance = abundance, logFC = logFC, label = label)
mito &lt;- grep(&quot;MT-&quot;, rownames(sce), value = TRUE)
p &lt;- ggplot(df, aes(x = abundance, y = logFC, label = label)) + 
  geom_point(size = 2.5) +
  geom_point(data = df[mito,], color = &quot;dodgerblue&quot;, size = 2.5) + 
  xlab(&quot;Average count&quot;) + ylab(&quot;logFC (lost/kept)&quot;) + 
  theme_bw(base_size = 16) + 
  geom_text_repel()
print(p)
cat(&quot;\n\n&quot;)
}</code></pre>
<div id="nc96" class="section level3">
<h3>5NC96</h3>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/1f41de5ce67730f63c60c255c20a4850bb92a257/docs/figure/03-filtering.Rmd/unnamed-chunk-4-1.png" target="_blank">1f41de5</a>
</td>
<td>
khembach
</td>
<td>
2020-06-08
</td>
</tr>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/6d822af9ff45e71aaae0b22865306f8d3e6b49ee/docs/figure/03-filtering.Rmd/unnamed-chunk-4-1.png" target="_blank">6d822af</a>
</td>
<td>
khembach
</td>
<td>
2020-05-28
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nc96-1" class="section level3">
<h3>6NC96</h3>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-4-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>We decided to additionally filter the cells in sample 5 and 6.</p>
<pre class="r"><code>## filter the cells with less than 5000 UMIs in sample 5 and 6
colData(sce) %&gt;% data.frame() %&gt;% 
  filter(sample_id %in% c(&quot;5NC96&quot;, &quot;6NC96&quot;)) %&gt;% 
  group_by(sample_id, discard) %&gt;% 
  summarise(below_cutoff = sum(sum &lt; 5000))</code></pre>
<pre><code># A tibble: 4 x 3
# Groups:   sample_id [2]
  sample_id discard below_cutoff
  &lt;fct&gt;     &lt;lgl&gt;          &lt;int&gt;
1 5NC96     FALSE           2526
2 5NC96     TRUE             160
3 6NC96     FALSE            134
4 6NC96     TRUE            1221</code></pre>
<pre class="r"><code>colData(sce)$manual_discard_sum &lt;- colData(sce)$sum &lt; 5000 &amp; 
  colData(sce)$sample_id %in% c(&quot;5NC96&quot;, &quot;6NC96&quot;)
## filter the cells with less than 3000 detected genes in sample 5 and 6
colData(sce) %&gt;% data.frame() %&gt;% 
  filter(sample_id %in% c(&quot;5NC96&quot;, &quot;6NC96&quot;)) %&gt;% 
  group_by(sample_id, discard) %&gt;% 
  summarise(below_cutoff = sum(detected &lt; 2500) )</code></pre>
<pre><code># A tibble: 4 x 3
# Groups:   sample_id [2]
  sample_id discard below_cutoff
  &lt;fct&gt;     &lt;lgl&gt;          &lt;int&gt;
1 5NC96     FALSE           2561
2 5NC96     TRUE             202
3 6NC96     FALSE            183
4 6NC96     TRUE            1259</code></pre>
<pre class="r"><code>colData(sce)$manual_discard_detected &lt;- colData(sce)$detected &lt; 2500 &amp; 
  colData(sce)$sample_id %in% c(&quot;5NC96&quot;, &quot;6NC96&quot;)
## highlight all manually discarded cells
colData(sce)$manual_discard &lt;- colData(sce)$manual_discard_sum |
                                   colData(sce)$manual_discard_detected
plotColData(sce, x = &quot;sample_id&quot;, y = &quot;sum&quot;, colour_by = &quot;manual_discard&quot;) + 
  scale_y_log10()</code></pre>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;detected&quot;, colour_by = &quot;manual_discard&quot;) + 
  scale_y_log10()</code></pre>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## highlight all discarded cells
colData(sce)$discard &lt;- colData(sce)$manual_discard |
                                   colData(sce)$discard
plotColData(sce, x = &quot;sample_id&quot;, y = &quot;detected&quot;, colour_by = &quot;discard&quot;) + 
  scale_y_log10()</code></pre>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;sum&quot;, colour_by = &quot;discard&quot;) + 
  scale_y_log10()</code></pre>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;subsets_Mt_percent&quot;, 
            colour_by = &quot;discard&quot;)</code></pre>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-5-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>Plot the library size against the number of detected genes before and after filtering.</p>
<pre class="r"><code>cd &lt;- colData(sce)
layout(matrix(1:12, nrow = 3, byrow = TRUE))
for (i in levels(sce$sample_id)) {
  tmp &lt;- cd[cd$sample_id == i,]
  heatscatter(tmp$sum, tmp$detected, log = &quot;xy&quot;, 
              main = paste0(i, &quot;-unfiltered&quot;), xlab = &quot;total counts&quot;, 
              ylab = &quot;detected genes&quot;)
  heatscatter(tmp$sum[!tmp$discard], tmp$detected[!tmp$discard], 
              log = &quot;xy&quot;, main = paste0(i, &quot;-filtered&quot;), xlab = &quot;total counts&quot;, 
              ylab = &quot;detected genes&quot;)    
}</code></pre>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-6-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="ma-plot-of-all-discarded-over-retained-cells" class="section level2">
<h2>MA plot of all discarded over retained cells</h2>
<p>MA plot of all discarded cells (outlier cells or cells in second population in sample 5) over the retained cells. Mitochondrial genes in blue.</p>
<pre class="r"><code>lost &lt;- calculateAverage(counts(sce)[,colData(sce)$discard])
kept &lt;- calculateAverage(counts(sce)[,!colData(sce)$discard])

logged &lt;- cpm(cbind(lost, kept), log=TRUE, prior.count=2)
logFC &lt;- logged[,1] - logged[,2]
abundance &lt;- rowMeans(logged)

## only label genes with high logFC
label &lt;- rep(&quot;&quot;, length(logFC))
top &lt;- order(logFC, decreasing = TRUE)[1:20]
label[top]  &lt;- rowData(sce)$symbol[top]
df &lt;- data.frame(abundance = abundance, logFC = logFC, label = label)
mito &lt;- grep(&quot;MT-&quot;, rownames(sce), value = TRUE)
p &lt;- ggplot(df, aes(x = abundance, y = logFC, label = label)) + 
  geom_point(size = 2.5) +
  geom_point(data = df[mito,], color = &quot;dodgerblue&quot;, size = 2.5) + 
  xlab(&quot;Average count&quot;) + ylab(&quot;logFC (lost/kept)&quot;) + 
  theme_bw(base_size = 16) + 
  geom_text_repel()
print(p)</code></pre>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="ma-plot-per-sample" class="section level3 tabset">
<h3>MA plot per sample</h3>
<pre class="r"><code>for (s in levels(colData(sce)$sample_id)) {
  cat(&quot;#### &quot;, s, &quot;\n&quot;)
  part &lt;- sce[,colData(sce)$sample_id == s]
  lost &lt;- calculateAverage(counts(part)[,colData(part)$discard])
  kept &lt;- calculateAverage(counts(part)[,!colData(part)$discard])
  logged &lt;- cpm(cbind(lost, kept), log=TRUE, prior.count=2)
  logFC &lt;- logged[,1] - logged[,2]
  abundance &lt;- rowMeans(logged)
  ## only label genes with high logFC
  label &lt;- rep(&quot;&quot;, length(logFC))
  top &lt;- order(logFC, decreasing = TRUE)[1:20]
  label[top]  &lt;- rowData(part)$symbol[top]
  df &lt;- data.frame(abundance = abundance, logFC = logFC, label = label)
  mito &lt;- grep(&quot;MT-&quot;, rownames(part), value = TRUE)
  p &lt;- ggplot(df, aes(x = abundance, y = logFC, label = label)) + 
    geom_point(size = 2.5) +
    geom_point(data = df[mito,], color = &quot;dodgerblue&quot;, size = 2.5) + 
    xlab(&quot;Average count&quot;) + ylab(&quot;logFC (lost/kept)&quot;) + 
    theme_bw(base_size = 16) + 
    geom_text_repel()
  print(p) 
  cat(&quot;\n\n&quot;)
}</code></pre>
<div id="nsc" class="section level4">
<h4>1NSC</h4>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="nsc-1" class="section level4">
<h4>2NSC</h4>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="nc52" class="section level4">
<h4>3NC52</h4>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="nc52-1" class="section level4">
<h4>4NC52</h4>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-8-4.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="nc96-2" class="section level4">
<h4>5NC96</h4>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-8-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="nc96-3" class="section level4">
<h4>6NC96</h4>
<p><img src="figure/03-filtering.Rmd/unnamed-chunk-8-6.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="removal-of-outlier-cells" class="section level2">
<h2>Removal of outlier cells</h2>
<p>We remove the outlier cells and filter the genes:</p>
<pre class="r"><code>## summary of the kept cells
nr &lt;- table(cd$sample_id)
nr_fil &lt;- table(cd$sample_id[!cd$discard])
print(rbind(
    unfiltered = nr, filtered = nr_fil, 
    &quot;%&quot; = round(nr_fil / nr * 100, digits = 0)))</code></pre>
<pre><code>           1NSC 2NSC 3NC52 4NC52 5NC96 6NC96
unfiltered 8893 8854  9109  8865  6571  6289
filtered   8331 8408  8687  7438  3538  4595
%            94   95    95    84    54    73</code></pre>
<pre class="r"><code>## discard the outlier cells
dim(sce)</code></pre>
<pre><code>[1] 19375 48581</code></pre>
<pre class="r"><code>sce &lt;- sce[,!cd$discard]
dim(sce)</code></pre>
<pre><code>[1] 19375 40997</code></pre>
<pre class="r"><code>## we filter genes and require &gt; 1 count in at least 20 cells
sce &lt;- sce[rowSums(counts(sce) &gt; 1) &gt;= 20, ]
dim(sce)</code></pre>
<pre><code>[1] 13254 40997</code></pre>
</div>
<div id="save-data-to-rds" class="section level2">
<h2>Save data to RDS</h2>
<pre class="r"><code>saveRDS(sce, file.path(&quot;output&quot;, &quot;sce_03_filtering.rds&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.6 LTS

Matrix products: default
BLAS:   /usr/local/R/R-4.0.0/lib/libRblas.so
LAPACK: /usr/local/R/R-4.0.0/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] HDF5Array_1.16.0            rhdf5_2.32.0               
 [3] ggrepel_0.8.2               edgeR_3.30.0               
 [5] limma_3.44.1                dplyr_0.8.5                
 [7] LSD_4.0-0                   scater_1.16.0              
 [9] ggplot2_3.3.0               SingleCellExperiment_1.10.1
[11] SummarizedExperiment_1.18.1 DelayedArray_0.14.0        
[13] matrixStats_0.56.0          Biobase_2.48.0             
[15] GenomicRanges_1.40.0        GenomeInfoDb_1.24.0        
[17] IRanges_2.22.2              S4Vectors_0.26.1           
[19] BiocGenerics_0.34.0         workflowr_1.6.2            

loaded via a namespace (and not attached):
 [1] viridis_0.5.1             BiocSingular_1.4.0       
 [3] viridisLite_0.3.0         DelayedMatrixStats_1.10.0
 [5] assertthat_0.2.1          GenomeInfoDbData_1.2.3   
 [7] vipor_0.4.5               yaml_2.2.1               
 [9] pillar_1.4.4              backports_1.1.7          
[11] lattice_0.20-41           glue_1.4.1               
[13] digest_0.6.25             promises_1.1.0           
[15] XVector_0.28.0            colorspace_1.4-1         
[17] cowplot_1.0.0             htmltools_0.4.0          
[19] httpuv_1.5.2              Matrix_1.2-18            
[21] pkgconfig_2.0.3           zlibbioc_1.34.0          
[23] purrr_0.3.4               scales_1.1.1             
[25] whisker_0.4               later_1.0.0              
[27] BiocParallel_1.22.0       git2r_0.27.1             
[29] tibble_3.0.1              farver_2.0.3             
[31] ellipsis_0.3.1            withr_2.2.0              
[33] cli_2.0.2                 magrittr_1.5             
[35] crayon_1.3.4              evaluate_0.14            
[37] fansi_0.4.1               fs_1.4.1                 
[39] beeswarm_0.2.3            tools_4.0.0              
[41] lifecycle_0.2.0           stringr_1.4.0            
[43] Rhdf5lib_1.10.0           munsell_0.5.0            
[45] locfit_1.5-9.4            irlba_2.3.3              
[47] compiler_4.0.0            rsvd_1.0.3               
[49] rlang_0.4.6               grid_4.0.0               
[51] RCurl_1.98-1.2            BiocNeighbors_1.6.0      
[53] labeling_0.3              bitops_1.0-6             
[55] rmarkdown_2.1             gtable_0.3.0             
[57] codetools_0.2-16          R6_2.4.1                 
[59] gridExtra_2.3             knitr_1.28               
[61] utf8_1.1.4                rprojroot_1.3-2          
[63] stringi_1.4.6             ggbeeswarm_0.6.0         
[65] Rcpp_1.0.4.6              vctrs_0.3.0              
[67] tidyselect_1.1.0          xfun_0.14                </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
