<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Katharina Hembach" />


<title>Cluster annotation</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">neural_scRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/khembach/neural_scRNAseq">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Cluster annotation</h1>
<h4 class="author">Katharina Hembach</h4>
<h4 class="date">5/29/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-06-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>neural_scRNAseq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200522code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200522)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200522code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200522)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
dim-red
</li>
<li>
load-data
</li>
<li>
load-libs
</li>
<li>
scran-findMarkers
</li>
<li>
scran-markers-heatmap
</li>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
unnamed-chunk-1
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-3
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>05-annotation_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtreeb6767a67e1f5e494c3bb27a0eee656dc5298a4actargetblankb6767a6a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/khembach/neural_scRNAseq/tree/b6767a67e1f5e494c3bb27a0eee656dc5298a4ac" target="_blank">b6767a6</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtreeb6767a67e1f5e494c3bb27a0eee656dc5298a4actargetblankb6767a6a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/khembach/neural_scRNAseq/tree/b6767a67e1f5e494c3bb27a0eee656dc5298a4ac" target="_blank">b6767a6</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    .__workflowr.yml
    Ignored:    ._neural_scRNAseq.Rproj
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/._.DS_Store
    Ignored:    analysis/._01-preprocessing.Rmd
    Ignored:    analysis/._01-preprocessing.html
    Ignored:    analysis/._02.1-SampleQC.Rmd
    Ignored:    analysis/._04-clustering.Rmd
    Ignored:    analysis/._04-clustering.knit.md
    Ignored:    analysis/.__site.yml
    Ignored:    analysis/01-preprocessing_cache/
    Ignored:    analysis/02-1-SampleQC_cache/
    Ignored:    analysis/02-quality_control_cache/
    Ignored:    analysis/02.1-SampleQC_cache/
    Ignored:    analysis/03-filtering_cache/
    Ignored:    analysis/04-clustering_cache/
    Ignored:    analysis/05-annotation_cache/
    Ignored:    analysis/sample5_QC_cache/
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._metadata.csv
    Ignored:    data/._metadata.xlsx
    Ignored:    data/.smbdeleteAAA17ed8b4b
    Ignored:    data/data_sushi/
    Ignored:    data/filtered_feature_matrices/
    Ignored:    data/metadata.csv
    Ignored:    data/metadata.xlsx
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/figures/
    Ignored:    output/sce_01_preprocessing.rds
    Ignored:    output/sce_02_quality_control.rds
    Ignored:    output/sce_03_filtering.rds
    Ignored:    output/sce_preprocessing.rds
    Ignored:    output/so_04_clustering.rds

Untracked files:
    Untracked:  analysis/sample5_QC.Rmd
    Untracked:  scripts/

Unstaged changes:
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/05-annotation.Rmd</code>) and HTML (<code>docs/05-annotation.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/b6767a67e1f5e494c3bb27a0eee656dc5298a4ac/analysis/05-annotation.Rmd" target="_blank">b6767a6</a>
</td>
<td>
khembach
</td>
<td>
2020-06-10
</td>
<td>
wflow_publish(“analysis/05-annotation.Rmd”, verbose = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/khembach/neural_scRNAseq/419ac731213a1080529b48e43a1e0494cf7a10a5/docs/05-annotation.html" target="_blank">419ac73</a>
</td>
<td>
khembach
</td>
<td>
2020-06-09
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/khembach/neural_scRNAseq/a4d0e04302b05ad6152378accb91e9c994394e42/docs/05-annotation.html" target="_blank">a4d0e04</a>
</td>
<td>
khembach
</td>
<td>
2020-05-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/97d5a52e05366f9053223a3c456d563d00891f5a/analysis/05-annotation.Rmd" target="_blank">97d5a52</a>
</td>
<td>
khembach
</td>
<td>
2020-05-29
</td>
<td>
cluster analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>library(ComplexHeatmap)
library(cowplot)
library(ggplot2)
library(dplyr)
library(muscat)
library(purrr)
library(RColorBrewer)
library(viridis)
library(scran)
library(Seurat)
library(SingleCellExperiment)
library(stringr)</code></pre>
</div>
<div id="load-data-convert-to-sce" class="section level1">
<h1>Load data &amp; convert to SCE</h1>
<pre class="r"><code>so &lt;- readRDS(file.path(&quot;output&quot;, &quot;so_04_clustering.rds&quot;))
sce &lt;- as.SingleCellExperiment(so, assay = &quot;RNA&quot;)
colData(sce) &lt;- as.data.frame(colData(sce)) %&gt;% 
    mutate_if(is.character, as.factor) %&gt;% 
    DataFrame(row.names = colnames(sce))</code></pre>
</div>
<div id="nb.-of-clusters-by-resolution" class="section level1">
<h1>Nb. of clusters by resolution</h1>
<pre class="r"><code>cluster_cols &lt;- grep(&quot;res.[0-9]&quot;, colnames(colData(sce)), value = TRUE)
sapply(colData(sce)[cluster_cols], nlevels)</code></pre>
<pre><code>integrated_snn_res.0.1 integrated_snn_res.0.2 integrated_snn_res.0.4 
                     8                     12                     17 
integrated_snn_res.0.8   integrated_snn_res.1 integrated_snn_res.1.2 
                    24                     29                     31 
  integrated_snn_res.2 
                    39 </code></pre>
</div>
<div id="cluster-sample-counts" class="section level1">
<h1>Cluster-sample counts</h1>
<pre class="r"><code># set cluster IDs to resolution 0.4 clustering
so &lt;- SetIdent(so, value = &quot;integrated_snn_res.0.4&quot;)
so@meta.data$cluster_id &lt;- Idents(so)
sce$cluster_id &lt;- Idents(so)
(n_cells &lt;- table(sce$cluster_id, sce$sample_id))</code></pre>
<pre><code>    
     1NSC 2NSC 3NC52 4NC52 5NC96 6NC96
  0  4850 5041   186    98   308    82
  1     0    0  1552  1279   492   627
  2  1037 1011   337   288   563   391
  3    12   10   572   383  1806   372
  4  1351 1221    69    56    76    16
  5     0    0  1017   847   380   415
  6     2    9   628   620   528   774
  7   253  236   577   606   376   369
  8     0    0  1007   867   250   285
  9     0    0   924   764   327   379
  10  688  716   130   121   188   119
  11    3    3   582   524   211   248
  12    0    0   365   281   186   235
  13    0    0   339   247   141   174
  14    1    1   205   260   210   194
  15   51   70   148   153    64    89
  16   83   90    49    44    83    24</code></pre>
</div>
<div id="relative-cluster-abundances" class="section level1">
<h1>Relative cluster-abundances</h1>
<pre class="r"><code>fqs &lt;- prop.table(n_cells, margin = 2)
mat &lt;- as.matrix(unclass(fqs))
Heatmap(mat,
    col = rev(brewer.pal(11, &quot;RdGy&quot;)[-6]),
    name = &quot;Frequency&quot;,
    cluster_rows = FALSE,
    cluster_columns = FALSE,
    row_names_side = &quot;left&quot;,
    row_title = &quot;cluster_id&quot;,
    column_title = &quot;sample_id&quot;,
    column_title_side = &quot;bottom&quot;,
    rect_gp = gpar(col = &quot;white&quot;),
    cell_fun = function(i, j, x, y, width, height, fill)
        grid.text(round(mat[j, i] * 100, 2), x = x, y = y, 
            gp = gpar(col = &quot;white&quot;, fontsize = 8)))</code></pre>
<p><img src="figure/05-annotation.Rmd/unnamed-chunk-3-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/a4d0e04302b05ad6152378accb91e9c994394e42/docs/figure/05-annotation.Rmd/unnamed-chunk-3-1.png" target="_blank">a4d0e04</a>
</td>
<td>
khembach
</td>
<td>
2020-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cell-cycle" class="section level1">
<h1>Cell cycle</h1>
<p>We assign each cell a cell cycle scores and visualize them in the DR plots. We use known G2/M and S phase markers that come with the <code>Seurat</code> package. The markers are anticorrelated and cells that to not express the markers are supposed to be in G1 phase.</p>
<pre class="r"><code># A list of cell cycle markers, from Tirosh et al, 2015, is loaded with Seurat.  We can
# segregate this list into markers of G2/M phase and markers of S phase
(s_genes &lt;- cc.genes$s.genes)</code></pre>
<pre><code> [1] &quot;MCM5&quot;     &quot;PCNA&quot;     &quot;TYMS&quot;     &quot;FEN1&quot;     &quot;MCM2&quot;     &quot;MCM4&quot;    
 [7] &quot;RRM1&quot;     &quot;UNG&quot;      &quot;GINS2&quot;    &quot;MCM6&quot;     &quot;CDCA7&quot;    &quot;DTL&quot;     
[13] &quot;PRIM1&quot;    &quot;UHRF1&quot;    &quot;MLF1IP&quot;   &quot;HELLS&quot;    &quot;RFC2&quot;     &quot;RPA2&quot;    
[19] &quot;NASP&quot;     &quot;RAD51AP1&quot; &quot;GMNN&quot;     &quot;WDR76&quot;    &quot;SLBP&quot;     &quot;CCNE2&quot;   
[25] &quot;UBR7&quot;     &quot;POLD3&quot;    &quot;MSH2&quot;     &quot;ATAD2&quot;    &quot;RAD51&quot;    &quot;RRM2&quot;    
[31] &quot;CDC45&quot;    &quot;CDC6&quot;     &quot;EXO1&quot;     &quot;TIPIN&quot;    &quot;DSCC1&quot;    &quot;BLM&quot;     
[37] &quot;CASP8AP2&quot; &quot;USP1&quot;     &quot;CLSPN&quot;    &quot;POLA1&quot;    &quot;CHAF1B&quot;   &quot;BRIP1&quot;   
[43] &quot;E2F8&quot;    </code></pre>
<pre class="r"><code>(g2m_genes &lt;- cc.genes$g2m.genes)</code></pre>
<pre><code> [1] &quot;HMGB2&quot;   &quot;CDK1&quot;    &quot;NUSAP1&quot;  &quot;UBE2C&quot;   &quot;BIRC5&quot;   &quot;TPX2&quot;    &quot;TOP2A&quot;  
 [8] &quot;NDC80&quot;   &quot;CKS2&quot;    &quot;NUF2&quot;    &quot;CKS1B&quot;   &quot;MKI67&quot;   &quot;TMPO&quot;    &quot;CENPF&quot;  
[15] &quot;TACC3&quot;   &quot;FAM64A&quot;  &quot;SMC4&quot;    &quot;CCNB2&quot;   &quot;CKAP2L&quot;  &quot;CKAP2&quot;   &quot;AURKB&quot;  
[22] &quot;BUB1&quot;    &quot;KIF11&quot;   &quot;ANP32E&quot;  &quot;TUBB4B&quot;  &quot;GTSE1&quot;   &quot;KIF20B&quot;  &quot;HJURP&quot;  
[29] &quot;CDCA3&quot;   &quot;HN1&quot;     &quot;CDC20&quot;   &quot;TTK&quot;     &quot;CDC25C&quot;  &quot;KIF2C&quot;   &quot;RANGAP1&quot;
[36] &quot;NCAPD2&quot;  &quot;DLGAP5&quot;  &quot;CDCA2&quot;   &quot;CDCA8&quot;   &quot;ECT2&quot;    &quot;KIF23&quot;   &quot;HMMR&quot;   
[43] &quot;AURKA&quot;   &quot;PSRC1&quot;   &quot;ANLN&quot;    &quot;LBR&quot;     &quot;CKAP5&quot;   &quot;CENPE&quot;   &quot;CTCF&quot;   
[50] &quot;NEK2&quot;    &quot;G2E3&quot;    &quot;GAS2L3&quot;  &quot;CBX5&quot;    &quot;CENPA&quot;  </code></pre>
<pre class="r"><code># match the marker genes to the features
m &lt;- match(s_genes, str_split(rownames(GetAssayData(so, assay = &quot;RNA&quot;)), 
                              pattern = &quot;\\.&quot;, simplify = TRUE)[,2])
s_genes &lt;- rownames(GetAssayData(so, assay = &quot;RNA&quot;))[m]
s_genes &lt;- s_genes[!is.na(s_genes)] 
m &lt;- match(g2m_genes, str_split(rownames(GetAssayData(so, assay = &quot;RNA&quot;)), 
                              pattern = &quot;\\.&quot;, simplify = TRUE)[,2])
g2m_genes &lt;- rownames(GetAssayData(so, assay = &quot;RNA&quot;))[m]
g2m_genes &lt;- g2m_genes[!is.na(g2m_genes)] 

# change the default assay to the raw counts
DefaultAssay(so) &lt;- &quot;RNA&quot;
so &lt;- CellCycleScoring(so, s.features = s_genes, g2m.features = g2m_genes, 
                       set.ident = TRUE)
# change back to integrated
DefaultAssay(so) &lt;- &quot;integrated&quot;</code></pre>
</div>
<div id="dr-colored-by-cluster-id" class="section level1 tabset">
<h1>DR colored by cluster ID</h1>
<pre class="r"><code>cs &lt;- sample(colnames(so), 5e3)
.plot_dr &lt;- function(so, dr, id)
    DimPlot(so, cells = cs, group.by = id, reduction = dr, pt.size = 0.4) +
        guides(col = guide_legend(nrow = 11, 
            override.aes = list(size = 3, alpha = 1))) +
        theme_void() + theme(aspect.ratio = 1)
ids &lt;- c(&quot;cluster_id&quot;, &quot;group_id&quot;, &quot;sample_id&quot;, &quot;Phase&quot;)
for (id in ids) {
    cat(&quot;## &quot;, id, &quot;\n&quot;)
    p1 &lt;- .plot_dr(so, &quot;tsne&quot;, id)
    lgd &lt;- get_legend(p1)
    p1 &lt;- p1 + theme(legend.position = &quot;none&quot;)
    p2 &lt;- .plot_dr(so, &quot;umap&quot;, id) + theme(legend.position = &quot;none&quot;)
    ps &lt;- plot_grid(plotlist = list(p1, p2), nrow = 1)
    p &lt;- plot_grid(ps, lgd, nrow = 1, rel_widths = c(1, 0.2))
    print(p)
    cat(&quot;\n\n&quot;)
}</code></pre>
<div id="cluster_id" class="section level2">
<h2>cluster_id</h2>
<p><img src="figure/05-annotation.Rmd/dim-red-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dim-red-1">
Past versions of dim-red-1.png
</button>
</p>
<div id="fig-dim-red-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/419ac731213a1080529b48e43a1e0494cf7a10a5/docs/figure/05-annotation.Rmd/dim-red-1.png" target="_blank">419ac73</a>
</td>
<td>
khembach
</td>
<td>
2020-06-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/a4d0e04302b05ad6152378accb91e9c994394e42/docs/figure/05-annotation.Rmd/dim-red-1.png" target="_blank">a4d0e04</a>
</td>
<td>
khembach
</td>
<td>
2020-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="group_id" class="section level2">
<h2>group_id</h2>
<p><img src="figure/05-annotation.Rmd/dim-red-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dim-red-2">
Past versions of dim-red-2.png
</button>
</p>
<div id="fig-dim-red-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/419ac731213a1080529b48e43a1e0494cf7a10a5/docs/figure/05-annotation.Rmd/dim-red-2.png" target="_blank">419ac73</a>
</td>
<td>
khembach
</td>
<td>
2020-06-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/a4d0e04302b05ad6152378accb91e9c994394e42/docs/figure/05-annotation.Rmd/dim-red-2.png" target="_blank">a4d0e04</a>
</td>
<td>
khembach
</td>
<td>
2020-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample_id" class="section level2">
<h2>sample_id</h2>
<p><img src="figure/05-annotation.Rmd/dim-red-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dim-red-3">
Past versions of dim-red-3.png
</button>
</p>
<div id="fig-dim-red-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/419ac731213a1080529b48e43a1e0494cf7a10a5/docs/figure/05-annotation.Rmd/dim-red-3.png" target="_blank">419ac73</a>
</td>
<td>
khembach
</td>
<td>
2020-06-09
</td>
</tr>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/a4d0e04302b05ad6152378accb91e9c994394e42/docs/figure/05-annotation.Rmd/dim-red-3.png" target="_blank">a4d0e04</a>
</td>
<td>
khembach
</td>
<td>
2020-05-29
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="phase" class="section level2">
<h2>Phase</h2>
<p><img src="figure/05-annotation.Rmd/dim-red-4.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="find-markers-using-scran" class="section level1">
<h1>Find markers using <code>scran</code></h1>
<p>We identify candidate marker genes for each cluster that enable a separation of that group from all other groups.</p>
<pre class="r"><code>scran_markers &lt;- findMarkers(sce, 
    groups = sce$cluster_id, block = sce$sample_id, 
    direction = &quot;up&quot;, lfc = 2, full.stats = TRUE)</code></pre>
<div id="heatmap-of-mean-marker-exprs.-by-cluster" class="section level2">
<h2>Heatmap of mean marker-exprs. by cluster</h2>
<p>We aggregate the cells to pseudobulks and plot the average expression of the condidate marker genes in each of the clusters.</p>
<pre class="r"><code>gs &lt;- lapply(scran_markers, function(u) rownames(u)[u$Top == 1])
## candidate cluster markers
lapply(gs, function(x) str_split(x, pattern = &quot;\\.&quot;, simplify = TRUE)[,2])</code></pre>
<pre><code>$`0`
[1] &quot;SAMD11&quot; &quot;NEFL&quot;   &quot;VIM&quot;    &quot;CKB&quot;    &quot;TTYH1&quot; 

$`1`
[1] &quot;SAMD11&quot; &quot;STMN2&quot;  &quot;CRABP1&quot; &quot;ZFHX3&quot;  &quot;HOXB8&quot;  &quot;MT-CO2&quot;

$`2`
[1] &quot;S100A10&quot; &quot;S100A11&quot; &quot;CLU&quot;     &quot;VIM&quot;     &quot;TAGLN&quot;   &quot;CKB&quot;     &quot;TPM1&quot;   

$`3`
[1] &quot;SAMD11&quot;  &quot;STMN2&quot;   &quot;SNCG&quot;    &quot;RTN1&quot;    &quot;HOXB5&quot;   &quot;ONECUT2&quot; &quot;PCP4&quot;   
[8] &quot;MT-CO2&quot; 

$`4`
[1] &quot;HMGB2&quot; &quot;VIM&quot;   &quot;CKB&quot;   &quot;TOP2A&quot;

$`5`
[1] &quot;STMN2&quot;  &quot;RTN1&quot;   &quot;MEIS2&quot;  &quot;HOXB9&quot;  &quot;PCP4&quot;   &quot;MT-ND3&quot;

$`6`
[1] &quot;C1orf61&quot; &quot;SPP1&quot;    &quot;FABP5&quot;   &quot;VIM&quot;     &quot;HOXB9&quot;   &quot;TTYH1&quot;   &quot;MT-ND4&quot; 

$`7`
[1] &quot;CLU&quot;   &quot;LY6H&quot;  &quot;VIM&quot;   &quot;TAGLN&quot; &quot;DLK1&quot;  &quot;CKB&quot;   &quot;METRN&quot;

$`8`
[1] &quot;SAMD11&quot; &quot;TAC1&quot;   &quot;STMN2&quot;  &quot;ZFHX3&quot;  &quot;HOXB8&quot;  &quot;LAMP5&quot;  &quot;MT-CO2&quot;

$`9`
[1] &quot;SAMD11&quot; &quot;FOXP1&quot;  &quot;STMN2&quot;  &quot;PCP4&quot;  

$`10`
[1] &quot;SAMD11&quot;   &quot;VGF&quot;      &quot;EIF4EBP1&quot; &quot;ANXA1&quot;    &quot;VIM&quot;      &quot;CKB&quot;      &quot;FTL&quot;     

$`11`
[1] &quot;ENC1&quot;    &quot;STMN2&quot;   &quot;NFIB&quot;    &quot;HOXB8&quot;   &quot;HOXB9&quot;   &quot;MT-RNR2&quot;

$`12`
[1] &quot;TAC1&quot;  &quot;STMN2&quot;

$`13`
[1] &quot;STMN2&quot;  &quot;SNCG&quot;   &quot;MPPED2&quot; &quot;PCP4&quot;  

$`14`
[1] &quot;STMN2&quot; &quot;DDIT3&quot;

$`15`
[1] &quot;C1orf61&quot; &quot;HES6&quot;    &quot;SOX2&quot;    &quot;VIM&quot;     &quot;CKB&quot;    

$`16`
[1] &quot;S100A11&quot; &quot;COL3A1&quot;  &quot;COL1A1&quot; </code></pre>
<pre class="r"><code>sub &lt;- sce[unique(unlist(gs)), ]
pbs &lt;- aggregateData(sub, assay = &quot;logcounts&quot;, by = &quot;cluster_id&quot;, fun = &quot;mean&quot;)
mat &lt;- t(muscat:::.scale(assay(pbs)))
## remove the Ensembl ID from the gene names
colnames(mat) &lt;- str_split(colnames(mat), pattern = &quot;\\.&quot;, simplify = TRUE)[,2] 
Heatmap(mat,
    name = &quot;scaled avg.\nexpression&quot;,
    col = viridis(10),
    cluster_rows = FALSE,
    cluster_columns = FALSE,
    row_names_side = &quot;left&quot;,
    row_title = &quot;cluster_id&quot;,
    rect_gp = gpar(col = &quot;white&quot;))</code></pre>
<p><img src="figure/05-annotation.Rmd/scran-markers-heatmap-1.png" width="1152" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.6 LTS

Matrix products: default
BLAS:   /usr/local/R/R-4.0.0/lib/libRblas.so
LAPACK: /usr/local/R/R-4.0.0/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] Seurat_3.1.5                scran_1.16.0               
 [3] SingleCellExperiment_1.10.1 SummarizedExperiment_1.18.1
 [5] DelayedArray_0.14.0         matrixStats_0.56.0         
 [7] Biobase_2.48.0              GenomicRanges_1.40.0       
 [9] GenomeInfoDb_1.24.0         IRanges_2.22.2             
[11] S4Vectors_0.26.1            BiocGenerics_0.34.0        
[13] viridis_0.5.1               viridisLite_0.3.0          
[15] RColorBrewer_1.1-2          purrr_0.3.4                
[17] muscat_1.2.0                dplyr_0.8.5                
[19] ggplot2_3.3.0               cowplot_1.0.0              
[21] ComplexHeatmap_2.4.2        workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] backports_1.1.7           circlize_0.4.9           
  [3] blme_1.0-4                igraph_1.2.5             
  [5] plyr_1.8.6                lazyeval_0.2.2           
  [7] TMB_1.7.16                splines_4.0.0            
  [9] BiocParallel_1.22.0       listenv_0.8.0            
 [11] scater_1.16.0             digest_0.6.25            
 [13] foreach_1.5.0             htmltools_0.4.0          
 [15] gdata_2.18.0              lmerTest_3.1-2           
 [17] magrittr_1.5              memoise_1.1.0            
 [19] cluster_2.1.0             doParallel_1.0.15        
 [21] ROCR_1.0-11               limma_3.44.1             
 [23] globals_0.12.5            annotate_1.66.0          
 [25] prettyunits_1.1.1         colorspace_1.4-1         
 [27] rappdirs_0.3.1            ggrepel_0.8.2            
 [29] blob_1.2.1                xfun_0.14                
 [31] jsonlite_1.6.1            crayon_1.3.4             
 [33] RCurl_1.98-1.2            genefilter_1.70.0        
 [35] lme4_1.1-23               zoo_1.8-8                
 [37] ape_5.3                   survival_3.1-12          
 [39] iterators_1.0.12          glue_1.4.1               
 [41] gtable_0.3.0              zlibbioc_1.34.0          
 [43] XVector_0.28.0            leiden_0.3.3             
 [45] GetoptLong_0.1.8          BiocSingular_1.4.0       
 [47] future.apply_1.5.0        shape_1.4.4              
 [49] scales_1.1.1              DBI_1.1.0                
 [51] edgeR_3.30.0              Rcpp_1.0.4.6             
 [53] xtable_1.8-4              progress_1.2.2           
 [55] clue_0.3-57               reticulate_1.16          
 [57] dqrng_0.2.1               bit_1.1-15.2             
 [59] rsvd_1.0.3                tsne_0.1-3               
 [61] htmlwidgets_1.5.1         httr_1.4.1               
 [63] gplots_3.0.3              ellipsis_0.3.1           
 [65] ica_1.0-2                 farver_2.0.3             
 [67] pkgconfig_2.0.3           XML_3.99-0.3             
 [69] uwot_0.1.8                locfit_1.5-9.4           
 [71] labeling_0.3              tidyselect_1.1.0         
 [73] rlang_0.4.6               reshape2_1.4.4           
 [75] later_1.0.0               AnnotationDbi_1.50.0     
 [77] munsell_0.5.0             tools_4.0.0              
 [79] RSQLite_2.2.0             ggridges_0.5.2           
 [81] evaluate_0.14             stringr_1.4.0            
 [83] yaml_2.2.1                knitr_1.28               
 [85] bit64_0.9-7               fs_1.4.1                 
 [87] fitdistrplus_1.1-1        caTools_1.18.0           
 [89] RANN_2.6.1                pbapply_1.4-2            
 [91] future_1.17.0             nlme_3.1-148             
 [93] whisker_0.4               pbkrtest_0.4-8.6         
 [95] compiler_4.0.0            plotly_4.9.2.1           
 [97] beeswarm_0.2.3            png_0.1-7                
 [99] variancePartition_1.18.0  tibble_3.0.1             
[101] statmod_1.4.34            geneplotter_1.66.0       
[103] stringi_1.4.6             lattice_0.20-41          
[105] Matrix_1.2-18             nloptr_1.2.2.1           
[107] vctrs_0.3.0               pillar_1.4.4             
[109] lifecycle_0.2.0           lmtest_0.9-37            
[111] GlobalOptions_0.1.1       RcppAnnoy_0.0.16         
[113] BiocNeighbors_1.6.0       data.table_1.12.8        
[115] bitops_1.0-6              irlba_2.3.3              
[117] patchwork_1.0.0           httpuv_1.5.2             
[119] colorRamps_2.3            R6_2.4.1                 
[121] promises_1.1.0            KernSmooth_2.23-17       
[123] gridExtra_2.3             vipor_0.4.5              
[125] codetools_0.2-16          boot_1.3-25              
[127] MASS_7.3-51.6             gtools_3.8.2             
[129] assertthat_0.2.1          DESeq2_1.28.1            
[131] rprojroot_1.3-2           rjson_0.2.20             
[133] withr_2.2.0               sctransform_0.2.1        
[135] GenomeInfoDbData_1.2.3    hms_0.5.3                
[137] tidyr_1.1.0               glmmTMB_1.0.1            
[139] minqa_1.2.4               rmarkdown_2.1            
[141] DelayedMatrixStats_1.10.0 Rtsne_0.15               
[143] git2r_0.27.1              numDeriv_2016.8-1.1      
[145] ggbeeswarm_0.6.0         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
