<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Katharina Hembach" />


<title>Cell hashing test quality control</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">neural_scRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/khembach/neural_scRNAseq">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Cell hashing test quality control</h1>
<h4 class="author">Katharina Hembach</h4>
<h4 class="date">5/14/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-05-14
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>neural_scRNAseq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200522code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200522)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200522code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200522)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtreed03901f2aaa95bf1a2c0401c85bf9e82f3d87a1dtargetblankd03901fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/khembach/neural_scRNAseq/tree/d03901f2aaa95bf1a2c0401c85bf9e82f3d87a1d" target="_blank">d03901f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtreed03901f2aaa95bf1a2c0401c85bf9e82f3d87a1dtargetblankd03901fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/khembach/neural_scRNAseq/tree/d03901f2aaa95bf1a2c0401c85bf9e82f3d87a1d" target="_blank">d03901f</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    ._Filtered.pdf
    Ignored:    ._Rplots.pdf
    Ignored:    ._Unfiltered.pdf
    Ignored:    .__workflowr.yml
    Ignored:    ._coverage.pdf
    Ignored:    ._coverage_sashimi.pdf
    Ignored:    ._coverage_sashimi.png
    Ignored:    ._neural_scRNAseq.Rproj
    Ignored:    ._pbDS_cell_level.pdf
    Ignored:    ._pbDS_top_expr_umap.pdf
    Ignored:    ._pbDS_upset.pdf
    Ignored:    ._sashimi.pdf
    Ignored:    ._stmn2.pdf
    Ignored:    ._tdp.pdf
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/._.DS_Store
    Ignored:    analysis/._01-preprocessing.Rmd
    Ignored:    analysis/._01-preprocessing.html
    Ignored:    analysis/._02.1-SampleQC.Rmd
    Ignored:    analysis/._03-filtering.Rmd
    Ignored:    analysis/._04-clustering.Rmd
    Ignored:    analysis/._04-clustering.knit.md
    Ignored:    analysis/._04.1-cell_cycle.Rmd
    Ignored:    analysis/._05-annotation.Rmd
    Ignored:    analysis/._07-cluster-analysis-all-timepoints.Rmd
    Ignored:    analysis/._Lam-0-NSC_no_integration.Rmd
    Ignored:    analysis/._Lam-01-NSC_integration.Rmd
    Ignored:    analysis/._Lam-02-NSC_annotation.Rmd
    Ignored:    analysis/._NSC-1-clustering.Rmd
    Ignored:    analysis/._NSC-2-annotation.Rmd
    Ignored:    analysis/.__site.yml
    Ignored:    analysis/._additional_filtering.Rmd
    Ignored:    analysis/._additional_filtering_clustering.Rmd
    Ignored:    analysis/._index.Rmd
    Ignored:    analysis/._organoid-01-1-qualtiy-control.Rmd
    Ignored:    analysis/._organoid-01-clustering.Rmd
    Ignored:    analysis/._organoid-02-integration.Rmd
    Ignored:    analysis/._organoid-03-cluster_analysis.Rmd
    Ignored:    analysis/._organoid-04-group_integration.Rmd
    Ignored:    analysis/._organoid-04-stage_integration.Rmd
    Ignored:    analysis/._organoid-05-group_integration_cluster_analysis.Rmd
    Ignored:    analysis/._organoid-05-stage_integration_cluster_analysis.Rmd
    Ignored:    analysis/._organoid-06-1-prepare-sce.Rmd
    Ignored:    analysis/._organoid-06-conos-analysis-Seurat.Rmd
    Ignored:    analysis/._organoid-06-conos-analysis-function.Rmd
    Ignored:    analysis/._organoid-06-conos-analysis.Rmd
    Ignored:    analysis/._organoid-06-group-integration-conos-analysis.Rmd
    Ignored:    analysis/._organoid-07-conos-visualization.Rmd
    Ignored:    analysis/._organoid-07-group-integration-conos-visualization.Rmd
    Ignored:    analysis/._organoid-08-conos-comparison.Rmd
    Ignored:    analysis/._organoid-0x-sample_integration.Rmd
    Ignored:    analysis/01-preprocessing_cache/
    Ignored:    analysis/02-1-SampleQC_cache/
    Ignored:    analysis/02-quality_control_cache/
    Ignored:    analysis/02.1-SampleQC_cache/
    Ignored:    analysis/03-filtering_cache/
    Ignored:    analysis/04-clustering_cache/
    Ignored:    analysis/04.1-cell_cycle_cache/
    Ignored:    analysis/05-annotation_cache/
    Ignored:    analysis/06-clustering-all-timepoints_cache/
    Ignored:    analysis/07-cluster-analysis-all-timepoints_cache/
    Ignored:    analysis/Lam-01-NSC_integration_cache/
    Ignored:    analysis/Lam-02-NSC_annotation_cache/
    Ignored:    analysis/NSC-1-clustering_cache/
    Ignored:    analysis/NSC-2-annotation_cache/
    Ignored:    analysis/TDP-01-preprocessing_cache/
    Ignored:    analysis/TDP-02-quality_control_cache/
    Ignored:    analysis/TDP-03-filtering_cache/
    Ignored:    analysis/TDP-04-clustering_cache/
    Ignored:    analysis/TDP-05-00-filtering-plasmid-QC_cache/
    Ignored:    analysis/TDP-05-plasmid_expression_cache/
    Ignored:    analysis/TDP-06-cluster_analysis_cache/
    Ignored:    analysis/TDP-07-01-STMN2_expression_cache/
    Ignored:    analysis/TDP-07-cluster_12_cache/
    Ignored:    analysis/TDP-08-00-clustering-HA-D96_cache/
    Ignored:    analysis/TDP-08-01-HA-D96-expression-changes_cache/
    Ignored:    analysis/TDP-08-02-TDP_target_genes_cache/
    Ignored:    analysis/TDP-08-clustering-timeline-HA_cache/
    Ignored:    analysis/additional_filtering_cache/
    Ignored:    analysis/additional_filtering_clustering_cache/
    Ignored:    analysis/figure/
    Ignored:    analysis/organoid-01-1-qualtiy-control_cache/
    Ignored:    analysis/organoid-01-clustering_cache/
    Ignored:    analysis/organoid-02-integration_cache/
    Ignored:    analysis/organoid-03-cluster_analysis_cache/
    Ignored:    analysis/organoid-04-group_integration_cache/
    Ignored:    analysis/organoid-04-stage_integration_cache/
    Ignored:    analysis/organoid-05-group_integration_cluster_analysis_cache/
    Ignored:    analysis/organoid-05-stage_integration_cluster_analysis_cache/
    Ignored:    analysis/organoid-06-conos-analysis_cache/
    Ignored:    analysis/organoid-06-conos-analysis_test_cache/
    Ignored:    analysis/organoid-06-group-integration-conos-analysis_cache/
    Ignored:    analysis/organoid-07-conos-visualization_cache/
    Ignored:    analysis/organoid-07-group-integration-conos-visualization_cache/
    Ignored:    analysis/organoid-08-conos-comparison_cache/
    Ignored:    analysis/organoid-0x-sample_integration_cache/
    Ignored:    analysis/sample5_QC_cache/
    Ignored:    analysis/timepoints-01-organoid-integration_cache/
    Ignored:    analysis/timepoints-02-cluster-analysis_cache/
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._.smbdeleteAAA17ed8b4b
    Ignored:    data/._Lam_figure2_markers.R
    Ignored:    data/._README.md
    Ignored:    data/._Reactive_astrocytes_markers.xlsx
    Ignored:    data/._known_NSC_markers.R
    Ignored:    data/._known_cell_type_markers.R
    Ignored:    data/._metadata.csv
    Ignored:    data/._virus_cell_tropism_markers.R
    Ignored:    data/._~$Reactive_astrocytes_markers.xlsx
    Ignored:    data/data_sushi/
    Ignored:    data/filtered_feature_matrices/
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/._NSC_cluster2_marker_genes.txt
    Ignored:    output/._TDP-06-no_integration_cluster12_marker_genes.txt
    Ignored:    output/._TDP-06-no_integration_cluster13_marker_genes.txt
    Ignored:    output/._organoid_integration_cluster1_marker_genes.txt
    Ignored:    output/._tbl_TDP-08-01-muscat_cluster_0.txt
    Ignored:    output/._tbl_TDP-08-01-muscat_cluster_1.txt
    Ignored:    output/._tbl_TDP-08-01-muscat_cluster_10.txt
    Ignored:    output/._tbl_TDP-08-01-muscat_cluster_11.txt
    Ignored:    output/._tbl_TDP-08-01-muscat_cluster_12.txt
    Ignored:    output/._tbl_TDP-08-01-muscat_cluster_13.txt
    Ignored:    output/._tbl_TDP-08-01-muscat_cluster_14.txt
    Ignored:    output/._tbl_TDP-08-01-muscat_cluster_5.txt
    Ignored:    output/._tbl_TDP-08-01-muscat_cluster_7.txt
    Ignored:    output/._tbl_TDP-08-01-muscat_cluster_8.txt
    Ignored:    output/._tbl_TDP-08-01-muscat_cluster_all.xlsx
    Ignored:    output/._tbl_TDP-08-02-targets_hek_cluster_0.txt
    Ignored:    output/._tbl_TDP-08-02-targets_hek_cluster_1.txt
    Ignored:    output/._tbl_TDP-08-02-targets_hek_cluster_10.txt
    Ignored:    output/._tbl_TDP-08-02-targets_hek_cluster_11.txt
    Ignored:    output/._tbl_TDP-08-02-targets_hek_cluster_12.txt
    Ignored:    output/._tbl_TDP-08-02-targets_hek_cluster_13.txt
    Ignored:    output/._tbl_TDP-08-02-targets_hek_cluster_14.txt
    Ignored:    output/._tbl_TDP-08-02-targets_hek_cluster_5.txt
    Ignored:    output/._tbl_TDP-08-02-targets_hek_cluster_7.txt
    Ignored:    output/._tbl_TDP-08-02-targets_hek_cluster_8.txt
    Ignored:    output/._tbl_TDP-08-02-targets_hek_cluster_all.xlsx
    Ignored:    output/._~$tbl_TDP-08-02-targets_hek_cluster_all.xlsx
    Ignored:    output/Lam-01-clustering.rds
    Ignored:    output/NSC_1_clustering.rds
    Ignored:    output/NSC_cluster1_marker_genes.txt
    Ignored:    output/NSC_cluster2_marker_genes.txt
    Ignored:    output/NSC_cluster3_marker_genes.txt
    Ignored:    output/NSC_cluster4_marker_genes.txt
    Ignored:    output/NSC_cluster5_marker_genes.txt
    Ignored:    output/NSC_cluster6_marker_genes.txt
    Ignored:    output/NSC_cluster7_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster0_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster10_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster11_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster12_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster13_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster14_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster15_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster16_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster17_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster1_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster2_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster3_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster4_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster5_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster6_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster7_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster8_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster9_marker_genes.txt
    Ignored:    output/TDP-06_scran_markers.rds
    Ignored:    output/additional_filtering.rds
    Ignored:    output/conos/
    Ignored:    output/conos_organoid-06-conos-analysis.rds
    Ignored:    output/conos_organoid-06-group-integration-conos-analysis.rds
    Ignored:    output/figures/
    Ignored:    output/organoid_integration_cluster10_marker_genes.txt
    Ignored:    output/organoid_integration_cluster11_marker_genes.txt
    Ignored:    output/organoid_integration_cluster12_marker_genes.txt
    Ignored:    output/organoid_integration_cluster13_marker_genes.txt
    Ignored:    output/organoid_integration_cluster14_marker_genes.txt
    Ignored:    output/organoid_integration_cluster15_marker_genes.txt
    Ignored:    output/organoid_integration_cluster16_marker_genes.txt
    Ignored:    output/organoid_integration_cluster17_marker_genes.txt
    Ignored:    output/organoid_integration_cluster1_marker_genes.txt
    Ignored:    output/organoid_integration_cluster2_marker_genes.txt
    Ignored:    output/organoid_integration_cluster3_marker_genes.txt
    Ignored:    output/organoid_integration_cluster4_marker_genes.txt
    Ignored:    output/organoid_integration_cluster5_marker_genes.txt
    Ignored:    output/organoid_integration_cluster6_marker_genes.txt
    Ignored:    output/organoid_integration_cluster7_marker_genes.txt
    Ignored:    output/organoid_integration_cluster8_marker_genes.txt
    Ignored:    output/organoid_integration_cluster9_marker_genes.txt
    Ignored:    output/res_TDP-08-01-muscat.rds
    Ignored:    output/sce_01_preprocessing.rds
    Ignored:    output/sce_02_quality_control.rds
    Ignored:    output/sce_03_filtering.rds
    Ignored:    output/sce_03_filtering_all_genes.rds
    Ignored:    output/sce_06-1-prepare-sce.rds
    Ignored:    output/sce_TDP-08-01-muscat.rds
    Ignored:    output/sce_TDP_01_preprocessing.rds
    Ignored:    output/sce_TDP_02_quality_control.rds
    Ignored:    output/sce_TDP_03_filtering.rds
    Ignored:    output/sce_TDP_03_filtering_all_genes.rds
    Ignored:    output/sce_organoid-01-clustering.rds
    Ignored:    output/sce_preprocessing.rds
    Ignored:    output/so_04-stage_integration.rds
    Ignored:    output/so_04_1_cell_cycle.rds
    Ignored:    output/so_04_clustering.rds
    Ignored:    output/so_06-clustering_all_timepoints.rds
    Ignored:    output/so_08-00_clustering_HA_D96.rds
    Ignored:    output/so_08-clustering_timeline_HA.rds
    Ignored:    output/so_0x-sample_integration.rds
    Ignored:    output/so_TDP-06-cluster-analysis.rds
    Ignored:    output/so_TDP_04_clustering.rds
    Ignored:    output/so_TDP_05_plasmid_expression.rds
    Ignored:    output/so_additional_filtering_clustering.rds
    Ignored:    output/so_integrated_organoid-02-integration.rds
    Ignored:    output/so_merged_organoid-02-integration.rds
    Ignored:    output/so_organoid-01-clustering.rds
    Ignored:    output/so_sample_organoid-01-clustering.rds
    Ignored:    output/so_timepoints-01-organoid_integration.rds
    Ignored:    output/tbl_TDP-08-01-muscat.rds
    Ignored:    output/tbl_TDP-08-01-muscat_cluster_0.txt
    Ignored:    output/tbl_TDP-08-01-muscat_cluster_1.txt
    Ignored:    output/tbl_TDP-08-01-muscat_cluster_10.txt
    Ignored:    output/tbl_TDP-08-01-muscat_cluster_11.txt
    Ignored:    output/tbl_TDP-08-01-muscat_cluster_12.txt
    Ignored:    output/tbl_TDP-08-01-muscat_cluster_13.txt
    Ignored:    output/tbl_TDP-08-01-muscat_cluster_14.txt
    Ignored:    output/tbl_TDP-08-01-muscat_cluster_5.txt
    Ignored:    output/tbl_TDP-08-01-muscat_cluster_7.txt
    Ignored:    output/tbl_TDP-08-01-muscat_cluster_8.txt
    Ignored:    output/tbl_TDP-08-01-muscat_cluster_all.xlsx
    Ignored:    output/tbl_TDP-08-02-targets_hek.rds
    Ignored:    output/tbl_TDP-08-02-targets_hek_cluster_0.txt
    Ignored:    output/tbl_TDP-08-02-targets_hek_cluster_1.txt
    Ignored:    output/tbl_TDP-08-02-targets_hek_cluster_10.txt
    Ignored:    output/tbl_TDP-08-02-targets_hek_cluster_11.txt
    Ignored:    output/tbl_TDP-08-02-targets_hek_cluster_12.txt
    Ignored:    output/tbl_TDP-08-02-targets_hek_cluster_13.txt
    Ignored:    output/tbl_TDP-08-02-targets_hek_cluster_14.txt
    Ignored:    output/tbl_TDP-08-02-targets_hek_cluster_5.txt
    Ignored:    output/tbl_TDP-08-02-targets_hek_cluster_7.txt
    Ignored:    output/tbl_TDP-08-02-targets_hek_cluster_8.txt
    Ignored:    output/tbl_TDP-08-02-targets_hek_cluster_all.xlsx
    Ignored:    output/~$tbl_TDP-08-02-targets_hek_cluster_all.xlsx
    Ignored:    scripts/.DS_Store
    Ignored:    scripts/._.DS_Store
    Ignored:    scripts/._bu_Rcode.R
    Ignored:    scripts/._plasmid_expression.sh
    Ignored:    scripts/._plasmid_expression_cell_hashing_test.sh
    Ignored:    scripts/._prepare_salmon_transcripts.R
    Ignored:    scripts/._prepare_salmon_transcripts_cell_hashing_test.R

Untracked files:
    Untracked:  Filtered.pdf
    Untracked:  Rplots.pdf
    Untracked:  Unfiltered
    Untracked:  Unfiltered.pdf
    Untracked:  analysis/Lam-0-NSC_no_integration.Rmd
    Untracked:  analysis/TDP-07-01-STMN2_expression copy.Rmd
    Untracked:  analysis/additional_filtering.Rmd
    Untracked:  analysis/additional_filtering_clustering.Rmd
    Untracked:  analysis/organoid-01-1-qualtiy-control.Rmd
    Untracked:  analysis/organoid-06-conos-analysis-Seurat.Rmd
    Untracked:  analysis/organoid-06-conos-analysis-function.Rmd
    Untracked:  analysis/organoid-07-conos-visualization.Rmd
    Untracked:  analysis/organoid-07-group-integration-conos-visualization.Rmd
    Untracked:  analysis/organoid-08-conos-comparison.Rmd
    Untracked:  analysis/organoid-0x-sample_integration.Rmd
    Untracked:  analysis/sample5_QC.Rmd
    Untracked:  coverage.pdf
    Untracked:  coverage_sashimi.pdf
    Untracked:  coverage_sashimi.png
    Untracked:  data/Homo_sapiens.GRCh38.98.sorted.gtf
    Untracked:  data/Kanton_et_al/
    Untracked:  data/Lam_et_al/
    Untracked:  data/Sep2020/
    Untracked:  data/cell_hashing_test/
    Untracked:  data/reference/
    Untracked:  data/virus_cell_tropism_markers.R
    Untracked:  data/~$Reactive_astrocytes_markers.xlsx
    Untracked:  pbDS_cell_level.pdf
    Untracked:  pbDS_heatmap.pdf
    Untracked:  pbDS_top_expr_umap.pdf
    Untracked:  pbDS_upset.pdf
    Untracked:  sashimi.pdf
    Untracked:  scripts/bu_Rcode.R
    Untracked:  scripts/bu_code.Rmd
    Untracked:  scripts/plasmid_expression_cell_hashing_test.sh
    Untracked:  scripts/prepare_salmon_transcripts_cell_hashing_test.R
    Untracked:  scripts/salmon-latest_linux_x86_64/
    Untracked:  stmn2.pdf
    Untracked:  tdp.pdf

Unstaged changes:
    Modified:   analysis/05-annotation.Rmd
    Modified:   analysis/TDP-04-clustering.Rmd
    Modified:   analysis/TDP-08-01-HA-D96-expression-changes.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/organoid-02-integration.Rmd
    Modified:   analysis/organoid-04-group_integration.Rmd
    Modified:   analysis/organoid-06-conos-analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/CH-test-01-preprocessing.Rmd</code>) and HTML (<code>docs/CH-test-01-preprocessing.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/d03901f2aaa95bf1a2c0401c85bf9e82f3d87a1d/analysis/CH-test-01-preprocessing.Rmd" target="_blank">d03901f</a>
</td>
<td>
khembach
</td>
<td>
2021-05-14
</td>
<td>
preprocessing and quality control of HTO test experiment
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>library(DropletUtils)
library(BiocParallel)
library(ggplot2)
library(scater)
library(readxl)
library(Seurat)
library(scales)
library(viridis)
library(dplyr)</code></pre>
</div>
<div id="importing-cellranger-output-and-metadata" class="section level1">
<h1>Importing CellRanger output and metadata</h1>
<pre class="r"><code>fs &lt;- file.path(&quot;data&quot;, &quot;cell_hashing_test&quot;, 
                &quot;CellRangerCount_57443_2021-05-12--11-37-28&quot;, &quot;HashTag_test&quot;, 
                &quot;filtered_feature_bc_matrix.h5&quot;)
names(fs) &lt;- &quot;cell_hashing_test&quot;
sce_raw &lt;- read10xCounts(samples = fs)

# rename colnames and dimnames
rowData(sce_raw)$Type &lt;- NULL
names(rowData(sce_raw)) &lt;- c(&quot;ensembl_id&quot;, &quot;symbol&quot;)
names(colData(sce_raw)) &lt;- c(&quot;sample_id&quot;, &quot;barcode&quot;)
sce_raw$sample_id &lt;- factor(sce_raw$sample_id)

# load metadata
meta &lt;- read_excel(file.path(&quot;data&quot;, &quot;cell_hashing_test&quot;, &quot;SampleName_feature_ref_MHP.xlsm&quot;))
m &lt;- match(meta$name, rowData(sce_raw)$symbol)

## separate gene counts from HTO counts
rowData(sce_raw) %&gt;% tail</code></pre>
<pre><code>DataFrame with 6 rows and 2 columns
                     ensembl_id      symbol
                    &lt;character&gt; &lt;character&gt;
ENSG00000276017 ENSG00000276017  AC007325.1
ENSG00000278817 ENSG00000278817  AC007325.4
ENSG00000277196 ENSG00000277196  AC007325.2
B0253                     B0253    Hashtag3
B0254                     B0254    Hashtag4
B0257                     B0257    Hashtag7</code></pre>
<pre class="r"><code>sce &lt;- sce_raw[-m,]
dimnames(sce) &lt;- list(with(rowData(sce), paste(ensembl_id, symbol, sep = &quot;.&quot;)),
                      with(colData(sce), paste(barcode, sample_id, sep = &quot;.&quot;)))</code></pre>
</div>
<div id="quality-control" class="section level1">
<h1>Quality control</h1>
<p>We compute cell-level QC.</p>
<pre class="r"><code># remove empty rows
sce &lt;- sce[rowSums(counts(sce) &gt; 0) &gt; 0, ]
dim(sce)</code></pre>
<pre><code>[1] 18824 17631</code></pre>
<pre class="r"><code>(mito &lt;- grep(&quot;MT-&quot;, rownames(sce), value = TRUE))</code></pre>
<pre><code> [1] &quot;ENSG00000210049.MT-TF&quot;   &quot;ENSG00000211459.MT-RNR1&quot;
 [3] &quot;ENSG00000210077.MT-TV&quot;   &quot;ENSG00000210082.MT-RNR2&quot;
 [5] &quot;ENSG00000209082.MT-TL1&quot;  &quot;ENSG00000198888.MT-ND1&quot; 
 [7] &quot;ENSG00000210100.MT-TI&quot;   &quot;ENSG00000210107.MT-TQ&quot;  
 [9] &quot;ENSG00000210112.MT-TM&quot;   &quot;ENSG00000198763.MT-ND2&quot; 
[11] &quot;ENSG00000210117.MT-TW&quot;   &quot;ENSG00000210127.MT-TA&quot;  
[13] &quot;ENSG00000210135.MT-TN&quot;   &quot;ENSG00000210140.MT-TC&quot;  
[15] &quot;ENSG00000210144.MT-TY&quot;   &quot;ENSG00000198804.MT-CO1&quot; 
[17] &quot;ENSG00000210151.MT-TS1&quot;  &quot;ENSG00000210154.MT-TD&quot;  
[19] &quot;ENSG00000198712.MT-CO2&quot;  &quot;ENSG00000210156.MT-TK&quot;  
[21] &quot;ENSG00000228253.MT-ATP8&quot; &quot;ENSG00000198899.MT-ATP6&quot;
[23] &quot;ENSG00000198938.MT-CO3&quot;  &quot;ENSG00000210164.MT-TG&quot;  
[25] &quot;ENSG00000198840.MT-ND3&quot;  &quot;ENSG00000210174.MT-TR&quot;  
[27] &quot;ENSG00000212907.MT-ND4L&quot; &quot;ENSG00000198886.MT-ND4&quot; 
[29] &quot;ENSG00000210176.MT-TH&quot;   &quot;ENSG00000210184.MT-TS2&quot; 
[31] &quot;ENSG00000210191.MT-TL2&quot;  &quot;ENSG00000198786.MT-ND5&quot; 
[33] &quot;ENSG00000198695.MT-ND6&quot;  &quot;ENSG00000210194.MT-TE&quot;  
[35] &quot;ENSG00000198727.MT-CYB&quot;  &quot;ENSG00000210195.MT-TT&quot;  
[37] &quot;ENSG00000210196.MT-TP&quot;  </code></pre>
<pre class="r"><code>sce &lt;- addPerCellQC(sce, subsets = list(Mt = mito))
# we compute the fraction of mitochondrial genes and the logit of it 
sce$subsets_Mt_fraction &lt;- (sce$subsets_Mt_percent + 0.001) /100
sce$subsets_Mt_fraction_logit &lt;- qlogis(sce$subsets_Mt_fraction + 0.001)
# library size
summary(sce$sum)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    500     642     862    5216    5818   96619 </code></pre>
<pre class="r"><code># number of detected genes per cell
summary(sce$detected)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    302     491     609    1650    2540    9081 </code></pre>
<pre class="r"><code># percentage of counts that come from mitochondrial genes:
summary(sce$subsets_Mt_percent)</code></pre>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
 0.01666  3.86824  7.70896  9.93049 13.29026 87.60776 </code></pre>
<div id="diagnostic-plots" class="section level2">
<h2>Diagnostic plots</h2>
<p>The number of counts per cell:</p>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;sum&quot;) + scale_y_log10()</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The number of genes:</p>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;detected&quot;) + scale_y_log10() </code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The percentage of mitochondrial genes:</p>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;subsets_Mt_percent&quot;)</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We plot the total number of counts against the number of detected genes and color by the fraction of mitochondrial genes:</p>
<pre class="r"><code>cd &lt;- data.frame(colData(sce))
ggplot(cd, aes(x = sum, y = detected, color = subsets_Mt_fraction)) +
  geom_point(alpha = 0.7) + 
  geom_density_2d(color = &quot;grey&quot;, bins = 6) +
  scale_x_log10() +
  scale_y_log10() +
  facet_wrap(~sample_id) + 
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
  xlab(&quot;sum of counts&quot;) + 
  ylab(&quot;number of detected genes&quot;) + 
  labs(color = &quot;mitochondrial fraction&quot;) +
  scale_color_viridis(trans = &quot;logit&quot;, breaks = c(0.01, 0.1, 0.25, 0.5, 0.75))</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We plot the total number of counts against the mitochondrial content. Well-behaved cells should have many expressed genes and a low fraction of mitochondrial genes. High mitochondrial content indicates empty or damaged cells.</p>
<pre class="r"><code>ggplot(cd, aes(x = sum, y = subsets_Mt_fraction)) +
  geom_point(color = &quot;darkgrey&quot;, alpha = 0.3) + 
  geom_density_2d(color = &quot;lightblue&quot;) +
  scale_x_log10() +
  scale_y_continuous(trans = &#39;logit&#39;, 
                     breaks = c(0.01, 0.05, 0.1, 0.2, 0.5, 0.75)) +
  facet_wrap(~sample_id) + 
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) + 
  xlab(&quot;sum of counts&quot;) + 
  ylab(&quot;logit(mitochondrial fraction)&quot;)</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="identification-of-outlier-cells" class="section level2">
<h2>Identification of outlier cells</h2>
<p>Based on the QC metrics, we now identify outlier cells:</p>
<pre class="r"><code>cols &lt;- c(&quot;sum&quot;, &quot;detected&quot;, &quot;subsets_Mt_percent&quot;)
log &lt;- c(TRUE, TRUE, FALSE)
type &lt;- c(&quot;lower&quot;, &quot;lower&quot;, &quot;higher&quot;)

drop_cols &lt;- paste0(cols, &quot;_drop&quot;)
for (i in seq_along(cols))
    colData(sce)[[drop_cols[i]]] &lt;- isOutlier(sce[[cols[i]]], 
        nmads = 1, type = type[i], log = log[i], batch = sce$sample_id)

# Overlap of outlier cells from two metrics
sapply(drop_cols, function(i) 
    sapply(drop_cols, function(j)
        sum(sce[[i]] &amp; sce[[j]])))</code></pre>
<pre><code>                        sum_drop detected_drop subsets_Mt_percent_drop
sum_drop                       0             0                       0
detected_drop                  0            53                      49
subsets_Mt_percent_drop        0            49                    3900</code></pre>
<pre class="r"><code>colData(sce)$discard &lt;- rowSums(data.frame(colData(sce)[,drop_cols])) &gt; 0
table(colData(sce)$discard)</code></pre>
<pre><code>
FALSE  TRUE 
13727  3904 </code></pre>
<pre class="r"><code>## Plot the metrics and highlight the discarded cells
plotColData(sce, x = &quot;sample_id&quot;, y = &quot;sum&quot;, colour_by = &quot;discard&quot;) + 
  scale_y_log10()</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/outlier-identification-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;detected&quot;, colour_by = &quot;discard&quot;) + 
  scale_y_log10()</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/outlier-identification-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;subsets_Mt_percent&quot;, 
            colour_by = &quot;discard&quot;)</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/outlier-identification-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## we manually filter filter the cells with less than 2000 UMIs
colData(sce)$manual_discard_sum &lt;- colData(sce)$sum &lt; 1000 
## filter the cells with less than 800 detected genes
colData(sce)$manual_discard_detected &lt;- colData(sce)$detected &lt; 800 


## highlight all manually discarded cells
colData(sce)$manual_discard &lt;- colData(sce)$manual_discard_sum |
                                   colData(sce)$manual_discard_detected
plotColData(sce, x = &quot;sample_id&quot;, y = &quot;sum&quot;, colour_by = &quot;manual_discard&quot;) + 
  scale_y_log10()</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/outlier-identification-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;detected&quot;, colour_by = &quot;manual_discard&quot;) + 
  scale_y_log10()</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/outlier-identification-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## highlight all discarded cells
colData(sce)$discard &lt;- colData(sce)$manual_discard |
                                   colData(sce)$discard
plotColData(sce, x = &quot;sample_id&quot;, y = &quot;detected&quot;, colour_by = &quot;discard&quot;) + 
  scale_y_log10()</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/outlier-identification-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;sum&quot;, colour_by = &quot;discard&quot;) + 
  scale_y_log10()</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/outlier-identification-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;subsets_Mt_percent&quot;, 
            colour_by = &quot;discard&quot;)</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/outlier-identification-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>table(colData(sce)$discard)</code></pre>
<pre><code>
FALSE  TRUE 
 5542 12089 </code></pre>
<p>We discard the outlier cells</p>
<pre class="r"><code>dim(sce)</code></pre>
<pre><code>[1] 18824 17631</code></pre>
<pre class="r"><code>sce_filtered &lt;- sce[,!sce$discard]
dim(sce_filtered)</code></pre>
<pre><code>[1] 18824  5542</code></pre>
</div>
</div>
<div id="create-seurat-object-and-split-gene-and-hto-counts" class="section level1">
<h1>Create Seurat object and split gene and HTO counts</h1>
<pre class="r"><code>## convert DDelayedMatrix to dgCMatrix for import into Seurat object
counts &lt;- as(counts(sce_filtered, withDimnames = FALSE), &quot;dgCMatrix&quot;)
colnames(counts) &lt;- colnames(counts(sce_filtered))
rownames(counts) &lt;- rownames(counts(sce_filtered))
                             
so &lt;- CreateSeuratObject(
    counts = counts,
    meta.data = data.frame(colData(sce_filtered)),
    project = &quot;cell_hashing_test&quot;)</code></pre>
<pre><code>Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
(&#39;-&#39;)</code></pre>
<pre class="r"><code>## add HTO data as independent assay
hto_counts &lt;- as(counts(sce_raw, withDimnames = FALSE)[m,with(colData(sce), 
                                    paste(barcode, sample_id, sep = &quot;.&quot;)) %in% 
                                colnames(sce_filtered)], &quot;dgCMatrix&quot;)
colnames(hto_counts) &lt;- colnames(sce_filtered)
rownames(hto_counts) &lt;- rownames(sce_raw)[m]
so[[&quot;HTO&quot;]] &lt;- CreateAssayObject(counts = hto_counts)</code></pre>
</div>
<div id="data-normalization" class="section level1">
<h1>Data normalization</h1>
<pre class="r"><code>DefaultAssay(so) &lt;- &quot;RNA&quot;
# Normalize RNA data with log normalization
so &lt;- NormalizeData(so)
# Find and scale variable features
so &lt;- FindVariableFeatures(so, selection.method = &quot;mean.var.plot&quot;)
so &lt;- ScaleData(so, features = VariableFeatures(so))

# Normalize HTO data, here we use centered log-ratio (CLR) transformation
so &lt;- NormalizeData(so, assay = &quot;HTO&quot;, normalization.method = &quot;CLR&quot;)</code></pre>
</div>
<div id="demultiplex-cells-based-on-hto-enrichment" class="section level1">
<h1>Demultiplex cells based on HTO enrichment</h1>
<pre class="r"><code>so &lt;- HTODemux(so, assay = &quot;HTO&quot;, positive.quantile = 0.99)</code></pre>
<div id="visualize-results" class="section level2">
<h2>Visualize results</h2>
<pre class="r"><code># Global classification results
table(so$HTO_classification.global)</code></pre>
<pre><code>
 Doublet Negative  Singlet 
     656     2119     2767 </code></pre>
<pre class="r"><code># Group cells based on the max HTO signal
Idents(so) &lt;- &quot;HTO_maxID&quot;
# Group cells based on the max HTO signal
RidgePlot(so, assay = &quot;HTO&quot;, features = rownames(so[[&quot;HTO&quot;]]), ncol = 3)</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/demultiplexing-viz-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># Visualize pairs of HTO signals to check mutual exclusivity in singlets
DefaultAssay(object = so) &lt;- &quot;HTO&quot;
FeatureScatter(so, feature1 = &quot;B0253&quot;, feature2 = &quot;B0254&quot;)</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/demultiplexing-viz-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeatureScatter(so, feature1 = &quot;B0257&quot;, feature2 = &quot;B0254&quot;)</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/demultiplexing-viz-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>FeatureScatter(so, feature1 = &quot;B0253&quot;, feature2 = &quot;B0257&quot;)</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/demultiplexing-viz-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## compare number of UMIs for singlet&#39;s, doublets and negative cells
Idents(so) &lt;- &quot;HTO_classification.global&quot;
VlnPlot(so, features = &quot;nCount_HTO&quot;, pt.size = 0.1, log = TRUE)</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/demultiplexing-viz-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>VlnPlot(so, features = &quot;nCount_RNA&quot;, pt.size = 0.1, log = TRUE)</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/demultiplexing-viz-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## tSNE for HTOs
# First, we will remove negative cells from the object
subs &lt;- subset(so, idents = &quot;Negative&quot;, invert = TRUE)
subs$HTO_classification %&gt;% table</code></pre>
<pre><code>.
      B0253 B0253_B0254 B0253_B0257       B0254 B0254_B0257       B0257 
        830         236         198         939         222         998 </code></pre>
<pre class="r"><code># Calculate a tSNE embedding of the HTO data
DefaultAssay(subs) &lt;- &quot;HTO&quot;
subs &lt;- ScaleData(subs, features = rownames(subs), 
    verbose = FALSE)
subs &lt;- RunPCA(subs, features = rownames(subs), approx = FALSE)</code></pre>
<pre><code>Warning in print.DimReduc(x = reduction.data, dims = ndims.print, nfeatures =
nfeatures.print): Only 3 dimensions have been computed.</code></pre>
<pre><code>Warning: Requested number is larger than the number of available items (3).
Setting to 3.

Warning: Requested number is larger than the number of available items (3).
Setting to 3.

Warning: Requested number is larger than the number of available items (3).
Setting to 3.</code></pre>
<pre class="r"><code>subs &lt;- RunTSNE(subs, dims = 1:3, perplexity = 100)
Idents(subs) &lt;- &quot;HTO_classification.global&quot;
DimPlot(subs)</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/demultiplexing-viz-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>Idents(subs) &lt;- &#39;HTO_classification&#39;
DimPlot(subs)</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/demultiplexing-viz-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># HTO heatmap
HTOHeatmap(so, assay = &quot;HTO&quot;, ncells = 5000)</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/demultiplexing-viz-9.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cluster-based-on-gene-counts-and-visualize-cells" class="section level2">
<h2>Cluster based on gene counts and visualize cells</h2>
<pre class="r"><code>DefaultAssay(so) &lt;- &quot;RNA&quot;
# Extract the singlets
subs &lt;- subset(so, idents = &quot;Singlet&quot;)
subs$HTO_classification %&gt;% table</code></pre>
<pre><code>.
B0253 B0254 B0257 
  830   939   998 </code></pre>
<pre class="r"><code># Select the top 1000 most variable features
subs &lt;- FindVariableFeatures(subs, selection.method = &quot;mean.var.plot&quot;)
# Scaling RNA data, we only scale the variable features here for efficiency
subs &lt;- ScaleData(subs, features = VariableFeatures(subs))
# Run PCA
subs &lt;- RunPCA(subs, features = VariableFeatures(subs))

# We select the top 10 PCs for clustering and tSNE based on PCElbowPlot
subs &lt;- FindNeighbors(subs, reduction = &quot;pca&quot;, dims = 1:10)
subs &lt;- FindClusters(subs, resolution = 0.6, verbose = FALSE)
subs &lt;- RunTSNE(subs, reduction = &quot;pca&quot;, dims = 1:10)
subs &lt;- RunUMAP(subs, reduction = &quot;pca&quot;, dims = 1:10)</code></pre>
<pre><code>Warning: The default method for RunUMAP has changed from calling Python UMAP via reticulate to the R-native UWOT using the cosine metric
To use Python UMAP via reticulate, set umap.method to &#39;umap-learn&#39; and metric to &#39;correlation&#39;
This message will be shown once per session</code></pre>
<pre class="r"><code># Projecting singlet identities on TSNE visualization
DimPlot(subs, group.by = &quot;HTO_classification&quot;, reduction = &quot;tsne&quot;)</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/gene-count-clustering-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>DimPlot(subs, group.by = &quot;HTO_classification&quot;, reduction = &quot;umap&quot;)</code></pre>
<p><img src="figure/CH-test-01-preprocessing.Rmd/gene-count-clustering-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="save-data-to-rds" class="section level2">
<h2>Save data to RDS</h2>
<pre class="r"><code>saveRDS(sce, file.path(&quot;output&quot;, &quot;CH-test-01-preprocessing.rds&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/local/R/R-4.0.5/lib/libRblas.so
LAPACK: /usr/local/R/R-4.0.5/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] dplyr_1.0.2                 viridis_0.5.1              
 [3] viridisLite_0.3.0           scales_1.1.1               
 [5] SeuratObject_4.0.1          Seurat_4.0.1               
 [7] readxl_1.3.1                scater_1.16.2              
 [9] ggplot2_3.3.2               BiocParallel_1.22.0        
[11] DropletUtils_1.8.0          SingleCellExperiment_1.10.1
[13] SummarizedExperiment_1.18.1 DelayedArray_0.14.0        
[15] matrixStats_0.56.0          Biobase_2.48.0             
[17] GenomicRanges_1.40.0        GenomeInfoDb_1.24.2        
[19] IRanges_2.22.2              S4Vectors_0.26.1           
[21] BiocGenerics_0.34.0         workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] backports_1.1.9           plyr_1.8.6               
  [3] igraph_1.2.5              lazyeval_0.2.2           
  [5] splines_4.0.5             listenv_0.8.0            
  [7] scattermore_0.7           digest_0.6.25            
  [9] htmltools_0.5.0           magrittr_1.5             
 [11] tensor_1.5                cluster_2.1.0            
 [13] ROCR_1.0-11               limma_3.44.3             
 [15] globals_0.12.5            R.utils_2.9.2            
 [17] spatstat.sparse_2.0-0     colorspace_1.4-1         
 [19] rappdirs_0.3.1            ggrepel_0.8.2            
 [21] xfun_0.15                 crayon_1.3.4             
 [23] RCurl_1.98-1.3            jsonlite_1.7.2           
 [25] spatstat.data_2.1-0       survival_3.2-3           
 [27] zoo_1.8-8                 glue_1.4.2               
 [29] polyclip_1.10-0           gtable_0.3.0             
 [31] zlibbioc_1.34.0           XVector_0.28.0           
 [33] leiden_0.3.3              BiocSingular_1.4.0       
 [35] Rhdf5lib_1.10.0           future.apply_1.6.0       
 [37] HDF5Array_1.16.1          abind_1.4-5              
 [39] edgeR_3.30.3              miniUI_0.1.1.1           
 [41] Rcpp_1.0.5                isoband_0.2.2            
 [43] xtable_1.8-4              reticulate_1.16          
 [45] spatstat.core_2.1-2       dqrng_0.2.1              
 [47] rsvd_1.0.3                htmlwidgets_1.5.1        
 [49] httr_1.4.2                RColorBrewer_1.1-2       
 [51] ellipsis_0.3.1            ica_1.0-2                
 [53] farver_2.0.3              pkgconfig_2.0.3          
 [55] R.methodsS3_1.8.0         uwot_0.1.10              
 [57] deldir_0.2-10             locfit_1.5-9.4           
 [59] labeling_0.3              tidyselect_1.1.0         
 [61] rlang_0.4.10              reshape2_1.4.4           
 [63] later_1.1.0.1             munsell_0.5.0            
 [65] cellranger_1.1.0          tools_4.0.5              
 [67] generics_0.0.2            ggridges_0.5.2           
 [69] evaluate_0.14             stringr_1.4.0            
 [71] fastmap_1.0.1             goftest_1.2-2            
 [73] yaml_2.2.1                knitr_1.29               
 [75] fs_1.5.0                  fitdistrplus_1.1-1       
 [77] purrr_0.3.4               RANN_2.6.1               
 [79] nlme_3.1-148              pbapply_1.4-2            
 [81] future_1.17.0             whisker_0.4              
 [83] mime_0.9                  R.oo_1.23.0              
 [85] compiler_4.0.5            beeswarm_0.2.3           
 [87] plotly_4.9.2.1            png_0.1-7                
 [89] spatstat.utils_2.1-0      tibble_3.0.3             
 [91] stringi_1.4.6             RSpectra_0.16-0          
 [93] lattice_0.20-41           Matrix_1.3-3             
 [95] vctrs_0.3.4               pillar_1.4.6             
 [97] lifecycle_1.0.0           spatstat.geom_2.1-0      
 [99] lmtest_0.9-37             RcppAnnoy_0.0.18         
[101] BiocNeighbors_1.6.0       data.table_1.12.8        
[103] cowplot_1.0.0             bitops_1.0-6             
[105] irlba_2.3.3               httpuv_1.5.4             
[107] patchwork_1.0.1           R6_2.4.1                 
[109] promises_1.1.1            KernSmooth_2.23-17       
[111] gridExtra_2.3             vipor_0.4.5              
[113] codetools_0.2-16          MASS_7.3-51.6            
[115] rhdf5_2.32.2              rprojroot_1.3-2          
[117] withr_2.4.1               sctransform_0.3.2        
[119] GenomeInfoDbData_1.2.3    mgcv_1.8-31              
[121] beachmat_2.4.0            rpart_4.1-15             
[123] grid_4.0.5                tidyr_1.1.0              
[125] rmarkdown_2.3             DelayedMatrixStats_1.10.1
[127] Rtsne_0.15                git2r_0.27.1             
[129] shiny_1.5.0               ggbeeswarm_0.6.0         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
