<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Katharina Hembach" />


<title>Integration NSC from Lam et al.</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">neural_scRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/khembach/neural_scRNAseq">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Integration NSC from Lam et al.</h1>
<h4 class="author">Katharina Hembach</h4>
<h4 class="date">7/3/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-07-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>neural_scRNAseq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200522code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200522)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200522code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200522)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtreed8bd3397b3cf440d3c7ec628b677f7858c001011targetblankd8bd339a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/khembach/neural_scRNAseq/tree/d8bd3397b3cf440d3c7ec628b677f7858c001011" target="_blank">d8bd339</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtreed8bd3397b3cf440d3c7ec628b677f7858c001011targetblankd8bd339a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/khembach/neural_scRNAseq/tree/d8bd3397b3cf440d3c7ec628b677f7858c001011" target="_blank">d8bd339</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    ._Rplots.pdf
    Ignored:    .__workflowr.yml
    Ignored:    ._neural_scRNAseq.Rproj
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/._.DS_Store
    Ignored:    analysis/._01-preprocessing.Rmd
    Ignored:    analysis/._01-preprocessing.html
    Ignored:    analysis/._02.1-SampleQC.Rmd
    Ignored:    analysis/._03-filtering.Rmd
    Ignored:    analysis/._04-clustering.Rmd
    Ignored:    analysis/._04-clustering.knit.md
    Ignored:    analysis/._04.1-cell_cycle.Rmd
    Ignored:    analysis/._05-annotation.Rmd
    Ignored:    analysis/._Lam-01-NSC_integration.Rmd
    Ignored:    analysis/._Lam-02-NSC_annotation.Rmd
    Ignored:    analysis/._NSC-1-clustering.Rmd
    Ignored:    analysis/._NSC-2-annotation.Rmd
    Ignored:    analysis/.__site.yml
    Ignored:    analysis/._additional_filtering.Rmd
    Ignored:    analysis/._additional_filtering_clustering.Rmd
    Ignored:    analysis/._index.Rmd
    Ignored:    analysis/01-preprocessing_cache/
    Ignored:    analysis/02-1-SampleQC_cache/
    Ignored:    analysis/02-quality_control_cache/
    Ignored:    analysis/02.1-SampleQC_cache/
    Ignored:    analysis/03-filtering_cache/
    Ignored:    analysis/04-clustering_cache/
    Ignored:    analysis/04.1-cell_cycle_cache/
    Ignored:    analysis/05-annotation_cache/
    Ignored:    analysis/Lam-02-NSC_annotation_cache/
    Ignored:    analysis/NSC-1-clustering_cache/
    Ignored:    analysis/NSC-2-annotation_cache/
    Ignored:    analysis/additional_filtering_cache/
    Ignored:    analysis/additional_filtering_clustering_cache/
    Ignored:    analysis/sample5_QC_cache/
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._.smbdeleteAAA17ed8b4b
    Ignored:    data/._Lam_figure2_markers.R
    Ignored:    data/._known_NSC_markers.R
    Ignored:    data/._known_cell_type_markers.R
    Ignored:    data/._metadata.csv
    Ignored:    data/data_sushi/
    Ignored:    data/filtered_feature_matrices/
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/Lam-01-clustering.rds
    Ignored:    output/NSC_1_clustering.rds
    Ignored:    output/additional_filtering.rds
    Ignored:    output/figures/
    Ignored:    output/sce_01_preprocessing.rds
    Ignored:    output/sce_02_quality_control.rds
    Ignored:    output/sce_03_filtering.rds
    Ignored:    output/sce_preprocessing.rds
    Ignored:    output/so_04_1_cell_cycle.rds
    Ignored:    output/so_04_clustering.rds
    Ignored:    output/so_additional_filtering_clustering.rds

Untracked files:
    Untracked:  Rplots.pdf
    Untracked:  analysis/additional_filtering.Rmd
    Untracked:  analysis/additional_filtering_clustering.Rmd
    Untracked:  analysis/sample5_QC.Rmd
    Untracked:  analysis/tabsets.Rmd
    Untracked:  data/Homo_sapiens.GRCh38.98.sorted.gtf
    Untracked:  data/Lam_et_al/
    Untracked:  data/Lam_figure2_markers.R
    Untracked:  scripts/

Unstaged changes:
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Lam-01-NSC_integration.Rmd</code>) and HTML (<code>docs/Lam-01-NSC_integration.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/d8bd3397b3cf440d3c7ec628b677f7858c001011/analysis/Lam-01-NSC_integration.Rmd" target="_blank">d8bd339</a>
</td>
<td>
khembach
</td>
<td>
2020-07-08
</td>
<td>
NSC integration with NES from Lam et al.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>library(cowplot)
library(ggplot2)
library(Seurat)
library(SingleCellExperiment)
library(stringr)
library(Seurat)
library(rtracklayer)
library(future)
library(biomaRt)
library(dplyr)
library(data.table)</code></pre>
<pre class="r"><code># increase future&#39;s maximum allowed size of exported globals to 4GB 
# the default is 2GB
options(future.globals.maxSize = 4096 * 1024 ^ 2)
# change the current plan to access parallelization
plan(&quot;multiprocess&quot;, workers = 20)</code></pre>
</div>
<div id="load-our-nsc-data" class="section level2">
<h2>Load our NSC data</h2>
<pre class="r"><code># sce &lt;- readRDS(file.path(&quot;output&quot;, &quot;sce_03_filtering.rds&quot;))
## our NSC from sample 1 and 2
# so &lt;- readRDS(file.path(&quot;output&quot;, &quot;NSC_1_clustering.rds&quot;))

sce &lt;- readRDS(file.path(&quot;output&quot;, &quot;sce_03_filtering.rds&quot;))

## subset the two NSC samples
sce &lt;- sce[,colData(sce)$sample_id %in% c(&quot;1NSC&quot;, &quot;2NSC&quot;)]
sce$sample_id &lt;- droplevels(sce$sample_id)
# ## we filter genes and require &gt; 1 count in at least 20 cells
# sce &lt;- sce[rowSums(counts(sce) &gt; 1) &gt;= 20, ]
# dim(sce)

# create SeuratObject
so &lt;- CreateSeuratObject(
    counts = counts(sce),
    meta.data = data.frame(colData(sce)),
    project = &quot;neural_cultures&quot;)</code></pre>
</div>
<div id="load-nes-data-and-map-symbold-to-ensembl-ids" class="section level2">
<h2>Load NES data and map symbold to Ensembl IDs</h2>
<pre class="r"><code>nes_meta &lt;- read.table(file.path(&quot;data&quot;, &quot;Lam_et_al&quot;, &quot;figure2&quot;, &quot;Figure_2_metadata.NES.Healthy.Cell.lines.healthy_NES.7.clusters.final.figure.txt&quot;))

nes_counts &lt;- read.table(file.path(&quot;data&quot;, &quot;Lam_et_al&quot;, &quot;figure2&quot;, &quot;Figure_2_NES.Healthy.Cell.lines.healthy_NES.7.clusters.final.figure.txt&quot;))

sce_nes &lt;- SingleCellExperiment(list(counts=nes_counts), colData = nes_meta)
dim(sce_nes)</code></pre>
<pre><code>[1] 14086   768</code></pre>
<pre class="r"><code>## I think Lam et al. used UCSC hg19 gene annotations for their analysis.
## --&gt; we map the gene symbols to hg19 Ensembl IDs
grch37 &lt;- biomaRt::useMart(biomart=&quot;ENSEMBL_MART_ENSEMBL&quot;, host=&quot;grch37.ensembl.org&quot;, 
                  path=&quot;/biomart/martservice&quot;, dataset=&quot;hsapiens_gene_ensembl&quot;)
res &lt;- getBM(attributes=c(&#39;hgnc_symbol&#39;, &#39;ensembl_gene_id&#39;), 
      filters = &#39;hgnc_symbol&#39;, 
      values = rownames(sce_nes), 
      mart = grch37)
table(rownames(sce_nes) %in% res$hgnc_symbol)</code></pre>
<pre><code>
FALSE  TRUE 
 1450 12636 </code></pre>
<pre class="r"><code>res_split &lt;- split(res$ensembl_gene_id, res$hgnc_symbol)

# we match our Ensembl IDs to the biomaRt IDs and only keep rows from Lam et al. 
# that match one of our IDs
a &lt;- res %&gt;% dplyr::left_join(data.frame(rowData(sce)), 
                              by = c(&quot;ensembl_gene_id&quot; = &quot;ensembl_id&quot;))
## if a symbols has multiple IDs, we keep the ID with the same symbol in our dataset
## find all duplicates
dups &lt;- duplicated(a$hgnc_symbol) | duplicated(a$hgnc_symbol, fromLast=TRUE)
## check the symbol in our dataset and remove the duplicate with wrong symbol
b_split &lt;- split(a[dups,], a[dups,]$hgnc_symbol)

## if GRCh38 symbol matches the hgnc_symbol, we keep the corresponding Ensembl ID
## otherwise we only keep the gene symbol without an ID
resolve_dups &lt;- function(b){
  res &lt;- b[which(b$hgnc_symbol == b$symbol),]
  if (nrow(res) != 1){
    res &lt;- b[1,]
    res[, c(&quot;ensembl_gene_id&quot;, &quot;symbol&quot;)] &lt;- &quot;&quot;
  }
  res
}

b_resolved &lt;- lapply(b_split, resolve_dups)
b_resolved &lt;- rbindlist(b_resolved)
id_map &lt;- rbind(a[!dups,], b_resolved)
id_map$symbol &lt;- NULL
colnames(id_map) &lt;- c(&quot;symbol&quot;, &quot;ensembl_id&quot;)

## Remove the features for which we don&#39;t have an ensembl ID
sce_nes &lt;- sce_nes[rownames(sce_nes) %in% id_map$symbol,]
dim(sce_nes)</code></pre>
<pre><code>[1] 12636   768</code></pre>
<pre class="r"><code>## We have to use identical gene symbols for all datasets if we want to integrate the data.
## We replace the Lam et al. gene symbols with GRCh38 symbols for all features 
## where the Ensembl IDs could be mapped.
gtf &lt;- import(file.path(&quot;data&quot;, &quot;Homo_sapiens.GRCh38.98.sorted.gtf&quot;))
gene &lt;- gtf[gtf$type == &quot;gene&quot;]
m &lt;- match(id_map$symbol, gene$gene_name)
id_map[!is.na(m),]$symbol &lt;- gene$gene_name[m[!is.na(m)]]

## adjust row and colData
colData(sce_nes)$sample_id &lt;- factor(&quot;NES&quot;)
rowData(sce_nes) &lt;- id_map[match(rownames(sce_nes), id_map$symbol),]
rownames(sce_nes) &lt;- with(rowData(sce_nes), 
                               paste(ensembl_id, symbol, sep = &quot;.&quot;))
sce_nes$group_id &lt;- &quot;NES&quot;
## how many genes could be mapped between datasets?
rownames(sce_nes) %in% rownames(sce) %&gt;% table</code></pre>
<pre><code>.
FALSE  TRUE 
 1867 10769 </code></pre>
<pre class="r"><code>## The data already comes with the number of RNAs and features
names(colData(sce_nes))[names(colData(sce_nes)) == &#39;BARCODE&#39;] &lt;- &#39;barcode&#39;
names(colData(sce_nes))[names(colData(sce_nes)) == &#39;nCount_RNA&#39;] &lt;- &#39;sum&#39;
names(colData(sce_nes))[names(colData(sce_nes)) == &#39;nFeature_RNA&#39;] &lt;- &#39;detected&#39;

# ## we filter genes and require &gt; 1 count in at least 5 cells
# sce_nes &lt;- sce_nes[rowSums(counts(sce_nes) &gt; 1) &gt;= 5, ]
# dim(sce_nes)

so_nes &lt;- CreateSeuratObject(
    counts = counts(sce_nes),
    meta.data = data.frame(colData(sce_nes)),
    project = &quot;neural_cultures&quot;)</code></pre>
<!-- # ## Ensembl GTF annotations -->
<!-- # # /home/Shared_taupo/data/annotation/Human/Ensembl_GRCh37.71/gtf/Homo_sapiens.GRCh37.71.gtf -->
<!-- # gtf <- import("/home/Shared_taupo/data/annotation/Human/GRCH37/gencode_v19/gencode.v19.annotation.gtf") -->
<!-- # # gtf <- import("/home/Shared_taupo/data/annotation/Human/Ensembl_GRCh37.71/gtf/Homo_sapiens.GRCh37.71.gtf") -->
<!-- # gene <- gtf[gtf$type == "gene"] -->
<!-- # map <- data.frame(gene_id = gtf$gene_id, gene_name = gtf$gene_name) -->
<!-- # map <- unique(map) -->
<!-- # -->
<!-- # ## we map the gene symbols to Ensembl IDs -->
<!-- # m <- match(rownames(sce_nes), map$gene_name) -->
<!-- # table(is.na(m)) -->
<!-- # rowData(sce_nes)$symbol <- rownames(sce_nes) -->
<!-- # rowData(sce_nes)$ensembl_id <- gene$gene_id[m] -->
<!-- # ## GRCH37/gencode_v19 -->
<!-- # FALSE  TRUE -->
<!-- # 12616  1470 -->
<!-- # #GRCh37.71 -->
<!-- # FALSE  TRUE -->
<!-- # 12423  1663 -->
</div>
<div id="normalisation" class="section level2">
<h2>Normalisation</h2>
<pre class="r"><code># split by sample
cells_by_sample &lt;- split(colnames(sce), sce$sample_id)
so &lt;- lapply(cells_by_sample, function(i) subset(so, cells = i))
## add the NES
so[[&quot;NES&quot;]] &lt;- so_nes

## log normalize the data using a scaling factor of 10000
so &lt;- lapply(so, NormalizeData, verbose = FALSE, scale.factor = 10000, 
             normalization.method = &quot;LogNormalize&quot;)</code></pre>
<pre class="r"><code>## Identify the top 2000 genes with high cell-to-cell variation
so &lt;- lapply(so, FindVariableFeatures, nfeatures = 2000, 
    selection.method = &quot;vst&quot;, verbose = FALSE)

## Plot variable features 
for (i in names(so)) {
  # Identify the 10 most highly variable genes
  top10 &lt;- head(VariableFeatures(so[[i]]), 10)
  p &lt;- VariableFeaturePlot(so[[i]])
  p &lt;- LabelPoints(plot = p, points = top10, 
              labels = str_split(top10, &quot;\\.&quot;, simplify = TRUE)[,2], 
              repel = TRUE)
  print(p)
}</code></pre>
<p><img src="figure/Lam-01-NSC_integration.Rmd/variable-features-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lam-01-NSC_integration.Rmd/variable-features-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Lam-01-NSC_integration.Rmd/variable-features-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># find anchors &amp; integrate
as &lt;- FindIntegrationAnchors(so, verbose = FALSE)
so &lt;- IntegrateData(anchorset = as, dims = seq_len(30), verbose = FALSE)

## We scale the data so that mean expression is 0 and variance is 1, across cells
## We also regress out the number of UMIs. 
## We don&#39;t have mitochondrial genes for the NES
DefaultAssay(so) &lt;- &quot;integrated&quot;
so &lt;- ScaleData(so, verbose = FALSE, vars.to.regress = &quot;sum&quot;)</code></pre>
</div>
<div id="dimension-reduction" class="section level2">
<h2>Dimension reduction</h2>
<p>We perform dimension reduction with t-SNE and UMAP based on PCA results.</p>
<pre class="r"><code>so &lt;- RunPCA(so, npcs = 30, verbose = FALSE)
so &lt;- RunTSNE(so, reduction = &quot;pca&quot;, dims = seq_len(20),
    seed.use = 1, do.fast = TRUE, verbose = FALSE)
so &lt;- RunUMAP(so, reduction = &quot;pca&quot;, dims = seq_len(20),
    seed.use = 1, verbose = FALSE)</code></pre>
<div id="plot-pca-results" class="section level3">
<h3>Plot PCA results</h3>
<pre class="r"><code># top genes that are associated with the first two PCs
VizDimLoadings(so, dims = 1:2, reduction = &quot;pca&quot;)</code></pre>
<p><img src="figure/Lam-01-NSC_integration.Rmd/unnamed-chunk-2-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## PCA plot 
DimPlot(so, reduction = &quot;pca&quot;, group.by = &quot;sample_id&quot;)</code></pre>
<p><img src="figure/Lam-01-NSC_integration.Rmd/unnamed-chunk-3-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code># elbow plot with the ranking of PCs based on the % of variance explained
ElbowPlot(so, ndims = 30)</code></pre>
<p><img src="figure/Lam-01-NSC_integration.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## heatmaps of the top 20 PCs and the 500 most extreme cells for each component
DimHeatmap(so, dims = 1:20, cells = 500, balanced = TRUE, nfeatures = 20 )</code></pre>
<p><img src="figure/Lam-01-NSC_integration.Rmd/unnamed-chunk-5-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="clustering" class="section level2">
<h2>Clustering</h2>
<p>We cluster the cells using the reduced PCA dimensions.</p>
<pre class="r"><code>so &lt;- FindNeighbors(so, reduction = &quot;pca&quot;, dims = seq_len(20), verbose = FALSE)
for (res in c(0.1, 0.2, 0.4, 0.8, 1, 1.2, 2))
    so &lt;- FindClusters(so, resolution = res, random.seed = 1, verbose = FALSE)</code></pre>
</div>
<div id="dimension-reduction-plots" class="section level2">
<h2>Dimension reduction plots</h2>
<p>We plot the dimension reduction (DR) and color by sample, group and cluster ID</p>
<pre class="r"><code>thm &lt;- theme(aspect.ratio = 1, legend.position = &quot;none&quot;)
ps &lt;- lapply(c(&quot;sample_id&quot;, &quot;group_id&quot;, &quot;ident&quot;), function(u) {
    p1 &lt;- DimPlot(so, reduction = &quot;tsne&quot;, group.by = u) + thm
    p2 &lt;- DimPlot(so, reduction = &quot;umap&quot;, group.by = u)
    lgd &lt;- get_legend(p2)
    p2 &lt;- p2 + thm
    list(p1, p2, lgd)
    plot_grid(p1, p2, lgd, nrow = 1,
        rel_widths = c(1, 1, 0.5))
})
plot_grid(plotlist = ps, ncol = 1)</code></pre>
<p><img src="figure/Lam-01-NSC_integration.Rmd/dr-plots-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="qc-on-dr-plots" class="section level2 tabset">
<h2>QC on DR plots</h2>
<pre class="r"><code>cs &lt;- sample(colnames(so), 1e4) ## subsample cells
.plot_features &lt;- function(so, dr, id) {
    FeaturePlot(so, cells = cs, features = id, reduction = dr, pt.size = 0.4, 
                cols = c(&quot;grey&quot;, &quot;blue&quot;)) +
        guides(col = guide_legend(nrow = 11, 
            override.aes = list(size = 3, alpha = 1))) +
        theme_void() + theme(aspect.ratio = 1)
}
ids &lt;- c(&quot;sum&quot;, &quot;detected&quot;, &quot;subsets_Mt_percent&quot;)
for (id in ids) {
    cat(&quot;## &quot;, id, &quot;\n&quot;)
    p1 &lt;- .plot_features(so, &quot;tsne&quot;, id)
    lgd &lt;- get_legend(p1)
    p1 &lt;- p1 + theme(legend.position = &quot;none&quot;) + ggtitle(&quot;tSNE&quot;)
    p2 &lt;- .plot_features(so, &quot;umap&quot;, id) + theme(legend.position = &quot;none&quot;) + 
      ggtitle(&quot;UMAP&quot;)
    ps &lt;- plot_grid(plotlist = list(p1, p2), nrow = 1)
    p &lt;- plot_grid(ps, lgd, nrow = 1, rel_widths = c(1, 0.2))
    print(p)
    cat(&quot;\n\n&quot;)
}</code></pre>
</div>
<div id="sum" class="section level2">
<h2>sum</h2>
<p><img src="figure/Lam-01-NSC_integration.Rmd/DR-QC-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="detected" class="section level2">
<h2>detected</h2>
<p><img src="figure/Lam-01-NSC_integration.Rmd/DR-QC-2.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="subsets_mt_percent" class="section level2">
<h2>subsets_Mt_percent</h2>
<p><img src="figure/Lam-01-NSC_integration.Rmd/DR-QC-3.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="save-seurat-object-to-rds" class="section level2">
<h2>Save Seurat object to RDS</h2>
<pre class="r"><code>saveRDS(so, file.path(&quot;output&quot;, &quot;Lam-01-clustering.rds&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.6 LTS

Matrix products: default
BLAS:   /usr/local/R/R-4.0.0/lib/libRblas.so
LAPACK: /usr/local/R/R-4.0.0/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] HDF5Array_1.16.1            rhdf5_2.32.2               
 [3] data.table_1.12.8           dplyr_1.0.0                
 [5] biomaRt_2.44.1              future_1.17.0              
 [7] rtracklayer_1.48.0          stringr_1.4.0              
 [9] SingleCellExperiment_1.10.1 SummarizedExperiment_1.18.1
[11] DelayedArray_0.14.0         matrixStats_0.56.0         
[13] Biobase_2.48.0              GenomicRanges_1.40.0       
[15] GenomeInfoDb_1.24.2         IRanges_2.22.2             
[17] S4Vectors_0.26.1            BiocGenerics_0.34.0        
[19] Seurat_3.1.5                ggplot2_3.3.2              
[21] cowplot_1.0.0               workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] backports_1.1.8          BiocFileCache_1.12.0     plyr_1.8.6              
  [4] igraph_1.2.5             lazyeval_0.2.2           splines_4.0.0           
  [7] BiocParallel_1.22.0      listenv_0.8.0            digest_0.6.25           
 [10] htmltools_0.5.0          magrittr_1.5             memoise_1.1.0           
 [13] cluster_2.1.0            ROCR_1.0-11              globals_0.12.5          
 [16] Biostrings_2.56.0        askpass_1.1              prettyunits_1.1.1       
 [19] colorspace_1.4-1         blob_1.2.1               rappdirs_0.3.1          
 [22] ggrepel_0.8.2            xfun_0.15                crayon_1.3.4            
 [25] RCurl_1.98-1.2           jsonlite_1.7.0           survival_3.2-3          
 [28] zoo_1.8-8                ape_5.4                  glue_1.4.1              
 [31] gtable_0.3.0             zlibbioc_1.34.0          XVector_0.28.0          
 [34] leiden_0.3.3             Rhdf5lib_1.10.0          future.apply_1.6.0      
 [37] scales_1.1.1             DBI_1.1.0                Rcpp_1.0.4.6            
 [40] viridisLite_0.3.0        progress_1.2.2           reticulate_1.16         
 [43] bit_1.1-15.2             rsvd_1.0.3               tsne_0.1-3              
 [46] htmlwidgets_1.5.1        httr_1.4.1               RColorBrewer_1.1-2      
 [49] ellipsis_0.3.1           ica_1.0-2                farver_2.0.3            
 [52] pkgconfig_2.0.3          XML_3.99-0.4             uwot_0.1.8              
 [55] dbplyr_1.4.4             tidyselect_1.1.0         labeling_0.3            
 [58] rlang_0.4.6              reshape2_1.4.4           later_1.1.0.1           
 [61] AnnotationDbi_1.50.1     munsell_0.5.0            tools_4.0.0             
 [64] generics_0.0.2           RSQLite_2.2.0            ggridges_0.5.2          
 [67] evaluate_0.14            yaml_2.2.1               knitr_1.29              
 [70] bit64_0.9-7              fs_1.4.2                 fitdistrplus_1.1-1      
 [73] purrr_0.3.4              RANN_2.6.1               pbapply_1.4-2           
 [76] nlme_3.1-148             whisker_0.4              compiler_4.0.0          
 [79] plotly_4.9.2.1           curl_4.3                 png_0.1-7               
 [82] tibble_3.0.1             stringi_1.4.6            RSpectra_0.16-0         
 [85] lattice_0.20-41          Matrix_1.2-18            vctrs_0.3.1             
 [88] pillar_1.4.4             lifecycle_0.2.0          lmtest_0.9-37           
 [91] RcppAnnoy_0.0.16         bitops_1.0-6             irlba_2.3.3             
 [94] httpuv_1.5.4             patchwork_1.0.1          R6_2.4.1                
 [97] promises_1.1.1           KernSmooth_2.23-17       gridExtra_2.3           
[100] codetools_0.2-16         MASS_7.3-51.6            assertthat_0.2.1        
[103] openssl_1.4.2            rprojroot_1.3-2          withr_2.2.0             
[106] GenomicAlignments_1.24.0 sctransform_0.2.1        Rsamtools_2.4.0         
[109] GenomeInfoDbData_1.2.3   hms_0.5.3                grid_4.0.0              
[112] tidyr_1.1.0              rmarkdown_2.3            Rtsne_0.15              
[115] git2r_0.27.1            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
