<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Katharina Hembach" />


<title>Organoid Stage integration cluster analysis</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">neural_scRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/khembach/neural_scRNAseq">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Organoid Stage integration cluster analysis</h1>
<h4 class="author">Katharina Hembach</h4>
<h4 class="date">9/18/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-09-21
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>neural_scRNAseq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200522code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200522)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200522code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200522)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
cluster-before-after-organoid-integration
</li>
<li>
integration-dim-red
</li>
<li>
load-data
</li>
<li>
load-libs
</li>
<li>
scran-findMarkers
</li>
<li>
scran-markers-heatmap
</li>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
unnamed-chunk-1
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-3
</li>
<li>
unnamed-chunk-4
</li>
<li>
unnamed-chunk-5
</li>
<li>
unnamed-chunk-6
</li>
<li>
unnamed-chunk-7
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>organoid-05-stage_integration_cluster_analysis_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtree6d64ace722915edc9dfbe99d5853508d827c4fa5targetblank6d64acea"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/khembach/neural_scRNAseq/tree/6d64ace722915edc9dfbe99d5853508d827c4fa5" target="_blank">6d64ace</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtree6d64ace722915edc9dfbe99d5853508d827c4fa5targetblank6d64acea" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/khembach/neural_scRNAseq/tree/6d64ace722915edc9dfbe99d5853508d827c4fa5" target="_blank">6d64ace</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    ._Rplots.pdf
    Ignored:    ._Rplots_largeViz.pdf
    Ignored:    ._Rplots_separate.pdf
    Ignored:    .__workflowr.yml
    Ignored:    ._neural_scRNAseq.Rproj
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/._.DS_Store
    Ignored:    analysis/._01-preprocessing.Rmd
    Ignored:    analysis/._01-preprocessing.html
    Ignored:    analysis/._02.1-SampleQC.Rmd
    Ignored:    analysis/._03-filtering.Rmd
    Ignored:    analysis/._04-clustering.Rmd
    Ignored:    analysis/._04-clustering.knit.md
    Ignored:    analysis/._04.1-cell_cycle.Rmd
    Ignored:    analysis/._05-annotation.Rmd
    Ignored:    analysis/._Lam-0-NSC_no_integration.Rmd
    Ignored:    analysis/._Lam-01-NSC_integration.Rmd
    Ignored:    analysis/._Lam-02-NSC_annotation.Rmd
    Ignored:    analysis/._NSC-1-clustering.Rmd
    Ignored:    analysis/._NSC-2-annotation.Rmd
    Ignored:    analysis/.__site.yml
    Ignored:    analysis/._additional_filtering.Rmd
    Ignored:    analysis/._additional_filtering_clustering.Rmd
    Ignored:    analysis/._index.Rmd
    Ignored:    analysis/._organoid-01-clustering.Rmd
    Ignored:    analysis/._organoid-02-integration.Rmd
    Ignored:    analysis/._organoid-03-cluster_analysis.Rmd
    Ignored:    analysis/._organoid-04-group_integration.Rmd
    Ignored:    analysis/._organoid-04-stage_integration.Rmd
    Ignored:    analysis/._organoid-05-group_integration_cluster_analysis.Rmd
    Ignored:    analysis/._organoid-05-stage_integration_cluster_analysis.Rmd
    Ignored:    analysis/._organoid-06-1-prepare-sce.Rmd
    Ignored:    analysis/._organoid-06-conos-analysis-Seurat.Rmd
    Ignored:    analysis/._organoid-06-conos-analysis-function.Rmd
    Ignored:    analysis/._organoid-06-conos-analysis.Rmd
    Ignored:    analysis/._organoid-06-group-integration-conos-analysis.Rmd
    Ignored:    analysis/._organoid-07-conos-visualization.Rmd
    Ignored:    analysis/._organoid-07-group-integration-conos-visualization.Rmd
    Ignored:    analysis/._organoid-08-conos-comparison.Rmd
    Ignored:    analysis/._organoid-0x-sample_integration.Rmd
    Ignored:    analysis/01-preprocessing_cache/
    Ignored:    analysis/02-1-SampleQC_cache/
    Ignored:    analysis/02-quality_control_cache/
    Ignored:    analysis/02.1-SampleQC_cache/
    Ignored:    analysis/03-filtering_cache/
    Ignored:    analysis/04-clustering_cache/
    Ignored:    analysis/04.1-cell_cycle_cache/
    Ignored:    analysis/05-annotation_cache/
    Ignored:    analysis/Lam-01-NSC_integration_cache/
    Ignored:    analysis/Lam-02-NSC_annotation_cache/
    Ignored:    analysis/NSC-1-clustering_cache/
    Ignored:    analysis/NSC-2-annotation_cache/
    Ignored:    analysis/additional_filtering_cache/
    Ignored:    analysis/additional_filtering_clustering_cache/
    Ignored:    analysis/organoid-01-clustering_cache/
    Ignored:    analysis/organoid-02-integration_cache/
    Ignored:    analysis/organoid-03-cluster_analysis_cache/
    Ignored:    analysis/organoid-04-group_integration_cache/
    Ignored:    analysis/organoid-04-stage_integration_cache/
    Ignored:    analysis/organoid-05-group_integration_cluster_analysis_cache/
    Ignored:    analysis/organoid-05-stage_integration_cluster_analysis_cache/
    Ignored:    analysis/organoid-06-conos-analysis_cache/
    Ignored:    analysis/organoid-06-conos-analysis_test_cache/
    Ignored:    analysis/organoid-06-group-integration-conos-analysis_cache/
    Ignored:    analysis/organoid-07-conos-visualization_cache/
    Ignored:    analysis/organoid-07-group-integration-conos-visualization_cache/
    Ignored:    analysis/organoid-08-conos-comparison_cache/
    Ignored:    analysis/organoid-0x-sample_integration_cache/
    Ignored:    analysis/sample5_QC_cache/
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._.smbdeleteAAA17ed8b4b
    Ignored:    data/._Lam_figure2_markers.R
    Ignored:    data/._known_NSC_markers.R
    Ignored:    data/._known_cell_type_markers.R
    Ignored:    data/._metadata.csv
    Ignored:    data/data_sushi/
    Ignored:    data/filtered_feature_matrices/
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/._NSC_cluster1_marker_genes.txt
    Ignored:    output/._organoid_integration_cluster1_marker_genes.txt
    Ignored:    output/Lam-01-clustering.rds
    Ignored:    output/NSC_1_clustering.rds
    Ignored:    output/NSC_cluster1_marker_genes.txt
    Ignored:    output/NSC_cluster2_marker_genes.txt
    Ignored:    output/NSC_cluster3_marker_genes.txt
    Ignored:    output/NSC_cluster4_marker_genes.txt
    Ignored:    output/NSC_cluster5_marker_genes.txt
    Ignored:    output/NSC_cluster6_marker_genes.txt
    Ignored:    output/NSC_cluster7_marker_genes.txt
    Ignored:    output/additional_filtering.rds
    Ignored:    output/conos/
    Ignored:    output/conos_organoid-06-conos-analysis.rds
    Ignored:    output/conos_organoid-06-group-integration-conos-analysis.rds
    Ignored:    output/figures/
    Ignored:    output/organoid_integration_cluster10_marker_genes.txt
    Ignored:    output/organoid_integration_cluster11_marker_genes.txt
    Ignored:    output/organoid_integration_cluster12_marker_genes.txt
    Ignored:    output/organoid_integration_cluster13_marker_genes.txt
    Ignored:    output/organoid_integration_cluster14_marker_genes.txt
    Ignored:    output/organoid_integration_cluster15_marker_genes.txt
    Ignored:    output/organoid_integration_cluster16_marker_genes.txt
    Ignored:    output/organoid_integration_cluster17_marker_genes.txt
    Ignored:    output/organoid_integration_cluster1_marker_genes.txt
    Ignored:    output/organoid_integration_cluster2_marker_genes.txt
    Ignored:    output/organoid_integration_cluster3_marker_genes.txt
    Ignored:    output/organoid_integration_cluster4_marker_genes.txt
    Ignored:    output/organoid_integration_cluster5_marker_genes.txt
    Ignored:    output/organoid_integration_cluster6_marker_genes.txt
    Ignored:    output/organoid_integration_cluster7_marker_genes.txt
    Ignored:    output/organoid_integration_cluster8_marker_genes.txt
    Ignored:    output/organoid_integration_cluster9_marker_genes.txt
    Ignored:    output/sce_01_preprocessing.rds
    Ignored:    output/sce_02_quality_control.rds
    Ignored:    output/sce_03_filtering.rds
    Ignored:    output/sce_03_filtering_all_genes.rds
    Ignored:    output/sce_06-1-prepare-sce.rds
    Ignored:    output/sce_organoid-01-clustering.rds
    Ignored:    output/sce_preprocessing.rds
    Ignored:    output/so_04-group_integration.rds
    Ignored:    output/so_04-stage_integration.rds
    Ignored:    output/so_04_1_cell_cycle.rds
    Ignored:    output/so_04_clustering.rds
    Ignored:    output/so_0x-sample_integration.rds
    Ignored:    output/so_additional_filtering_clustering.rds
    Ignored:    output/so_integrated_organoid-02-integration.rds
    Ignored:    output/so_merged_organoid-02-integration.rds
    Ignored:    output/so_organoid-01-clustering.rds
    Ignored:    output/so_sample_organoid-01-clustering.rds

Untracked files:
    Untracked:  Rplots.pdf
    Untracked:  Rplots_largeViz.pdf
    Untracked:  Rplots_separate.pdf
    Untracked:  analysis/Lam-0-NSC_no_integration.Rmd
    Untracked:  analysis/additional_filtering.Rmd
    Untracked:  analysis/additional_filtering_clustering.Rmd
    Untracked:  analysis/organoid-06-conos-analysis-Seurat.Rmd
    Untracked:  analysis/organoid-06-conos-analysis-function.Rmd
    Untracked:  analysis/organoid-07-conos-visualization.Rmd
    Untracked:  analysis/organoid-07-group-integration-conos-visualization.Rmd
    Untracked:  analysis/organoid-08-conos-comparison.Rmd
    Untracked:  analysis/organoid-0x-sample_integration.Rmd
    Untracked:  analysis/sample5_QC.Rmd
    Untracked:  data/Homo_sapiens.GRCh38.98.sorted.gtf
    Untracked:  data/Kanton_et_al/
    Untracked:  data/Lam_et_al/
    Untracked:  scripts/

Unstaged changes:
    Modified:   analysis/05-annotation.Rmd
    Modified:   analysis/Lam-02-NSC_annotation.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/organoid-02-integration.Rmd
    Modified:   analysis/organoid-04-group_integration.Rmd
    Modified:   analysis/organoid-06-conos-analysis.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/organoid-05-stage_integration_cluster_analysis.Rmd</code>) and HTML (<code>docs/organoid-05-stage_integration_cluster_analysis.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/6d64ace722915edc9dfbe99d5853508d827c4fa5/analysis/organoid-05-stage_integration_cluster_analysis.Rmd" target="_blank">6d64ace</a>
</td>
<td>
khembach
</td>
<td>
2020-09-21
</td>
<td>
fix factor levels
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/khembach/neural_scRNAseq/ff520aefa753dff305aab2255f2cfae85084232f/docs/organoid-05-stage_integration_cluster_analysis.html" target="_blank">ff520ae</a>
</td>
<td>
khembach
</td>
<td>
2020-09-20
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/f91df7bcab4cf89c5ed6dc14c87cccf175555083/analysis/organoid-05-stage_integration_cluster_analysis.Rmd" target="_blank">f91df7b</a>
</td>
<td>
khembach
</td>
<td>
2020-09-20
</td>
<td>
cluster abundances of cell line and stage integration
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>library(ComplexHeatmap)
library(cowplot)
library(ggplot2)
library(dplyr)
library(muscat)
library(RColorBrewer)
library(Seurat)
library(SingleCellExperiment)
library(scran)
library(stringr)
library(viridis)</code></pre>
</div>
<div id="load-data-convert-to-sce" class="section level1">
<h1>Load data &amp; convert to SCE</h1>
<pre class="r"><code>so &lt;- readRDS(file.path(&quot;output&quot;, &quot;so_04-stage_integration.rds&quot;))
sce &lt;- as.SingleCellExperiment(so, assay = &quot;RNA&quot;)
colData(sce) &lt;- as.data.frame(colData(sce)) %&gt;% 
    mutate_if(is.character, as.factor) %&gt;% 
    DataFrame(row.names = colnames(sce))</code></pre>
</div>
<div id="cluster-sample-counts" class="section level1">
<h1>Cluster-sample counts</h1>
<pre class="r"><code># set cluster IDs to resolution 0.4 clustering
so &lt;- SetIdent(so, value = &quot;integrated_snn_res.0.4&quot;)
so@meta.data$cluster_id &lt;- Idents(so)
sce$cluster_id &lt;- Idents(so)
(n_cells &lt;- table(sce$cluster_id, sce$sample_id))</code></pre>
<pre><code>    
     1NSC 2NSC 3NC52 4NC52 5NC96 6NC96 h409B2_120d_org1 h409B2_128d_org1
  0    44   38  5980  4810  1611  2512               82               79
  1  3130 3184  1105   952   926  1105               31               48
  10  701  703    37    31     5     2               58               31
  11  436  446   287   254   226   241               50               98
  12    6    9     3     3     0     1               94              104
  13   99  107    41    33    22    19               37                0
  2  1922 1949   111    70    88    35              186               96
  3  1007  993   325   364   388   333              160                2
  4    56   41   374   356    89   155              345              492
  5     0    0    33    27     7     8              141              624
  6    43   34   129   315    90    94              349              585
  7   847  868    79    57    59    33               33               29
  8     0    0    10    32     2     3              699              592
  9    40   36   173   134    25    54              288              340
    
     h409B2_60d_org1 h409B2_67d_org1 h409B2_EB h409B2_iPSCs
  0              803             731         3            1
  1              126              72       362          690
  10             174             104        54          117
  11             115              72         7           18
  12              95              85         0            0
  13               5               9         0            0
  2              406             316       148          425
  3              156              62        60          123
  4              753             620         0            0
  5              592             315         0            0
  6              216             254        51          192
  7              108              79       170          376
  8              555             363         0            0
  9              488             382         0            1
    
     h409B2_neuroectoderm h409B2_neuroepithelium h409B2_Org_32d H9_128d_org1
  0                     1                      1            172          486
  1                   300                     79            240            3
  10                   61                     41            148           65
  11                   29                      8             38           36
  12                    0                      0             29          149
  13                    1                      9             11            0
  2                   196                    100            581          118
  3                   107                    143            268            5
  4                     0                      4            489          159
  5                     0                      0             70         1452
  6                    68                     24             40          894
  7                   123                     34            108           23
  8                     0                      0             27          182
  9                     0                      0            195          243
    
     H9_60d_org1 H9_67d_org1 H9_EB H9_iPSCs H9_neuroectoderm H9_neuroepithelium
  0          551         269    11        5                0                  0
  1           53          37  1065      845              300                106
  10         131         104   144      128              183                 78
  11          31          68    38       50              116                 38
  12         332         153     0        0                0                  0
  13           0          24     0        1                1                  0
  2          275         275   523      415              570                225
  3           12          94   199      132              409                296
  4          670         614     2        1               30                 36
  5         1252         534     0        0                0                  0
  6          219         534    31      314               85                 66
  7          118          69   432      466              172                 65
  8          713        1089     0        0                0                  0
  9          500         443     0        0                5                 10
    
     H9_Org_32d
  0         222
  1         104
  10        135
  11        181
  12         28
  13          4
  2         327
  3         437
  4         545
  5         131
  6         176
  7          92
  8          15
  9         257</code></pre>
</div>
<div id="relative-cluster-abundances" class="section level1">
<h1>Relative cluster-abundances</h1>
<pre class="r"><code>fqs &lt;- prop.table(n_cells, margin = 2)
mat &lt;- as.matrix(unclass(fqs))
Heatmap(mat,
    col = rev(brewer.pal(11, &quot;RdGy&quot;)[-6]),
    name = &quot;Frequency&quot;,
    cluster_rows = FALSE,
    cluster_columns = FALSE,
    row_names_side = &quot;left&quot;,
    row_title = &quot;cluster_id&quot;,
    column_title = &quot;sample_id&quot;,
    column_title_side = &quot;bottom&quot;,
    rect_gp = gpar(col = &quot;white&quot;),
    cell_fun = function(i, j, x, y, width, height, fill)
        grid.text(round(mat[j, i] * 100, 2), x = x, y = y, 
            gp = gpar(col = &quot;white&quot;, fontsize = 10)))</code></pre>
<p><img src="figure/organoid-05-stage_integration_cluster_analysis.Rmd/unnamed-chunk-2-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>n_cells &lt;- table(sce$sample_id, sce$cluster_id)
fqs &lt;- prop.table(n_cells, margin = 2)
mat &lt;- as.matrix(unclass(fqs))
Heatmap(mat,
    col = rev(brewer.pal(11, &quot;RdGy&quot;)[-6]),
    name = &quot;Frequency&quot;,
    cluster_rows = FALSE,
    cluster_columns = FALSE,
    row_names_side = &quot;left&quot;,
    row_title = &quot;sample_id&quot;,
    column_title = &quot;cluster_id&quot;,
    column_title_side = &quot;bottom&quot;,
    rect_gp = gpar(col = &quot;white&quot;),
    cell_fun = function(i, j, x, y, width, height, fill)
        grid.text(round(mat[j, i] * 100, 2), x = x, y = y, 
            gp = gpar(col = &quot;white&quot;, fontsize = 10)))</code></pre>
<p><img src="figure/organoid-05-stage_integration_cluster_analysis.Rmd/unnamed-chunk-3-1.png" width="864" style="display: block; margin: auto;" /></p>
<pre class="r"><code>(n_cells_group &lt;- table(sce$cluster_id, sce$group_id))</code></pre>
<pre><code>    
       P22   D52   D96    H9 409b2
  0     82 10790  4123  1544  1873
  1   6314  2057  2031  2513  1948
  10  1404    68     7   968   788
  11   882   541   467   558   435
  12    15     6     1   662   407
  13   206    74    41    30    72
  2   3871   181   123  2728  2454
  3   2000   689   721  1584  1081
  4     97   730   244  2057  2703
  5      0    60    15  3369  1742
  6     77   444   184  2319  1779
  7   1715   136    92  1437  1060
  8      0    42     5  1999  2236
  9     76   307    79  1458  1694</code></pre>
<pre class="r"><code>fqs &lt;- prop.table(n_cells_group, margin = 2)
mat &lt;- as.matrix(unclass(fqs))
Heatmap(mat,
    col = rev(brewer.pal(11, &quot;RdGy&quot;)[-6]),
    name = &quot;Frequency&quot;,
    cluster_rows = FALSE,
    cluster_columns = FALSE,
    row_names_side = &quot;left&quot;,
    row_title = &quot;cluster_id&quot;,
    column_title = &quot;group_id&quot;,
    column_title_side = &quot;bottom&quot;,
    rect_gp = gpar(col = &quot;white&quot;),
    cell_fun = function(i, j, x, y, width, height, fill)
        grid.text(round(mat[j, i] * 100, 2), x = x, y = y, 
            gp = gpar(col = &quot;white&quot;, fontsize = 10)))</code></pre>
<p><img src="figure/organoid-05-stage_integration_cluster_analysis.Rmd/unnamed-chunk-4-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>(n_cells_group &lt;- table(sce$cluster_id, sce$Stage))</code></pre>
<pre><code>    
       P22   D52   D96 iPSCs    EB Neuroectoderm Neuroepithelium Organoid-1M
  0     82 10790  4123     6    14             1               1         394
  1   6314  2057  2031  1535  1427           600             185         344
  10  1404    68     7   245   198           244             119         283
  11   882   541   467    68    45           145              46         219
  12    15     6     1     0     0             0               0          57
  13   206    74    41     1     0             2               9          15
  2   3871   181   123   840   671           766             325         908
  3   2000   689   721   255   259           516             439         705
  4     97   730   244     1     2            30              40        1034
  5      0    60    15     0     0             0               0         201
  6     77   444   184   506    82           153              90         216
  7   1715   136    92   842   602           295              99         200
  8      0    42     5     0     0             0               0          42
  9     76   307    79     1     0             5              10         452
    
     Organoid-2M Organoid-4M
  0         2354         647
  1          288          82
  10         513         154
  11         286         184
  12         665         347
  13          38          37
  2         1272         400
  3          324         167
  4         2657         996
  5         2693        2217
  6         1223        1828
  7          374          85
  8         2720        1473
  9         1813         871</code></pre>
<pre class="r"><code>fqs &lt;- prop.table(n_cells_group, margin = 2)
mat &lt;- as.matrix(unclass(fqs))
Heatmap(mat,
    col = rev(brewer.pal(11, &quot;RdGy&quot;)[-6]),
    name = &quot;Frequency&quot;,
    cluster_rows = FALSE,
    cluster_columns = FALSE,
    row_names_side = &quot;left&quot;,
    row_title = &quot;cluster_id&quot;,
    column_title = &quot;Stage&quot;,
    column_title_side = &quot;bottom&quot;,
    rect_gp = gpar(col = &quot;white&quot;),
    cell_fun = function(i, j, x, y, width, height, fill)
        grid.text(round(mat[j, i] * 100, 2), x = x, y = y, 
            gp = gpar(col = &quot;white&quot;, fontsize = 10)))</code></pre>
<p><img src="figure/organoid-05-stage_integration_cluster_analysis.Rmd/unnamed-chunk-5-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/ff520aefa753dff305aab2255f2cfae85084232f/docs/figure/organoid-05-stage_integration_cluster_analysis.Rmd/unnamed-chunk-5-1.png" target="_blank">ff520ae</a>
</td>
<td>
khembach
</td>
<td>
2020-09-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>n_cells_lineage &lt;- table(sce$cluster_id, sce$cl_FullLineage)
fqs &lt;- prop.table(n_cells_lineage, margin = 2)
mat &lt;- as.matrix(unclass(fqs))
cn &lt;- colnames(mat)
Heatmap(mat,
    col = rev(brewer.pal(11, &quot;RdGy&quot;)[-6]),
    name = &quot;Frequency&quot;,
    cluster_rows = FALSE,
    cluster_columns = FALSE,
    show_column_names = FALSE, 
    row_names_side = &quot;left&quot;,
    row_title = &quot;cluster_id&quot;,
    column_title = &quot;cl_FullLineage&quot;,
    column_title_side = &quot;bottom&quot;,
    rect_gp = gpar(col = &quot;white&quot;),
    cell_fun = function(i, j, x, y, width, height, fill)
        grid.text(round(mat[j, i] * 100, 1), x = x, y = y, 
            gp = gpar(col = &quot;white&quot;, fontsize = 10)),
    bottom_annotation = HeatmapAnnotation(
      text = anno_text(cn, rot = 80, just = &quot;right&quot;)))</code></pre>
<p><img src="figure/organoid-05-stage_integration_cluster_analysis.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>n_cells_lineage &lt;- table(sce$cl_FullLineage, sce$cluster_id)
fqs &lt;- prop.table(n_cells_lineage, margin = 2)
mat &lt;- as.matrix(unclass(fqs))
Heatmap(mat,
    col = rev(brewer.pal(11, &quot;RdGy&quot;)[-6]),
    name = &quot;Frequency&quot;,
    cluster_rows = FALSE,
    cluster_columns = FALSE,
    row_names_side = &quot;left&quot;,
    row_title = &quot;cl_FullLineage&quot;,
    row_names_rot = 10,
    column_title = &quot;cluster_id&quot;,
    column_title_side = &quot;bottom&quot;,
    rect_gp = gpar(col = &quot;white&quot;),
    cell_fun = function(i, j, x, y, width, height, fill)
        grid.text(round(mat[j, i] * 100, 1), x = x, y = y, 
            gp = gpar(col = &quot;white&quot;, fontsize = 10)))</code></pre>
<p><img src="figure/organoid-05-stage_integration_cluster_analysis.Rmd/unnamed-chunk-6-2.png" width="960" style="display: block; margin: auto;" /></p>
<div id="evaluation-of-cluster-before-and-after-integration" class="section level2">
<h2>Evaluation of cluster before and after integration</h2>
<p>We evaluate if cells which were together in a cluster before the integration of the organoid cells are still in the same cluster after integration.</p>
<pre class="r"><code>## Load the Seurat object from the integrated analysis of our 6 samples
so_before &lt;- readRDS(file.path(&quot;output&quot;, &quot;so_04_clustering.rds&quot;))
so_before &lt;- SetIdent(so_before, value = &quot;integrated_snn_res.0.4&quot;)
so_before@meta.data$cluster_id &lt;- Idents(so_before)
table(so_before@meta.data$cluster_id)</code></pre>
<pre><code>
    0     1     2     3     4     5     6     7     8     9    10    11    12 
11194  3938  2856  2763  2608  2481  2467  2332  1948  1870  1340  1328  1176 
   13    14    15    16 
  976   915   488   317 </code></pre>
<pre class="r"><code>## subset to our cells
cs &lt;- which(so@meta.data$integration_group %in% c(&quot;P22&quot;, &quot;D52&quot;, &quot;D96&quot;))
sub &lt;- subset(so, cells = cs)
table(sub@meta.data$cluster_id)</code></pre>
<pre><code>
    0     1    10    11    12    13     2     3     4     5     6     7     8 
14995 10402  1479  1890    22   321  4175  3410  1071    75   705  1943    47 
    9 
  462 </code></pre>
<pre class="r"><code>## join the cluster_ids from both clustering runs
before &lt;- data.frame(cell = colnames(so_before), 
                     cluster_before = so_before@meta.data[,c(&quot;cluster_id&quot;)])
after &lt;- data.frame(cell = colnames(sub), 
                    cluster_after = sub@meta.data[,c(&quot;cluster_id&quot;)])
clusters &lt;- before %&gt;% full_join(after)

## check if cells from the same cluster are still in the same cluster
(n_clusters &lt;- table(clusters$cluster_after, clusters$cluster_before))</code></pre>
<pre><code>    
        0    1    2    3    4    5    6    7    8    9   10   11   12   13   14
  0    70 3703    0    1    0 2129 2280 2189    1 1716    4  105 1057  924  816
  1  5978   87    1  639 1972   48   83   36  863   69  160  174   51   22   16
  10   99    1 1345    4    0    1    0    1    0    0   26    0    0    0    0
  11  100   18    0   31   38   26   17   34   61   20  871  619   15    4   33
  12    0    0   21    1    0    0    0    0    0    0    0    0    0    0    0
  13    0    0    0   57    0    0    0    0    0    0    0    0    0    0    0
  2  2281   34 1297  220   21   24   12    2   49   15  164   22   18    7    6
  3  1029    7   12 1628  125    1    2    1  456    7    9   71    9    0    0
  4    99    0    0   85  417    0    0    0  457    0   11    1    0    0    0
  5     0    7    0    0    0   43   10   12    0    2    0    0    0    1    0
  6    48   22    0   22   18   47   27   47   37   23   25  320   16   15   37
  7  1446   48  180   48   17   25   34    6   24   17   70   15    4    1    2
  8     0   11    0    0    0   23    2    4    0    1    0    0    4    2    0
  9    44    0    0   27    0  114    0    0    0    0    0    1    2    0    5
    
       15   16
  0     0    0
  1   202    1
  10    1    1
  11    3    0
  12    0    0
  13    0  264
  2     2    1
  3     3   50
  4     1    0
  5     0    0
  6     1    0
  7     6    0
  8     0    0
  9   269    0</code></pre>
<pre class="r"><code>fqs &lt;- prop.table(n_clusters, margin = 2)
mat &lt;- as.matrix(unclass(fqs))
Heatmap(mat,
    col = rev(brewer.pal(11, &quot;RdGy&quot;)[-6]),
    name = &quot;Frequency&quot;,
    cluster_rows = FALSE,
    cluster_columns = FALSE,
    row_names_side = &quot;left&quot;,
    row_title = &quot;clusters after organoid integration&quot;,
    column_title = &quot;clusters before organoid integration&quot;,
    column_title_side = &quot;bottom&quot;,
    rect_gp = gpar(col = &quot;white&quot;),
    cell_fun = function(i, j, x, y, width, height, fill)
        grid.text(round(mat[j, i] * 100, 2), x = x, y = y, 
            gp = gpar(col = &quot;white&quot;, fontsize = 8)))</code></pre>
<p><img src="figure/organoid-05-stage_integration_cluster_analysis.Rmd/cluster-before-after-organoid-integration-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>## add the old cluster identities to the Seurat object
so@meta.data$cluster_id_before &lt;- so_before@meta.data$cluster_id[
  match(colnames(so), colnames(so_before))]</code></pre>
</div>
</div>
<div id="dimension-reduction-plots" class="section level1 tabset">
<h1>Dimension reduction plots</h1>
<pre class="r"><code>so$integration_group &lt;- factor(so$integration_group, 
        levels = c(&quot;P22&quot;, &quot;D52&quot;, &quot;D96&quot;, &quot;iPSCs_409b2&quot;, &quot;iPSCs_H9&quot;, &quot;EB_409b2&quot;, 
                   &quot;EB_H9&quot;, &quot;Neuroectoderm_409b2&quot;, &quot;Neuroectoderm_H9&quot;, 
                   &quot;Neuroepithelium_409b2&quot;, &quot;Neuroepithelium_H9&quot;,
                   &quot;Organoid-1M_409b2&quot;, &quot;Organoid-1M_H9&quot;, &quot;Organoid-2M_409b2&quot;, 
                   &quot;Organoid-2M_H9&quot;, &quot;Organoid-4M_409b2&quot;, &quot;Organoid-4M_H9&quot;))


cs &lt;- sample(colnames(so), 10e3)
.plot_dr &lt;- function(so, dr, id)
    DimPlot(so, cells = cs, group.by = id, reduction = dr, pt.size = 0.4) +
        guides(col = guide_legend(nrow = 11, 
            override.aes = list(size = 3, alpha = 1))) +
        theme_void() + theme(aspect.ratio = 1)
ids &lt;- c(&quot;integration_group&quot;, &quot;group_id&quot;, &quot;cl_FullLineage&quot;, &quot;cluster_id&quot;, 
         &quot;cluster_id_before&quot;)
for (id in ids) {
    cat(&quot;## &quot;, id, &quot;\n&quot;)
    p1 &lt;- .plot_dr(so, &quot;tsne&quot;, id)
    lgd &lt;- get_legend(p1)
    p1 &lt;- p1 + theme(legend.position = &quot;none&quot;)
    p2 &lt;- .plot_dr(so, &quot;umap&quot;, id) + theme(legend.position = &quot;none&quot;)
    ps &lt;- plot_grid(plotlist = list(p1, p2), nrow = 1)
    p &lt;- plot_grid(ps, lgd, nrow = 1, rel_widths = c(1, 0.5))
    print(p)
    cat(&quot;\n\n&quot;)
}</code></pre>
<div id="integration_group" class="section level2">
<h2>integration_group</h2>
<p><img src="figure/organoid-05-stage_integration_cluster_analysis.Rmd/integration-dim-red-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-integration-dim-red-1">
Past versions of integration-dim-red-1.png
</button>
</p>
<div id="fig-integration-dim-red-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/ff520aefa753dff305aab2255f2cfae85084232f/docs/figure/organoid-05-stage_integration_cluster_analysis.Rmd/integration-dim-red-1.png" target="_blank">ff520ae</a>
</td>
<td>
khembach
</td>
<td>
2020-09-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="group_id" class="section level2">
<h2>group_id</h2>
<p><img src="figure/organoid-05-stage_integration_cluster_analysis.Rmd/integration-dim-red-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-integration-dim-red-2">
Past versions of integration-dim-red-2.png
</button>
</p>
<div id="fig-integration-dim-red-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/ff520aefa753dff305aab2255f2cfae85084232f/docs/figure/organoid-05-stage_integration_cluster_analysis.Rmd/integration-dim-red-2.png" target="_blank">ff520ae</a>
</td>
<td>
khembach
</td>
<td>
2020-09-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cl_fulllineage" class="section level2">
<h2>cl_FullLineage</h2>
<p><img src="figure/organoid-05-stage_integration_cluster_analysis.Rmd/integration-dim-red-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-integration-dim-red-3">
Past versions of integration-dim-red-3.png
</button>
</p>
<div id="fig-integration-dim-red-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/ff520aefa753dff305aab2255f2cfae85084232f/docs/figure/organoid-05-stage_integration_cluster_analysis.Rmd/integration-dim-red-3.png" target="_blank">ff520ae</a>
</td>
<td>
khembach
</td>
<td>
2020-09-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster_id" class="section level2">
<h2>cluster_id</h2>
<p><img src="figure/organoid-05-stage_integration_cluster_analysis.Rmd/integration-dim-red-4.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-integration-dim-red-4">
Past versions of integration-dim-red-4.png
</button>
</p>
<div id="fig-integration-dim-red-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/ff520aefa753dff305aab2255f2cfae85084232f/docs/figure/organoid-05-stage_integration_cluster_analysis.Rmd/integration-dim-red-4.png" target="_blank">ff520ae</a>
</td>
<td>
khembach
</td>
<td>
2020-09-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cluster_id_before" class="section level2">
<h2>cluster_id_before</h2>
<p><img src="figure/organoid-05-stage_integration_cluster_analysis.Rmd/integration-dim-red-5.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-integration-dim-red-5">
Past versions of integration-dim-red-5.png
</button>
</p>
<div id="fig-integration-dim-red-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/ff520aefa753dff305aab2255f2cfae85084232f/docs/figure/organoid-05-stage_integration_cluster_analysis.Rmd/integration-dim-red-5.png" target="_blank">ff520ae</a>
</td>
<td>
khembach
</td>
<td>
2020-09-20
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="find-markers-using-scran" class="section level1">
<h1>Find markers using <code>scran</code></h1>
<p>We identify candidate marker genes for each cluster.</p>
<pre class="r"><code>scran_markers &lt;- findMarkers(sce, 
    groups = sce$cluster_id, block = sce$sample_id, 
    direction = &quot;up&quot;, lfc = 2, full.stats = TRUE)</code></pre>
<div id="heatmap-of-mean-marker-exprs.-by-cluster" class="section level2">
<h2>Heatmap of mean marker-exprs. by cluster</h2>
<p>We aggregate the cells to pseudobulks and plot the average expression of the candidate marker genes in each of the clusters.</p>
<pre class="r"><code>gs &lt;- lapply(scran_markers, function(u) rownames(u)[u$Top == 1])
## candidate cluster markers
lapply(gs, function(x) str_split(x, pattern = &quot;\\.&quot;, simplify = TRUE)[,2])</code></pre>
<pre><code>$`0`
[1] &quot;SAMD11&quot; &quot;STMN2&quot;  &quot;RTN1&quot;   &quot;SEZ6L2&quot; &quot;MT-CO3&quot;

$`1`
[1] &quot;SAMD11&quot; &quot;VIM&quot;    &quot;CKB&quot;    &quot;MT-CO1&quot;

$`10`
[1] &quot;CENPF&quot; &quot;HMGB2&quot; &quot;VIM&quot;   &quot;UBE2C&quot;

$`11`
[1] &quot;SAMD11&quot; &quot;PTN&quot;    &quot;VIM&quot;    &quot;SLC3A2&quot; &quot;CKB&quot;   

$`12`
[1] &quot;HES6&quot;     &quot;HMGB2&quot;    &quot;HIST1H4C&quot; &quot;CKB&quot;     

$`13`
[1] &quot;S100A11&quot; &quot;COL3A1&quot;  &quot;LGALS1&quot; 

$`2`
[1] &quot;SAMD11&quot;   &quot;HIST1H4C&quot; &quot;VIM&quot;      &quot;CKB&quot;      &quot;MT-CO3&quot;  

$`3`
[1] &quot;SAMD11&quot; &quot;SPARC&quot;  &quot;PTN&quot;    &quot;CLU&quot;    &quot;VIM&quot;    &quot;MT-CO1&quot;

$`4`
[1] &quot;SAMD11&quot;  &quot;C1orf61&quot; &quot;FABP7&quot;   &quot;VIM&quot;     &quot;TTYH1&quot;  

$`5`
[1] &quot;SAMD11&quot;   &quot;STMN2&quot;    &quot;MT-ATP6&quot;  &quot;DLX6-AS1&quot;

$`6`
[1] &quot;SAMD11&quot; &quot;IGFBP2&quot; &quot;TUBB2B&quot; &quot;STMN2&quot;  &quot;RTN1&quot;   &quot;ALDOA&quot; 

$`7`
[1] &quot;SAMD11&quot; &quot;HMGB2&quot;  &quot;PTTG1&quot;  &quot;VIM&quot;    &quot;CKB&quot;    &quot;MT-CO1&quot;

$`8`
[1] &quot;STMN2&quot;   &quot;NFIB&quot;    &quot;NEUROD2&quot; &quot;MT-CO3&quot; 

$`9`
[1] &quot;HES6&quot;  &quot;SOX4&quot;  &quot;STMN2&quot; &quot;CKB&quot;  </code></pre>
<pre class="r"><code>sub &lt;- sce[unique(unlist(gs)), ]
pbs &lt;- aggregateData(sub, assay = &quot;logcounts&quot;, by = &quot;cluster_id&quot;, fun = &quot;mean&quot;)
mat &lt;- t(muscat:::.scale(assay(pbs)))
## remove the Ensembl ID from the gene names
colnames(mat) &lt;- str_split(colnames(mat), pattern = &quot;\\.&quot;, simplify = TRUE)[,2] 
Heatmap(mat,
    name = &quot;scaled avg.\nexpression&quot;,
    col = viridis(10),
    cluster_rows = FALSE,
    cluster_columns = FALSE,
    row_names_side = &quot;left&quot;,
    row_title = &quot;cluster_id&quot;,
    rect_gp = gpar(col = &quot;white&quot;))</code></pre>
<p><img src="figure/organoid-05-stage_integration_cluster_analysis.Rmd/scran-markers-heatmap-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>We write tables with the top marker genes per cluster.</p>
<pre class="r"><code>gs2 &lt;- lapply(scran_markers, function(u) u[u$Top %in% 1:3,])
for (i in seq_along(gs2)) {
  write.table(x = gs2[[i]] %&gt;% as.data.frame %&gt;% 
                dplyr::mutate(gene = rownames(gs2[[i]])) %&gt;%
                dplyr::relocate(gene), 
              file =  file.path(&quot;output&quot;, 
                                paste0(&quot;organoid_integration_cluster&quot;, i, &quot;_marker_genes.txt&quot;)),
              sep = &quot;\t&quot;, quote = FALSE, row.names = FALSE)
}</code></pre>
<p>Heatmap including marker genes of rank 2 and 3.</p>
<pre class="r"><code>gs &lt;- lapply(scran_markers, function(u) rownames(u)[u$Top %in% 1:3])
## candidate cluster markers
lapply(gs, function(x) str_split(x, pattern = &quot;\\.&quot;, simplify = TRUE)[,2])</code></pre>
<pre><code>$`0`
 [1] &quot;SAMD11&quot;  &quot;STMN2&quot;   &quot;RTN1&quot;    &quot;SEZ6L2&quot;  &quot;MT-CO3&quot;  &quot;NOC2L&quot;   &quot;STMN1&quot;  
 [8] &quot;GAP43&quot;   &quot;DCX&quot;     &quot;MT-ATP6&quot; &quot;KLHL17&quot;  &quot;NSG1&quot;    &quot;STMN4&quot;   &quot;MT-CO2&quot; 
[15] &quot;HMP19&quot;  

$`1`
 [1] &quot;SAMD11&quot; &quot;VIM&quot;    &quot;CKB&quot;    &quot;MT-CO1&quot; &quot;NOC2L&quot;  &quot;SOX2&quot;   &quot;PTN&quot;    &quot;CLU&quot;   
 [9] &quot;TTYH1&quot;  &quot;MT-CO2&quot; &quot;KLHL17&quot; &quot;FABP7&quot;  &quot;PTPRZ1&quot; &quot;MT-CO3&quot;

$`10`
 [1] &quot;CENPF&quot;   &quot;HMGB2&quot;   &quot;VIM&quot;     &quot;UBE2C&quot;   &quot;SAMD11&quot;  &quot;CCNB1&quot;   &quot;PTTG1&quot;  
 [8] &quot;TOP2A&quot;   &quot;NOC2L&quot;   &quot;CKB&quot;     &quot;ARL6IP1&quot;

$`11`
 [1] &quot;SAMD11&quot;   &quot;PTN&quot;      &quot;VIM&quot;      &quot;SLC3A2&quot;   &quot;CKB&quot;      &quot;NOC2L&quot;   
 [7] &quot;CLU&quot;      &quot;EIF4EBP1&quot; &quot;MT-CO1&quot;   &quot;KLHL17&quot;   &quot;ENO1&quot;     &quot;SOX2&quot;    
[13] &quot;TUBB2B&quot;   &quot;BNIP3&quot;    &quot;DDIT3&quot;    &quot;TTYH1&quot;    &quot;MT-CO2&quot;  

$`12`
[1] &quot;HES6&quot;     &quot;HMGB2&quot;    &quot;HIST1H4C&quot; &quot;CKB&quot;      &quot;HMGN2&quot;    &quot;SOX4&quot;     &quot;TUBA1B&quot;  
[8] &quot;CDK1&quot;     &quot;TOP2A&quot;   

$`13`
[1] &quot;S100A11&quot; &quot;COL3A1&quot;  &quot;LGALS1&quot;  &quot;COL1A1&quot;  &quot;TAGLN&quot;   &quot;TPM1&quot;   

$`2`
 [1] &quot;SAMD11&quot;   &quot;HIST1H4C&quot; &quot;VIM&quot;      &quot;CKB&quot;      &quot;MT-CO3&quot;   &quot;NOC2L&quot;   
 [7] &quot;HMGB2&quot;    &quot;PTN&quot;      &quot;MT-CO1&quot;   &quot;KLHL17&quot;   &quot;SOX2&quot;     &quot;FABP7&quot;   
[13] &quot;MT-CO2&quot;   &quot;KIAA0101&quot;

$`3`
 [1] &quot;SAMD11&quot; &quot;SPARC&quot;  &quot;PTN&quot;    &quot;CLU&quot;    &quot;VIM&quot;    &quot;MT-CO1&quot; &quot;NOC2L&quot;  &quot;HES1&quot;  
 [9] &quot;CKB&quot;    &quot;MT-CO2&quot; &quot;KLHL17&quot; &quot;METRN&quot;  &quot;MT-CO3&quot; &quot;TTR&quot;   

$`4`
 [1] &quot;SAMD11&quot;  &quot;C1orf61&quot; &quot;FABP7&quot;   &quot;VIM&quot;     &quot;TTYH1&quot;   &quot;NOC2L&quot;   &quot;SOX2&quot;   
 [8] &quot;PTN&quot;     &quot;CKB&quot;     &quot;KLHL17&quot; 

$`5`
 [1] &quot;SAMD11&quot;   &quot;STMN2&quot;    &quot;MT-ATP6&quot;  &quot;DLX6-AS1&quot; &quot;NOC2L&quot;    &quot;SOX4&quot;    
 [7] &quot;RTN1&quot;     &quot;DCX&quot;      &quot;MT-CO3&quot;   &quot;DLX5&quot;     &quot;KLHL17&quot;   &quot;MLLT11&quot;  
[13] &quot;CD24&quot;     &quot;MT-CYB&quot;   &quot;DLX2&quot;    

$`6`
 [1] &quot;SAMD11&quot; &quot;IGFBP2&quot; &quot;TUBB2B&quot; &quot;STMN2&quot;  &quot;RTN1&quot;   &quot;ALDOA&quot;  &quot;NOC2L&quot;  &quot;ENO1&quot;  
 [9] &quot;CKB&quot;    &quot;DCX&quot;    &quot;KLHL17&quot; &quot;GPM6A&quot;  &quot;SOX4&quot;   &quot;BEX1&quot;  

$`7`
 [1] &quot;SAMD11&quot; &quot;HMGB2&quot;  &quot;PTTG1&quot;  &quot;VIM&quot;    &quot;CKB&quot;    &quot;MT-CO1&quot; &quot;NOC2L&quot;  &quot;CENPF&quot; 
 [9] &quot;PTN&quot;    &quot;MT-CO2&quot; &quot;KLHL17&quot; &quot;FABP7&quot;  &quot;UBE2S&quot;  &quot;MT-CO3&quot;

$`8`
 [1] &quot;STMN2&quot;   &quot;NFIB&quot;    &quot;NEUROD2&quot; &quot;MT-CO3&quot;  &quot;NEUROD6&quot; &quot;CSRP2&quot;   &quot;RTN1&quot;   
 [8] &quot;MT-ATP6&quot; &quot;SAMD11&quot;  &quot;SOX4&quot;    &quot;DCX&quot;     &quot;MT-CO1&quot; 

$`9`
 [1] &quot;HES6&quot;   &quot;SOX4&quot;   &quot;STMN2&quot;  &quot;CKB&quot;    &quot;SAMD11&quot; &quot;MLLT11&quot; &quot;FABP7&quot;  &quot;VIM&quot;   
 [9] &quot;MT-CO3&quot; &quot;NOC2L&quot;  &quot;SOX11&quot;  &quot;MT-CO2&quot;</code></pre>
<pre class="r"><code>sub &lt;- sce[unique(unlist(gs)), ]
pbs &lt;- aggregateData(sub, assay = &quot;logcounts&quot;, by = &quot;cluster_id&quot;, fun = &quot;mean&quot;)
mat &lt;- t(muscat:::.scale(assay(pbs)))
## remove the Ensembl ID from the gene names
colnames(mat) &lt;- str_split(colnames(mat), pattern = &quot;\\.&quot;, simplify = TRUE)[,2] 
Heatmap(mat,
    name = &quot;scaled avg.\nexpression&quot;,
    col = viridis(10),
    cluster_rows = FALSE,
    cluster_columns = FALSE,
    row_names_side = &quot;left&quot;,
    row_title = &quot;cluster_id&quot;,
    rect_gp = gpar(col = &quot;white&quot;))</code></pre>
<p><img src="figure/organoid-05-stage_integration_cluster_analysis.Rmd/unnamed-chunk-8-1.png" width="1344" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.6 LTS

Matrix products: default
BLAS:   /usr/local/R/R-4.0.0/lib/libRblas.so
LAPACK: /usr/local/R/R-4.0.0/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] viridis_0.5.1               viridisLite_0.3.0          
 [3] stringr_1.4.0               scran_1.16.0               
 [5] SingleCellExperiment_1.10.1 SummarizedExperiment_1.18.1
 [7] DelayedArray_0.14.0         matrixStats_0.56.0         
 [9] Biobase_2.48.0              GenomicRanges_1.40.0       
[11] GenomeInfoDb_1.24.2         IRanges_2.22.2             
[13] S4Vectors_0.26.1            BiocGenerics_0.34.0        
[15] Seurat_3.1.5                RColorBrewer_1.1-2         
[17] muscat_1.2.1                dplyr_1.0.2                
[19] ggplot2_3.3.2               cowplot_1.0.0              
[21] ComplexHeatmap_2.4.2        workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] backports_1.1.9           circlize_0.4.10          
  [3] blme_1.0-4                igraph_1.2.5             
  [5] plyr_1.8.6                lazyeval_0.2.2           
  [7] TMB_1.7.16                splines_4.0.0            
  [9] BiocParallel_1.22.0       listenv_0.8.0            
 [11] scater_1.16.2             digest_0.6.25            
 [13] foreach_1.5.0             htmltools_0.5.0          
 [15] gdata_2.18.0              lmerTest_3.1-2           
 [17] magrittr_1.5              memoise_1.1.0            
 [19] cluster_2.1.0             doParallel_1.0.15        
 [21] ROCR_1.0-11               limma_3.44.3             
 [23] globals_0.12.5            annotate_1.66.0          
 [25] prettyunits_1.1.1         colorspace_1.4-1         
 [27] rappdirs_0.3.1            ggrepel_0.8.2            
 [29] blob_1.2.1                xfun_0.15                
 [31] jsonlite_1.7.0            crayon_1.3.4             
 [33] RCurl_1.98-1.2            genefilter_1.70.0        
 [35] lme4_1.1-23               zoo_1.8-8                
 [37] ape_5.4                   survival_3.2-3           
 [39] iterators_1.0.12          glue_1.4.2               
 [41] gtable_0.3.0              zlibbioc_1.34.0          
 [43] XVector_0.28.0            leiden_0.3.3             
 [45] GetoptLong_1.0.1          BiocSingular_1.4.0       
 [47] future.apply_1.6.0        shape_1.4.4              
 [49] scales_1.1.1              DBI_1.1.0                
 [51] edgeR_3.30.3              Rcpp_1.0.5               
 [53] xtable_1.8-4              progress_1.2.2           
 [55] clue_0.3-57               dqrng_0.2.1              
 [57] reticulate_1.16           bit_1.1-15.2             
 [59] rsvd_1.0.3                tsne_0.1-3               
 [61] htmlwidgets_1.5.1         httr_1.4.1               
 [63] gplots_3.0.4              ellipsis_0.3.1           
 [65] ica_1.0-2                 farver_2.0.3             
 [67] pkgconfig_2.0.3           XML_3.99-0.4             
 [69] uwot_0.1.8                locfit_1.5-9.4           
 [71] labeling_0.3              tidyselect_1.1.0         
 [73] rlang_0.4.7               reshape2_1.4.4           
 [75] later_1.1.0.1             AnnotationDbi_1.50.1     
 [77] munsell_0.5.0             tools_4.0.0              
 [79] generics_0.0.2            RSQLite_2.2.0            
 [81] ggridges_0.5.2            evaluate_0.14            
 [83] yaml_2.2.1                knitr_1.29               
 [85] bit64_0.9-7               fs_1.4.2                 
 [87] fitdistrplus_1.1-1        caTools_1.18.0           
 [89] RANN_2.6.1                purrr_0.3.4              
 [91] pbapply_1.4-2             future_1.17.0            
 [93] nlme_3.1-148              whisker_0.4              
 [95] pbkrtest_0.4-8.6          compiler_4.0.0           
 [97] plotly_4.9.2.1            beeswarm_0.2.3           
 [99] png_0.1-7                 variancePartition_1.18.2 
[101] tibble_3.0.3              statmod_1.4.34           
[103] geneplotter_1.66.0        stringi_1.4.6            
[105] lattice_0.20-41           Matrix_1.2-18            
[107] nloptr_1.2.2.2            vctrs_0.3.4              
[109] pillar_1.4.6              lifecycle_0.2.0          
[111] lmtest_0.9-37             GlobalOptions_0.1.2      
[113] RcppAnnoy_0.0.16          BiocNeighbors_1.6.0      
[115] data.table_1.12.8         bitops_1.0-6             
[117] irlba_2.3.3               patchwork_1.0.1          
[119] httpuv_1.5.4              colorRamps_2.3           
[121] R6_2.4.1                  promises_1.1.1           
[123] KernSmooth_2.23-17        gridExtra_2.3            
[125] vipor_0.4.5               codetools_0.2-16         
[127] boot_1.3-25               MASS_7.3-51.6            
[129] gtools_3.8.2              DESeq2_1.28.1            
[131] rprojroot_1.3-2           rjson_0.2.20             
[133] withr_2.2.0               sctransform_0.2.1        
[135] GenomeInfoDbData_1.2.3    hms_0.5.3                
[137] tidyr_1.1.0               glmmTMB_1.0.2.1          
[139] minqa_1.2.4               rmarkdown_2.3            
[141] DelayedMatrixStats_1.10.1 Rtsne_0.15               
[143] git2r_0.27.1              numDeriv_2016.8-1.1      
[145] ggbeeswarm_0.6.0         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
