<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Katharina Hembach" />


<title>Organoid integration</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">neural_scRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/khembach/neural_scRNAseq">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Organoid integration</h1>
<h4 class="author">Katharina Hembach</h4>
<h4 class="date">8/10/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-08-14
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>neural_scRNAseq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200522code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200522)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200522code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200522)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
clustering
</li>
<li>
dim-red
</li>
<li>
dimension-reduction
</li>
<li>
future-setup
</li>
<li>
import
</li>
<li>
integration
</li>
<li>
normalization
</li>
<li>
reproduce-plots-clustering
</li>
<li>
reproduce-plots-dim-red
</li>
<li>
reproduce-plots-DR
</li>
<li>
reproduce-plots-integration
</li>
<li>
reproduce-plots-normalisation
</li>
<li>
save-objects
</li>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
unnamed-chunk-1
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-3
</li>
<li>
unnamed-chunk-4
</li>
<li>
unnamed-chunk-5
</li>
<li>
unnamed-chunk-6
</li>
<li>
unnamed-chunk-7
</li>
<li>
unnamed-chunk-8
</li>
<li>
unnamed-chunk-9
</li>
<li>
variable-features
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>organoid-01-clustering_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtree638915fa4f8e11001cf1f113dfd8534778aa8561targetblank638915fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/khembach/neural_scRNAseq/tree/638915fa4f8e11001cf1f113dfd8534778aa8561" target="_blank">638915f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtree638915fa4f8e11001cf1f113dfd8534778aa8561targetblank638915fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/khembach/neural_scRNAseq/tree/638915fa4f8e11001cf1f113dfd8534778aa8561" target="_blank">638915f</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    ._Rplots.pdf
    Ignored:    .__workflowr.yml
    Ignored:    ._neural_scRNAseq.Rproj
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/._.DS_Store
    Ignored:    analysis/._01-preprocessing.Rmd
    Ignored:    analysis/._01-preprocessing.html
    Ignored:    analysis/._02.1-SampleQC.Rmd
    Ignored:    analysis/._03-filtering.Rmd
    Ignored:    analysis/._04-clustering.Rmd
    Ignored:    analysis/._04-clustering.knit.md
    Ignored:    analysis/._04.1-cell_cycle.Rmd
    Ignored:    analysis/._05-annotation.Rmd
    Ignored:    analysis/._Lam-0-NSC_no_integration.Rmd
    Ignored:    analysis/._Lam-01-NSC_integration.Rmd
    Ignored:    analysis/._Lam-02-NSC_annotation.Rmd
    Ignored:    analysis/._NSC-1-clustering.Rmd
    Ignored:    analysis/._NSC-2-annotation.Rmd
    Ignored:    analysis/.__site.yml
    Ignored:    analysis/._additional_filtering.Rmd
    Ignored:    analysis/._additional_filtering_clustering.Rmd
    Ignored:    analysis/._index.Rmd
    Ignored:    analysis/._organoid-01-clustering.Rmd
    Ignored:    analysis/._organoid-02-integration.Rmd
    Ignored:    analysis/01-preprocessing_cache/
    Ignored:    analysis/02-1-SampleQC_cache/
    Ignored:    analysis/02-quality_control_cache/
    Ignored:    analysis/02.1-SampleQC_cache/
    Ignored:    analysis/03-filtering_cache/
    Ignored:    analysis/04-clustering_cache/
    Ignored:    analysis/04.1-cell_cycle_cache/
    Ignored:    analysis/05-annotation_cache/
    Ignored:    analysis/Lam-01-NSC_integration_cache/
    Ignored:    analysis/Lam-02-NSC_annotation_cache/
    Ignored:    analysis/NSC-1-clustering_cache/
    Ignored:    analysis/NSC-2-annotation_cache/
    Ignored:    analysis/additional_filtering_cache/
    Ignored:    analysis/additional_filtering_clustering_cache/
    Ignored:    analysis/organoid-01-clustering_cache/
    Ignored:    analysis/organoid-02-integration_cache/
    Ignored:    analysis/sample5_QC_cache/
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._.smbdeleteAAA17ed8b4b
    Ignored:    data/._Lam_figure2_markers.R
    Ignored:    data/._known_NSC_markers.R
    Ignored:    data/._known_cell_type_markers.R
    Ignored:    data/._metadata.csv
    Ignored:    data/data_sushi/
    Ignored:    data/filtered_feature_matrices/
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/._NSC_cluster1_marker_genes.txt
    Ignored:    output/Lam-01-clustering.rds
    Ignored:    output/NSC_1_clustering.rds
    Ignored:    output/NSC_cluster1_marker_genes.txt
    Ignored:    output/NSC_cluster2_marker_genes.txt
    Ignored:    output/NSC_cluster3_marker_genes.txt
    Ignored:    output/NSC_cluster4_marker_genes.txt
    Ignored:    output/NSC_cluster5_marker_genes.txt
    Ignored:    output/NSC_cluster6_marker_genes.txt
    Ignored:    output/NSC_cluster7_marker_genes.txt
    Ignored:    output/additional_filtering.rds
    Ignored:    output/figures/
    Ignored:    output/sce_01_preprocessing.rds
    Ignored:    output/sce_02_quality_control.rds
    Ignored:    output/sce_03_filtering.rds
    Ignored:    output/sce_organoid-01-clustering.rds
    Ignored:    output/sce_preprocessing.rds
    Ignored:    output/so_04_1_cell_cycle.rds
    Ignored:    output/so_04_clustering.rds
    Ignored:    output/so_additional_filtering_clustering.rds
    Ignored:    output/so_merged_organoid-02-integration.rds
    Ignored:    output/so_organoid-01-clustering.rds
    Ignored:    output/so_sample_organoid-01-clustering.rds

Untracked files:
    Untracked:  Rplots.pdf
    Untracked:  analysis/Lam-0-NSC_no_integration.Rmd
    Untracked:  analysis/additional_filtering.Rmd
    Untracked:  analysis/additional_filtering_clustering.Rmd
    Untracked:  analysis/organoid-01-integration.Rmd
    Untracked:  analysis/organoid-02-integration.Rmd
    Untracked:  analysis/sample5_QC.Rmd
    Untracked:  data/Homo_sapiens.GRCh38.98.sorted.gtf
    Untracked:  data/Kanton_et_al/
    Untracked:  data/Lam_et_al/
    Untracked:  scripts/

Unstaged changes:
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/organoid-01-clustering.Rmd</code>) and HTML (<code>docs/organoid-01-clustering.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/638915fa4f8e11001cf1f113dfd8534778aa8561/analysis/organoid-01-clustering.Rmd" target="_blank">638915f</a>
</td>
<td>
khembach
</td>
<td>
2020-08-14
</td>
<td>
organoid clustering
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/khembach/neural_scRNAseq/5b37055b675c0a2c7273daf196d71bce8efaf677/docs/organoid-01-clustering.html" target="_blank">5b37055</a>
</td>
<td>
khembach
</td>
<td>
2020-08-12
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/ec375b2501236d4314f4b376874418c059b43a37/analysis/organoid-01-clustering.Rmd" target="_blank">ec375b2</a>
</td>
<td>
khembach
</td>
<td>
2020-08-12
</td>
<td>
Cluster organoid data fromn Kanton et al. 2019
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>library(DropletUtils)
library(scDblFinder)
library(BiocParallel)
library(ggplot2)
library(scater)
library(dplyr)
library(cowplot)
library(ggplot2)
library(Seurat)
library(SingleCellExperiment)
library(stringr)
library(Seurat)
library(rtracklayer)
library(future)
library(data.table)</code></pre>
<pre class="r"><code># increase future&#39;s maximum allowed size of exported globals to 5GB 
# the default is 2GB
options(future.globals.maxSize = 5000 * 1024 ^ 2)
# change the current plan to access parallelization
plan(&quot;multiprocess&quot;, workers = 20)</code></pre>
</div>
<div id="importing-cellranger-output-and-metadata" class="section level2">
<h2>Importing CellRanger output and metadata</h2>
<pre class="r"><code>fs &lt;- file.path(&quot;data&quot;, &quot;Kanton_et_al&quot;, &quot;files&quot;)
sce &lt;- read10xCounts(samples = fs, sample.names = &quot;organoids&quot;)

# rename colnames and dimnames
names(rowData(sce)) &lt;- c(&quot;ensembl_id&quot;, &quot;symbol&quot;, &quot;type&quot;)
names(colData(sce)) &lt;- c(&quot;sample_id&quot;, &quot;barcode&quot;)

# load metadata
meta &lt;- read.csv(file.path(&quot;data&quot;, &quot;Kanton_et_al&quot;, &quot;metadata.csv&quot;))
colData(sce) &lt;- cbind(colData(sce), meta[, colnames(meta != &quot;Barcode&quot;)])

sce$sample_id &lt;- factor(sce$sample_id)
dimnames(sce) &lt;- list(with(rowData(sce), paste(ensembl_id, symbol, sep = &quot;.&quot;)),
                      with(colData(sce), paste(barcode, Sample, sep = &quot;.&quot;)))</code></pre>
</div>
<div id="overview-of-the-data" class="section level2">
<h2>Overview of the data</h2>
<pre class="r"><code>table(colData(sce)$Line)</code></pre>
<pre><code>
  409b2      H9   Hoik1   Kucg2 Sc102a1   Sojd3   Wibj2 
  24983   25543    3583    4704   10408    4144   17211 </code></pre>
<pre class="r"><code>table(colData(sce)$Stage)</code></pre>
<pre><code>
             EB           iPSCs   Neuroectoderm Neuroepithelium     Organoid-1M 
           3300            4300            2757            1363            5070 
    Organoid-2M     Organoid-4M 
          62306           11480 </code></pre>
<pre class="r"><code>table(colData(sce)$Sample)</code></pre>
<pre><code>
      h409B2_120d_org1       h409B2_128d_org1        h409B2_60d_org1 
                  3179                   3915                   6354 
       h409B2_67d_org1              h409B2_EB           h409B2_iPSCs 
                  4992                    855                   1943 
  h409B2_neuroectoderm h409B2_neuroepithelium         h409B2_Org_32d 
                   886                    443                   2416 
          H9_128d_org1            H9_60d_org1            H9_67d_org1 
                  4386                   5899                   5011 
                 H9_EB               H9_iPSCs       H9_neuroectoderm 
                  2445                   2357                   1871 
    H9_neuroepithelium             H9_Org_32d          hoik_HipSci_1 
                   920                   2654                   1168 
         hoik_HipSci_2          hoik_HipSci_3          kucg_HipSci_1 
                  1533                    882                   1456 
         kucg_HipSci_2          kucg_HipSci_3       SC102A1_65d_org1 
                  1322                   1926                   5231 
      SC102A1_65d_org2          sojd_HipSci_1          sojd_HipSci_2 
                  5177                   1247                   1122 
         sojd_HipSci_3          wibj_64d_org1          wibj_64d_org2 
                  1775                   5823                   6272 
         wibj_HipSci_1          wibj_HipSci_2          wibj_HipSci_3 
                  1774                   1516                   1826 </code></pre>
<pre class="r"><code>table(colData(sce)$PredCellType)</code></pre>
<pre><code>
     Astrocyte   Astrocyte/RG        Choroid     Choroid/RG             EN 
           651              4           1725              1          27146 
       EN/Glyc          EN/IN    Endothelial           Glyc        Glyc/IN 
            11              7              5          15292              2 
Glyc/Microglia     Glyc/Mural        Glyc/RG             IN         IN/IPC 
             1              1              4          14482              1 
  IN/Microglia            IPC         IPC/RG      Microglia          Mural 
             1          13549              2            344            210 
           OPC             RG 
            45          17092 </code></pre>
<pre class="r"><code>## The cells for Fig. 1a-d, Extended Data Fig. 2
table(colData(sce)$in_FullLineage)</code></pre>
<pre><code>
FALSE  TRUE 
47078 43498 </code></pre>
<pre class="r"><code>## The cells for Fig. 1e
table(colData(sce)$in_LineComp)</code></pre>
<pre><code>
FALSE  TRUE 
41423 49153 </code></pre>
<pre class="r"><code>## Cluster labels for Extended Data Fig. 2
table(colData(sce)$cl_FullLineage)</code></pre>
<pre><code>
choroid plexus/mesenchymal-like cells                    cortical neurons 1 
                                  371                                  2967 
                   cortical neurons 2            cycling dorsal progenitors 
                                 2557                                  1701 
          cycling ventral progenitors ectodermal/neuroectodermal-like cells 
                                 1022                                   870 
  gliogenic/outer RGCs and astrocytes        IPs and early cortical neurons 
                                 1936                                  1710 
             midbrain/hindbrain cells            neuroepithelial-like cells 
                                 3296                                  3230 
                   retina progenitors                                RGCs 3 
                                 1145                                  3562 
                         RGCs early 1                          RGCs early 2 
                                 2600                                   885 
                         stem cells 1                          stem cells 2 
                                 1279                                  5652 
                         stem cells 3     ventral progenitors and neurons 1 
                                  672                                  1576 
    ventral progenitors and neurons 2     ventral progenitors and neurons 3 
                                 1899                                  4568 </code></pre>
<pre class="r"><code>## Cluster labels
table(colData(sce)$cl_LineComp)</code></pre>
<pre><code>
Cerebellar glutamatergic neurons 1 Cerebellar glutamatergic neurons 2 
                              2237                                923 
                      Cortical IPs                   Cortical neurons 
                              6407                              12148 
                     Cortical NPCs                   LGE interneurons 
                              9104                               3081 
              MGE/CGE interneurons                 Mid-hindbrain NPCs 
                              6587                               1857 
   Midbrain GABAergic interneurons                     Purkinje cells 
                               458                                600 
                      Ventral NPCs 
                              3574 </code></pre>
<p>The dataset consists of 10X scRNA-seq from organoid development using embryonic stem cells (H9) and an iPSC (409b2) line. The cells for figure 1 a-d are labeled in column <code>in_FullLineage</code> and figure 1e in <code>in_LineComp</code>.</p>
</div>
<div id="quality-control" class="section level2">
<h2>Quality control</h2>
<p>We remove undetected genes and check cell-level QC that came with the data.</p>
<pre class="r"><code>sce &lt;- sce[rowSums(counts(sce) &gt; 0) &gt; 0, ]
dim(sce)</code></pre>
<pre><code>[1] 28601 90576</code></pre>
<pre class="r"><code># library size
summary(sce$nUMI)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   1743    4803    6918    8417   10385   59266 </code></pre>
<pre class="r"><code># number of detected genes per cell
summary(sce$nGene)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    223    1906    2453    2601    3148    5999 </code></pre>
<pre class="r"><code># percentage of counts that come from mitochondrial genes:
summary(sce$PercentMito)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
0.00000 0.01688 0.02674 0.02770 0.03678 0.09998 </code></pre>
<p>It seems the cells are already filtered based on the number of detected genes. From the methods section of the paper: &quot;Cells with more than 6,000 or less than 200 detected genes, as well as those with mitochondrial transcripts proportion higher than 5% were excluded&quot;</p>
<div id="diagnostic-plots" class="section level3">
<h3>Diagnostic plots</h3>
<p>The number of counts per cell:</p>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;nUMI&quot;) + scale_y_log10()</code></pre>
<p><img src="figure/organoid-01-clustering.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/5b37055b675c0a2c7273daf196d71bce8efaf677/docs/figure/organoid-01-clustering.Rmd/unnamed-chunk-4-1.png" target="_blank">5b37055</a>
</td>
<td>
khembach
</td>
<td>
2020-08-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The number of genes:</p>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;nGene&quot;) + scale_y_log10() </code></pre>
<p><img src="figure/organoid-01-clustering.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/5b37055b675c0a2c7273daf196d71bce8efaf677/docs/figure/organoid-01-clustering.Rmd/unnamed-chunk-5-1.png" target="_blank">5b37055</a>
</td>
<td>
khembach
</td>
<td>
2020-08-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>The percentage of mitochondrial genes:</p>
<pre class="r"><code>plotColData(sce, x = &quot;sample_id&quot;, y = &quot;PercentMito&quot;)</code></pre>
<p><img src="figure/organoid-01-clustering.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/5b37055b675c0a2c7273daf196d71bce8efaf677/docs/figure/organoid-01-clustering.Rmd/unnamed-chunk-6-1.png" target="_blank">5b37055</a>
</td>
<td>
khembach
</td>
<td>
2020-08-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="normalization" class="section level2">
<h2>Normalization</h2>
<p>We try to recreate Extended Data Fig. 2 a-e.</p>
<pre class="r"><code>## subset the cells
sce_all &lt;- sce
sce &lt;- sce[, sce$in_FullLineage]
dim(sce)</code></pre>
<pre><code>[1] 28601 43498</code></pre>
<pre class="r"><code># create SeuratObject
so &lt;- CreateSeuratObject(
    counts = counts(sce),
    meta.data = data.frame(colData(sce)),
    project = &quot;organoids&quot;)

# split by sample
cells_by_sample &lt;- split(colnames(sce), sce$Sample)
so &lt;- lapply(cells_by_sample, function(i) subset(so, cells = i))

## log normalize the data using a scaling factor of 10000
so &lt;- lapply(so, NormalizeData, verbose = FALSE, scale.factor = 10000, 
             normalization.method = &quot;LogNormalize&quot;)</code></pre>
</div>
<div id="integration-of-h9-and-409b2" class="section level2">
<h2>Integration of H9 and 409b2</h2>
<pre class="r"><code>so_all &lt;- so
sub_samples &lt;- unique(c(colData(sce)[sce$Line %in% c(&quot;H9&quot;,&quot;409b2&quot;),]$Sample))
so &lt;- so[sub_samples]

## Identify the top 2000 genes with high cell-to-cell variation
so &lt;- lapply(so, FindVariableFeatures, nfeatures = 2000, 
    selection.method = &quot;vst&quot;, verbose = FALSE)</code></pre>
<pre class="r"><code># find anchors &amp; integrate
as &lt;- FindIntegrationAnchors(so, verbose = FALSE)
so &lt;- IntegrateData(anchorset = as, dims = seq_len(20), verbose = FALSE)
# 
# ## We scale the data so that mean expression is 0 and variance is 1, across cells
# ## We also regress out the number of UMIs. 
# ## We don&#39;t have mitochondrial genes for the NES
# DefaultAssay(so) &lt;- &quot;integrated&quot;
so &lt;- ScaleData(so, verbose = FALSE, vars.to.regress = c(&quot;nGene&quot;, &quot;PercentMito&quot;))</code></pre>
</div>
<div id="dimension-reduction" class="section level2">
<h2>Dimension reduction</h2>
<p>We perform dimension reduction with t-SNE and UMAP based on PCA results.</p>
<pre class="r"><code>so &lt;- RunPCA(so, npcs = 30, verbose = FALSE)
so &lt;- RunTSNE(so, reduction = &quot;pca&quot;, dims = seq_len(20),
    seed.use = 1, do.fast = TRUE, verbose = FALSE)
so &lt;- RunUMAP(so, reduction = &quot;pca&quot;, dims = seq_len(20),
    seed.use = 1, verbose = FALSE)</code></pre>
<div id="plot-pca-results" class="section level3">
<h3>Plot PCA results</h3>
<pre class="r"><code>## PCA plot 
DimPlot(so, reduction = &quot;pca&quot;, group.by = &quot;sample_id&quot;)</code></pre>
<p><img src="figure/organoid-01-clustering.Rmd/unnamed-chunk-7-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/5b37055b675c0a2c7273daf196d71bce8efaf677/docs/figure/organoid-01-clustering.Rmd/unnamed-chunk-7-1.png" target="_blank">5b37055</a>
</td>
<td>
khembach
</td>
<td>
2020-08-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># elbow plot with the ranking of PCs based on the % of variance explained
ElbowPlot(so, ndims = 30)</code></pre>
<p><img src="figure/organoid-01-clustering.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/5b37055b675c0a2c7273daf196d71bce8efaf677/docs/figure/organoid-01-clustering.Rmd/unnamed-chunk-8-1.png" target="_blank">5b37055</a>
</td>
<td>
khembach
</td>
<td>
2020-08-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="clustering" class="section level2">
<h2>Clustering</h2>
<p>We cluster the cells using the reduced PCA dimensions.</p>
<pre class="r"><code>so &lt;- FindNeighbors(so, reduction = &quot;pca&quot;, dims = seq_len(20), verbose = FALSE)
for (res in c(0.4, 0.6, 0.8))
    so &lt;- FindClusters(so, resolution = res, random.seed = 1, verbose = FALSE)</code></pre>
</div>
<div id="dimension-reduction-plots" class="section level2 tabset">
<h2>Dimension reduction plots</h2>
<p>We plot the dimension reduction (DR) and color by cell line, sample, predicted cell type, cluster ID</p>
<pre class="r"><code># set cluster IDs to resolution 0.4 clustering
so &lt;- SetIdent(so, value = &quot;integrated_snn_res.0.6&quot;)
so@meta.data$cluster_id &lt;- Idents(so)

cs &lt;- sample(colnames(so), 10e3)
.plot_dr &lt;- function(so, dr, id)
    DimPlot(so, cells = cs, group.by = id, reduction = dr, pt.size = 0.4) +
        guides(col = guide_legend(nrow = 11, 
            override.aes = list(size = 3, alpha = 1))) +
        theme_void() + theme(aspect.ratio = 1)
ids &lt;- c(&quot;Line&quot;, &quot;Sample&quot;, &quot;Stage&quot;,&quot;cl_FullLineage&quot;, &quot;ident&quot;)
for (id in ids) {
    cat(&quot;### &quot;, id, &quot;\n&quot;)
    p1 &lt;- .plot_dr(so, &quot;tsne&quot;, id)
    lgd &lt;- get_legend(p1)
    p1 &lt;- p1 + theme(legend.position = &quot;none&quot;)
    p2 &lt;- .plot_dr(so, &quot;umap&quot;, id) + theme(legend.position = &quot;none&quot;)
    ps &lt;- plot_grid(plotlist = list(p1, p2), nrow = 1)
    p &lt;- plot_grid(ps, lgd, nrow = 1, rel_widths = c(1, 0.5))
    print(p)
    cat(&quot;\n\n&quot;)
}</code></pre>
<div id="line" class="section level3">
<h3>Line</h3>
<p><img src="figure/organoid-01-clustering.Rmd/dim-red-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dim-red-1">
Past versions of dim-red-1.png
</button>
</p>
<div id="fig-dim-red-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/5b37055b675c0a2c7273daf196d71bce8efaf677/docs/figure/organoid-01-clustering.Rmd/dim-red-1.png" target="_blank">5b37055</a>
</td>
<td>
khembach
</td>
<td>
2020-08-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="sample" class="section level3">
<h3>Sample</h3>
<p><img src="figure/organoid-01-clustering.Rmd/dim-red-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dim-red-2">
Past versions of dim-red-2.png
</button>
</p>
<div id="fig-dim-red-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/5b37055b675c0a2c7273daf196d71bce8efaf677/docs/figure/organoid-01-clustering.Rmd/dim-red-2.png" target="_blank">5b37055</a>
</td>
<td>
khembach
</td>
<td>
2020-08-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="stage" class="section level3">
<h3>Stage</h3>
<p><img src="figure/organoid-01-clustering.Rmd/dim-red-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dim-red-3">
Past versions of dim-red-3.png
</button>
</p>
<div id="fig-dim-red-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/5b37055b675c0a2c7273daf196d71bce8efaf677/docs/figure/organoid-01-clustering.Rmd/dim-red-3.png" target="_blank">5b37055</a>
</td>
<td>
khembach
</td>
<td>
2020-08-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="cl_fulllineage" class="section level3">
<h3>cl_FullLineage</h3>
<p><img src="figure/organoid-01-clustering.Rmd/dim-red-4.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dim-red-4">
Past versions of dim-red-4.png
</button>
</p>
<div id="fig-dim-red-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/5b37055b675c0a2c7273daf196d71bce8efaf677/docs/figure/organoid-01-clustering.Rmd/dim-red-4.png" target="_blank">5b37055</a>
</td>
<td>
khembach
</td>
<td>
2020-08-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="ident" class="section level3">
<h3>ident</h3>
<p><img src="figure/organoid-01-clustering.Rmd/dim-red-5.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-dim-red-5">
Past versions of dim-red-5.png
</button>
</p>
<div id="fig-dim-red-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/5b37055b675c0a2c7273daf196d71bce8efaf677/docs/figure/organoid-01-clustering.Rmd/dim-red-5.png" target="_blank">5b37055</a>
</td>
<td>
khembach
</td>
<td>
2020-08-12
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>so_sample &lt;- so</code></pre>
</div>
</div>
<div id="reproduction-of-the-paper-figures" class="section level2">
<h2>Reproduction of the paper figures</h2>
<p>In the paper, the authors used CCA to integrate the H9 and 409b2 cells in a single tSNE plot. We try to use the exact same approach to see if we can reproduce their Extended data figure 2a.<br />
</p>
<pre class="r"><code>## They describe their pipeline as follows:
## Seurat v3 CCA
## log-normalisation
## highly variable genes with vst (for 409b2 and H9 cells)
## integration using top 20 CCs using the Seurat method by identifying anchors and integrating the datasets
## scale data
## PCA
## clustering based on top 20 PCs, resolution of 0.6
## feature plots with non-integrated expression values

# create SeuratObject for the H9 and 409b2 cells
so &lt;- CreateSeuratObject(
    counts = counts(sce),
    meta.data = data.frame(colData(sce)),
    project = &quot;organoids&quot;)

# split by cell line instead of sample as before
cells_by_sample &lt;- split(colnames(sce), sce$Line)
so &lt;- lapply(cells_by_sample, function(i) subset(so, cells = i))

## log normalize the data using a scaling factor of 10000
so &lt;- lapply(so, NormalizeData, verbose = FALSE, scale.factor = 10000, 
             normalization.method = &quot;LogNormalize&quot;)

## Identify the top 2000 genes with high cell-to-cell variation
so &lt;- lapply(so, FindVariableFeatures, nfeatures = 2000, 
    selection.method = &quot;vst&quot;, verbose = FALSE)</code></pre>
<pre class="r"><code># find anchors &amp; integrate
as &lt;- FindIntegrationAnchors(so, verbose = FALSE, dims = 1:20)
so &lt;- IntegrateData(anchorset = as, dims = seq_len(20), verbose = FALSE)
# ## We scale the data without regressing out any factors
so &lt;- ScaleData(so, verbose = FALSE)</code></pre>
<pre class="r"><code>so &lt;- RunPCA(so, npcs = 30, verbose = FALSE)
so &lt;- RunTSNE(so, reduction = &quot;pca&quot;, dims = seq_len(20),
    seed.use = 1, do.fast = TRUE, verbose = FALSE)
so &lt;- RunUMAP(so, reduction = &quot;pca&quot;, dims = seq_len(20),
    seed.use = 1, verbose = FALSE)

## PCA plot 
DimPlot(so, reduction = &quot;pca&quot;, group.by = &quot;sample_id&quot;)</code></pre>
<p><img src="figure/organoid-01-clustering.Rmd/reproduce-plots-DR-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># elbow plot with the ranking of PCs based on the % of variance explained
ElbowPlot(so, ndims = 30)</code></pre>
<p><img src="figure/organoid-01-clustering.Rmd/reproduce-plots-DR-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>so &lt;- FindNeighbors(so, reduction = &quot;pca&quot;, dims = seq_len(20), verbose = FALSE)
so &lt;- FindClusters(so, resolution = 0.6, random.seed = 1, verbose = FALSE)</code></pre>
</div>
<div id="dimension-reduction-plots-1" class="section level2 tabset">
<h2>Dimension reduction plots</h2>
<p>We plot the dimension reduction (DR) and color by cell line, sample, predicted cell type, cluster ID</p>
<pre class="r"><code># set cluster IDs to resolution 0.4 clustering
so &lt;- SetIdent(so, value = &quot;integrated_snn_res.0.6&quot;)
so@meta.data$cluster_id &lt;- Idents(so)

cs &lt;- sample(colnames(so), 10e3)
.plot_dr &lt;- function(so, dr, id)
    DimPlot(so, cells = cs, group.by = id, reduction = dr, pt.size = 0.4) +
        guides(col = guide_legend(nrow = 11, 
            override.aes = list(size = 3, alpha = 1))) +
        theme_void() + theme(aspect.ratio = 1)
ids &lt;- c(&quot;Line&quot;, &quot;Sample&quot;, &quot;Stage&quot;,&quot;cl_FullLineage&quot;, &quot;ident&quot;)
for (id in ids) {
    cat(&quot;### &quot;, id, &quot;\n&quot;)
    p1 &lt;- .plot_dr(so, &quot;tsne&quot;, id)
    lgd &lt;- get_legend(p1)
    p1 &lt;- p1 + theme(legend.position = &quot;none&quot;)
    p2 &lt;- .plot_dr(so, &quot;umap&quot;, id) + theme(legend.position = &quot;none&quot;)
    ps &lt;- plot_grid(plotlist = list(p1, p2), nrow = 1)
    p &lt;- plot_grid(ps, lgd, nrow = 1, rel_widths = c(1, 0.5))
    print(p)
    cat(&quot;\n\n&quot;)
}</code></pre>
<div id="line-1" class="section level3">
<h3>Line</h3>
<p><img src="figure/organoid-01-clustering.Rmd/reproduce-plots-dim-red-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="sample-1" class="section level3">
<h3>Sample</h3>
<p><img src="figure/organoid-01-clustering.Rmd/reproduce-plots-dim-red-2.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="stage-1" class="section level3">
<h3>Stage</h3>
<p><img src="figure/organoid-01-clustering.Rmd/reproduce-plots-dim-red-3.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="cl_fulllineage-1" class="section level3">
<h3>cl_FullLineage</h3>
<p><img src="figure/organoid-01-clustering.Rmd/reproduce-plots-dim-red-4.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="ident-1" class="section level3">
<h3>ident</h3>
<p><img src="figure/organoid-01-clustering.Rmd/reproduce-plots-dim-red-5.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="save-seurat-object-to-rds" class="section level2">
<h2>Save Seurat object to RDS</h2>
<pre class="r"><code>## Seurat object integrated by sample
saveRDS(so_sample, file.path(&quot;output&quot;, &quot;so_sample_organoid-01-clustering.rds&quot;))
## Seurat object integrated by cell line
saveRDS(so, file.path(&quot;output&quot;, &quot;so_organoid-01-clustering.rds&quot;))
saveRDS(sce_all, file.path(&quot;output&quot;, &quot;sce_organoid-01-clustering.rds&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.6 LTS

Matrix products: default
BLAS:   /usr/local/R/R-4.0.0/lib/libRblas.so
LAPACK: /usr/local/R/R-4.0.0/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] data.table_1.12.8           future_1.17.0              
 [3] rtracklayer_1.48.0          stringr_1.4.0              
 [5] Seurat_3.1.5                cowplot_1.0.0              
 [7] dplyr_1.0.0                 scater_1.16.2              
 [9] ggplot2_3.3.2               BiocParallel_1.22.0        
[11] scDblFinder_1.2.0           DropletUtils_1.8.0         
[13] SingleCellExperiment_1.10.1 SummarizedExperiment_1.18.1
[15] DelayedArray_0.14.0         matrixStats_0.56.0         
[17] Biobase_2.48.0              GenomicRanges_1.40.0       
[19] GenomeInfoDb_1.24.2         IRanges_2.22.2             
[21] S4Vectors_0.26.1            BiocGenerics_0.34.0        
[23] workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] backports_1.1.8           plyr_1.8.6               
  [3] igraph_1.2.5              lazyeval_0.2.2           
  [5] splines_4.0.0             listenv_0.8.0            
  [7] digest_0.6.25             htmltools_0.5.0          
  [9] viridis_0.5.1             magrittr_1.5             
 [11] cluster_2.1.0             ROCR_1.0-11              
 [13] limma_3.44.3              globals_0.12.5           
 [15] Biostrings_2.56.0         R.utils_2.9.2            
 [17] colorspace_1.4-1          rappdirs_0.3.1           
 [19] ggrepel_0.8.2             xfun_0.15                
 [21] crayon_1.3.4              RCurl_1.98-1.2           
 [23] jsonlite_1.7.0            survival_3.2-3           
 [25] zoo_1.8-8                 ape_5.4                  
 [27] glue_1.4.1                gtable_0.3.0             
 [29] zlibbioc_1.34.0           XVector_0.28.0           
 [31] leiden_0.3.3              BiocSingular_1.4.0       
 [33] Rhdf5lib_1.10.0           future.apply_1.6.0       
 [35] HDF5Array_1.16.1          scales_1.1.1             
 [37] edgeR_3.30.3              Rcpp_1.0.4.6             
 [39] viridisLite_0.3.0         reticulate_1.16          
 [41] dqrng_0.2.1               rsvd_1.0.3               
 [43] tsne_0.1-3                htmlwidgets_1.5.1        
 [45] httr_1.4.1                RColorBrewer_1.1-2       
 [47] ellipsis_0.3.1            ica_1.0-2                
 [49] farver_2.0.3              pkgconfig_2.0.3          
 [51] XML_3.99-0.4              R.methodsS3_1.8.0        
 [53] uwot_0.1.8                locfit_1.5-9.4           
 [55] labeling_0.3              tidyselect_1.1.0         
 [57] rlang_0.4.6               reshape2_1.4.4           
 [59] later_1.1.0.1             munsell_0.5.0            
 [61] tools_4.0.0               generics_0.0.2           
 [63] ggridges_0.5.2            evaluate_0.14            
 [65] yaml_2.2.1                knitr_1.29               
 [67] fs_1.4.2                  fitdistrplus_1.1-1       
 [69] purrr_0.3.4               randomForest_4.6-14      
 [71] RANN_2.6.1                pbapply_1.4-2            
 [73] nlme_3.1-148              whisker_0.4              
 [75] scran_1.16.0              R.oo_1.23.0              
 [77] compiler_4.0.0            beeswarm_0.2.3           
 [79] plotly_4.9.2.1            png_0.1-7                
 [81] tibble_3.0.1              statmod_1.4.34           
 [83] stringi_1.4.6             RSpectra_0.16-0          
 [85] lattice_0.20-41           Matrix_1.2-18            
 [87] vctrs_0.3.1               pillar_1.4.4             
 [89] lifecycle_0.2.0           lmtest_0.9-37            
 [91] RcppAnnoy_0.0.16          BiocNeighbors_1.6.0      
 [93] bitops_1.0-6              irlba_2.3.3              
 [95] httpuv_1.5.4              patchwork_1.0.1          
 [97] R6_2.4.1                  promises_1.1.1           
 [99] KernSmooth_2.23-17        gridExtra_2.3            
[101] vipor_0.4.5               codetools_0.2-16         
[103] MASS_7.3-51.6             rhdf5_2.32.2             
[105] rprojroot_1.3-2           withr_2.2.0              
[107] GenomicAlignments_1.24.0  Rsamtools_2.4.0          
[109] sctransform_0.2.1         GenomeInfoDbData_1.2.3   
[111] grid_4.0.0                tidyr_1.1.0              
[113] rmarkdown_2.3             DelayedMatrixStats_1.10.1
[115] Rtsne_0.15                git2r_0.27.1             
[117] ggbeeswarm_0.6.0         </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
