<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Katharina Hembach" />

<meta name="date" content="2021-12-02" />

<title>Stathmin 2 read coverage</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">neural_scRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/khembach/neural_scRNAseq">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Stathmin 2 read coverage</h1>
<h4 class="author">Katharina Hembach</h4>
<h4 class="date">25/02/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>neural_scRNAseq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200522code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200522)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200522code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200522)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtree85bfc607e67067dddf0d9eefe90b2a30a3e7d67btargetblank85bfc60a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/khembach/neural_scRNAseq/tree/85bfc607e67067dddf0d9eefe90b2a30a3e7d67b" target="_blank">85bfc60</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtree85bfc607e67067dddf0d9eefe90b2a30a3e7d67btargetblank85bfc60a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/khembach/neural_scRNAseq/tree/85bfc607e67067dddf0d9eefe90b2a30a3e7d67b" target="_blank">85bfc60</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    ._Rplots.pdf
    Ignored:    ._Unfiltered.pdf
    Ignored:    .__workflowr.yml
    Ignored:    ._coverage.pdf
    Ignored:    ._coverage_sashimi.pdf
    Ignored:    ._coverage_sashimi.png
    Ignored:    ._neural_scRNAseq.Rproj
    Ignored:    ._sashimi.pdf
    Ignored:    ._stmn2.pdf
    Ignored:    ._tdp.pdf
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/._.DS_Store
    Ignored:    analysis/._01-preprocessing.Rmd
    Ignored:    analysis/._01-preprocessing.html
    Ignored:    analysis/._02.1-SampleQC.Rmd
    Ignored:    analysis/._03-filtering.Rmd
    Ignored:    analysis/._04-clustering.Rmd
    Ignored:    analysis/._04-clustering.knit.md
    Ignored:    analysis/._04.1-cell_cycle.Rmd
    Ignored:    analysis/._05-annotation.Rmd
    Ignored:    analysis/._Lam-0-NSC_no_integration.Rmd
    Ignored:    analysis/._Lam-01-NSC_integration.Rmd
    Ignored:    analysis/._Lam-02-NSC_annotation.Rmd
    Ignored:    analysis/._NSC-1-clustering.Rmd
    Ignored:    analysis/._NSC-2-annotation.Rmd
    Ignored:    analysis/.__site.yml
    Ignored:    analysis/._additional_filtering.Rmd
    Ignored:    analysis/._additional_filtering_clustering.Rmd
    Ignored:    analysis/._index.Rmd
    Ignored:    analysis/._organoid-01-1-qualtiy-control.Rmd
    Ignored:    analysis/._organoid-01-clustering.Rmd
    Ignored:    analysis/._organoid-02-integration.Rmd
    Ignored:    analysis/._organoid-03-cluster_analysis.Rmd
    Ignored:    analysis/._organoid-04-group_integration.Rmd
    Ignored:    analysis/._organoid-04-stage_integration.Rmd
    Ignored:    analysis/._organoid-05-group_integration_cluster_analysis.Rmd
    Ignored:    analysis/._organoid-05-stage_integration_cluster_analysis.Rmd
    Ignored:    analysis/._organoid-06-1-prepare-sce.Rmd
    Ignored:    analysis/._organoid-06-conos-analysis-Seurat.Rmd
    Ignored:    analysis/._organoid-06-conos-analysis-function.Rmd
    Ignored:    analysis/._organoid-06-conos-analysis.Rmd
    Ignored:    analysis/._organoid-06-group-integration-conos-analysis.Rmd
    Ignored:    analysis/._organoid-07-conos-visualization.Rmd
    Ignored:    analysis/._organoid-07-group-integration-conos-visualization.Rmd
    Ignored:    analysis/._organoid-08-conos-comparison.Rmd
    Ignored:    analysis/._organoid-0x-sample_integration.Rmd
    Ignored:    analysis/01-preprocessing_cache/
    Ignored:    analysis/02-1-SampleQC_cache/
    Ignored:    analysis/02-quality_control_cache/
    Ignored:    analysis/02.1-SampleQC_cache/
    Ignored:    analysis/03-filtering_cache/
    Ignored:    analysis/04-clustering_cache/
    Ignored:    analysis/04.1-cell_cycle_cache/
    Ignored:    analysis/05-annotation_cache/
    Ignored:    analysis/06-clustering-all-timepoints_cache/
    Ignored:    analysis/07-cluster-analysis-all-timepoints_cache/
    Ignored:    analysis/Lam-01-NSC_integration_cache/
    Ignored:    analysis/Lam-02-NSC_annotation_cache/
    Ignored:    analysis/NSC-1-clustering_cache/
    Ignored:    analysis/NSC-2-annotation_cache/
    Ignored:    analysis/TDP-01-preprocessing_cache/
    Ignored:    analysis/TDP-02-quality_control_cache/
    Ignored:    analysis/TDP-03-filtering_cache/
    Ignored:    analysis/TDP-04-clustering_cache/
    Ignored:    analysis/TDP-05-00-filtering-plasmid-QC_cache/
    Ignored:    analysis/TDP-05-plasmid_expression_cache/
    Ignored:    analysis/TDP-06-cluster_analysis_cache/
    Ignored:    analysis/TDP-07-cluster_12_cache/
    Ignored:    analysis/TDP-08-00-clustering-HA-D96_cache/
    Ignored:    analysis/TDP-08-clustering-timeline-HA_cache/
    Ignored:    analysis/additional_filtering_cache/
    Ignored:    analysis/additional_filtering_clustering_cache/
    Ignored:    analysis/organoid-01-1-qualtiy-control_cache/
    Ignored:    analysis/organoid-01-clustering_cache/
    Ignored:    analysis/organoid-02-integration_cache/
    Ignored:    analysis/organoid-03-cluster_analysis_cache/
    Ignored:    analysis/organoid-04-group_integration_cache/
    Ignored:    analysis/organoid-04-stage_integration_cache/
    Ignored:    analysis/organoid-05-group_integration_cluster_analysis_cache/
    Ignored:    analysis/organoid-05-stage_integration_cluster_analysis_cache/
    Ignored:    analysis/organoid-06-conos-analysis_cache/
    Ignored:    analysis/organoid-06-conos-analysis_test_cache/
    Ignored:    analysis/organoid-06-group-integration-conos-analysis_cache/
    Ignored:    analysis/organoid-07-conos-visualization_cache/
    Ignored:    analysis/organoid-07-group-integration-conos-visualization_cache/
    Ignored:    analysis/organoid-08-conos-comparison_cache/
    Ignored:    analysis/organoid-0x-sample_integration_cache/
    Ignored:    analysis/sample5_QC_cache/
    Ignored:    analysis/timepoints-01-organoid-integration_cache/
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._.smbdeleteAAA17ed8b4b
    Ignored:    data/._Lam_figure2_markers.R
    Ignored:    data/._Reactive_astrocytes_markers.xlsx
    Ignored:    data/._known_NSC_markers.R
    Ignored:    data/._known_cell_type_markers.R
    Ignored:    data/._metadata.csv
    Ignored:    data/._virus_cell_tropism_markers.R
    Ignored:    data/._~$Reactive_astrocytes_markers.xlsx
    Ignored:    data/data_sushi/
    Ignored:    data/filtered_feature_matrices/
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/._NSC_cluster2_marker_genes.txt
    Ignored:    output/._TDP-06-no_integration_cluster12_marker_genes.txt
    Ignored:    output/._TDP-06-no_integration_cluster13_marker_genes.txt
    Ignored:    output/._organoid_integration_cluster1_marker_genes.txt
    Ignored:    output/Lam-01-clustering.rds
    Ignored:    output/NSC_1_clustering.rds
    Ignored:    output/NSC_cluster1_marker_genes.txt
    Ignored:    output/NSC_cluster2_marker_genes.txt
    Ignored:    output/NSC_cluster3_marker_genes.txt
    Ignored:    output/NSC_cluster4_marker_genes.txt
    Ignored:    output/NSC_cluster5_marker_genes.txt
    Ignored:    output/NSC_cluster6_marker_genes.txt
    Ignored:    output/NSC_cluster7_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster0_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster10_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster11_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster12_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster13_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster14_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster15_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster16_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster17_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster1_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster2_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster3_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster4_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster5_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster6_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster7_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster8_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster9_marker_genes.txt
    Ignored:    output/TDP-06_scran_markers.rds
    Ignored:    output/additional_filtering.rds
    Ignored:    output/conos/
    Ignored:    output/conos_organoid-06-conos-analysis.rds
    Ignored:    output/conos_organoid-06-group-integration-conos-analysis.rds
    Ignored:    output/figures/
    Ignored:    output/organoid_integration_cluster10_marker_genes.txt
    Ignored:    output/organoid_integration_cluster11_marker_genes.txt
    Ignored:    output/organoid_integration_cluster12_marker_genes.txt
    Ignored:    output/organoid_integration_cluster13_marker_genes.txt
    Ignored:    output/organoid_integration_cluster14_marker_genes.txt
    Ignored:    output/organoid_integration_cluster15_marker_genes.txt
    Ignored:    output/organoid_integration_cluster16_marker_genes.txt
    Ignored:    output/organoid_integration_cluster17_marker_genes.txt
    Ignored:    output/organoid_integration_cluster1_marker_genes.txt
    Ignored:    output/organoid_integration_cluster2_marker_genes.txt
    Ignored:    output/organoid_integration_cluster3_marker_genes.txt
    Ignored:    output/organoid_integration_cluster4_marker_genes.txt
    Ignored:    output/organoid_integration_cluster5_marker_genes.txt
    Ignored:    output/organoid_integration_cluster6_marker_genes.txt
    Ignored:    output/organoid_integration_cluster7_marker_genes.txt
    Ignored:    output/organoid_integration_cluster8_marker_genes.txt
    Ignored:    output/organoid_integration_cluster9_marker_genes.txt
    Ignored:    output/sce_01_preprocessing.rds
    Ignored:    output/sce_02_quality_control.rds
    Ignored:    output/sce_03_filtering.rds
    Ignored:    output/sce_03_filtering_all_genes.rds
    Ignored:    output/sce_06-1-prepare-sce.rds
    Ignored:    output/sce_TDP_01_preprocessing.rds
    Ignored:    output/sce_TDP_02_quality_control.rds
    Ignored:    output/sce_TDP_03_filtering.rds
    Ignored:    output/sce_TDP_03_filtering_all_genes.rds
    Ignored:    output/sce_organoid-01-clustering.rds
    Ignored:    output/sce_preprocessing.rds
    Ignored:    output/so_04-group_integration.rds
    Ignored:    output/so_04-stage_integration.rds
    Ignored:    output/so_04_1_cell_cycle.rds
    Ignored:    output/so_04_clustering.rds
    Ignored:    output/so_06-clustering_all_timepoints.rds
    Ignored:    output/so_08-00_clustering_HA_D96.rds
    Ignored:    output/so_08-clustering_timeline_HA.rds
    Ignored:    output/so_0x-sample_integration.rds
    Ignored:    output/so_TDP-06-cluster-analysis.rds
    Ignored:    output/so_TDP_04_clustering.rds
    Ignored:    output/so_TDP_05_plasmid_expression.rds
    Ignored:    output/so_additional_filtering_clustering.rds
    Ignored:    output/so_integrated_organoid-02-integration.rds
    Ignored:    output/so_merged_organoid-02-integration.rds
    Ignored:    output/so_organoid-01-clustering.rds
    Ignored:    output/so_sample_organoid-01-clustering.rds
    Ignored:    scripts/.DS_Store
    Ignored:    scripts/._.DS_Store
    Ignored:    scripts/._bu_Rcode.R
    Ignored:    scripts/._plasmid_expression.sh

Untracked files:
    Untracked:  Filtered.pdf
    Untracked:  Rplots.pdf
    Untracked:  Unfiltered
    Untracked:  Unfiltered.pdf
    Untracked:  analysis/Lam-0-NSC_no_integration.Rmd
    Untracked:  analysis/TDP-07-01-STMN2_expression copy.Rmd
    Untracked:  analysis/TDP-08-01-HA-D96-expression-changes.Rmd
    Untracked:  analysis/additional_filtering.Rmd
    Untracked:  analysis/additional_filtering_clustering.Rmd
    Untracked:  analysis/organoid-01-1-qualtiy-control.Rmd
    Untracked:  analysis/organoid-06-conos-analysis-Seurat.Rmd
    Untracked:  analysis/organoid-06-conos-analysis-function.Rmd
    Untracked:  analysis/organoid-07-conos-visualization.Rmd
    Untracked:  analysis/organoid-07-group-integration-conos-visualization.Rmd
    Untracked:  analysis/organoid-08-conos-comparison.Rmd
    Untracked:  analysis/organoid-0x-sample_integration.Rmd
    Untracked:  analysis/sample5_QC.Rmd
    Untracked:  coverage.pdf
    Untracked:  coverage_sashimi.pdf
    Untracked:  coverage_sashimi.png
    Untracked:  data/Homo_sapiens.GRCh38.98.sorted.gtf
    Untracked:  data/Kanton_et_al/
    Untracked:  data/Lam_et_al/
    Untracked:  data/Sep2020/
    Untracked:  data/reference/
    Untracked:  data/virus_cell_tropism_markers.R
    Untracked:  data/~$Reactive_astrocytes_markers.xlsx
    Untracked:  sashimi.pdf
    Untracked:  scripts/bu_Rcode.R
    Untracked:  scripts/salmon-latest_linux_x86_64/
    Untracked:  stmn2.pdf
    Untracked:  tdp.pdf

Unstaged changes:
    Modified:   analysis/05-annotation.Rmd
    Modified:   analysis/Lam-02-NSC_annotation.Rmd
    Modified:   analysis/TDP-04-clustering.Rmd
    Modified:   analysis/TDP-06-cluster_analysis.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/organoid-02-integration.Rmd
    Modified:   analysis/organoid-04-group_integration.Rmd
    Modified:   analysis/organoid-06-conos-analysis.Rmd
    Modified:   analysis/timepoints-01-organoid-integration.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/TDP-07-01-STMN2_expression.Rmd</code>) and HTML (<code>docs/TDP-07-01-STMN2_expression.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/85bfc607e67067dddf0d9eefe90b2a30a3e7d67b/analysis/TDP-07-01-STMN2_expression.Rmd" target="_blank">85bfc60</a>
</td>
<td>
khembach
</td>
<td>
2021-02-25
</td>
<td>
add cryptic exon location and splice junctions to plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/khembach/neural_scRNAseq/a02eb1b19e76fcfb3648951cda280bde1453103f/docs/TDP-07-01-STMN2_expression.html" target="_blank">a02eb1b</a>
</td>
<td>
khembach
</td>
<td>
2021-02-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/f01a91a68643ad3dadd32ad8a99a3b84cc71905b/analysis/TDP-07-01-STMN2_expression.Rmd" target="_blank">f01a91a</a>
</td>
<td>
khembach
</td>
<td>
2021-02-25
</td>
<td>
add cryptic exon location and splice junctions to plot
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/khembach/neural_scRNAseq/5340095cabde7838184cf8f540ac0978eb007f7c/docs/TDP-07-01-STMN2_expression.html" target="_blank">5340095</a>
</td>
<td>
khembach
</td>
<td>
2021-02-19
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/4ddbb0db1f0a94a96be64987f00d34f3a9046086/analysis/TDP-07-01-STMN2_expression.Rmd" target="_blank">4ddbb0d</a>
</td>
<td>
khembach
</td>
<td>
2021-02-19
</td>
<td>
text size
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/khembach/neural_scRNAseq/a7c4a5b9fc772bcb6014673ab5ad289bd80202a3/docs/TDP-07-01-STMN2_expression.html" target="_blank">a7c4a5b</a>
</td>
<td>
khembach
</td>
<td>
2021-02-18
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/1755a273dea1064cbda0362daca89c86cf476e36/analysis/TDP-07-01-STMN2_expression.Rmd" target="_blank">1755a27</a>
</td>
<td>
khembach
</td>
<td>
2021-02-18
</td>
<td>
plot stathmin2 read coverage of cells from cluster 12
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>library(Seurat)
library(SingleCellExperiment)
library(dplyr)
library(Gviz)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(Rsamtools)
library(GenomicAlignments)
library(rtracklayer)</code></pre>
</div>
<div id="load-data-convert-to-sce" class="section level1">
<h1>Load data &amp; convert to SCE</h1>
<pre class="r"><code>so &lt;- readRDS(file.path(&quot;output&quot;, &quot;so_TDP_05_plasmid_expression.rds&quot;))
so &lt;- SetIdent(so, value = &quot;RNA_snn_res.0.4&quot;)
so@meta.data$cluster_id &lt;- Idents(so)</code></pre>
</div>
<div id="cells-from-cluster-12" class="section level1">
<h1>Cells from cluster 12</h1>
<p>We want to compare the stathmin2 read coverage of cells expressing TDP-HA (from cluster 12) and other neuronal cells without TDP-HA expression. For this, we randomly select 5 cells from each group and filter the corresponding stathmin2 reads from the BAM file.</p>
<pre class="r"><code>clus12 &lt;- subset(so, subset = cluster_id == &quot;12&quot;)
## from which sample do the cells come from?
clus12$sample_id %&gt;% table</code></pre>
<pre><code>.
 TDP2wON TDP4wOFF TDP4wONa TDP4wONb 
      97        3       88       36 </code></pre>
<pre class="r"><code>## what is the range of TDP-HA expression in all cells in cluster 12?
dat_ha &lt;- GetAssayData(object = clus12, slot = &quot;data&quot;)[&quot;TDP43-HA&quot;,]
summary(dat_ha)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  0.000   1.226   1.771   1.800   2.454   4.112 </code></pre>
<pre class="r"><code>## select cells with high TDP-HA expression
high &lt;- clus12[which(dat_ha &gt; 3.5)]
high$barcode %&gt;% head</code></pre>
<pre><code>AAAGGATGTGTCATTG-1.TDP2wON AAAGTGAAGAGATTCA-1.TDP2wON 
      &quot;AAAGGATGTGTCATTG-1&quot;       &quot;AAAGTGAAGAGATTCA-1&quot; 
AACAACCGTGGCCTCA-1.TDP2wON AAGAACAAGCGTACAG-1.TDP2wON 
      &quot;AACAACCGTGGCCTCA-1&quot;       &quot;AAGAACAAGCGTACAG-1&quot; 
AAGACTCGTTGCAACT-1.TDP2wON AATCACGCAAATGGAT-1.TDP2wON 
      &quot;AAGACTCGTTGCAACT-1&quot;       &quot;AATCACGCAAATGGAT-1&quot; </code></pre>
<pre class="r"><code>## cells with low TDP-HA expression
low &lt;- clus12[which(dat_ha &lt; 0.5 &amp; dat_ha &gt; 0)]
low$barcode %&gt;% head</code></pre>
<pre><code>AAAGGATGTGTCATTG-1.TDP2wON AAAGTGAAGAGATTCA-1.TDP2wON 
      &quot;AAAGGATGTGTCATTG-1&quot;       &quot;AAAGTGAAGAGATTCA-1&quot; 
AACAACCGTGGCCTCA-1.TDP2wON AAGAACAAGCGTACAG-1.TDP2wON 
      &quot;AACAACCGTGGCCTCA-1&quot;       &quot;AAGAACAAGCGTACAG-1&quot; 
AAGACTCGTTGCAACT-1.TDP2wON AATCACGCAAATGGAT-1.TDP2wON 
      &quot;AAGACTCGTTGCAACT-1&quot;       &quot;AATCACGCAAATGGAT-1&quot; </code></pre>
</div>
<div id="get-stathmin2-reads-of-selected-cells" class="section level1">
<h1>Get stathmin2 reads of selected cells</h1>
<p>We extract the reads covering the stathmin2 genes of the selected cells.</p>
<pre class="r"><code>bams &lt;- list(TDP4wOFF = file.path(&quot;data&quot;, &quot;Sep2020&quot;, &quot;CellRangerCount_50076_2020-09-22--15-40-54&quot;,
            &quot;no1_Neural_cuture_d_96_TDP-43-HA_4w_DOXoff&quot;,
            &quot;possorted_genome_bam.bam&quot;),
            TDP2wON = file.path(&quot;data&quot;, &quot;Sep2020&quot;, &quot;CellRangerCount_50076_2020-09-22--15-40-54&quot;,
            &quot;no2_Neural_cuture_d_96_TDP-43-HA_2w_DOXON&quot;,
            &quot;possorted_genome_bam.bam&quot;),
            TDP4wONa = file.path(&quot;data&quot;, &quot;Sep2020&quot;, &quot;CellRangerCount_50076_2020-09-22--15-40-54&quot;,
            &quot;no3_Neural_cuture_d_96_TDP-43-HA_4w_DOXONa&quot;,
            &quot;possorted_genome_bam.bam&quot;),
            TDP4wONb = file.path(&quot;data&quot;, &quot;Sep2020&quot;, &quot;CellRangerCount_50076_2020-09-22--15-40-54&quot;,
            &quot;no4_Neural_cuture_d_96_TDP-43-HA_4w_DOXONb&quot;,
            &quot;possorted_genome_bam.bam&quot;))

chr &lt;- &quot;chr8&quot;
region_start &lt;- 79611100
region_end &lt;- 79666200
stmn2 &lt;- GRanges(chr, IRanges(region_start, region_end), &quot;+&quot;)

# keep all reads from cells in cluster 12
param &lt;- ScanBamParam(which=stmn2, what = c(&quot;qname&quot;), tag = &quot;CB&quot;, 
                      tagFilter = list(CB = clus12$barcode))
gals &lt;- lapply(bams, function(x) {
  readGAlignments(x, use.names = TRUE, param=param)
  })

covs &lt;- lapply(gals, coverage)</code></pre>
</div>
<div id="ploting-with-gviz" class="section level1">
<h1>Ploting with Gviz</h1>
<p>Plot the stathmin2 transcripts and the read coverage of all cells from cluster 12.</p>
<pre class="r"><code>## gene annotations from UCSC
options(ucscChromosomeNames = FALSE)
eTrack &lt;- GeneRegionTrack(TxDb.Hsapiens.UCSC.hg38.knownGene, 
                          chromosome = chr, start = region_start, 
                          end = region_end, name = &quot;annotation&quot;)

coords &lt;- 79611100:79666201
dat &lt;- matrix(c(as.vector(covs[[1]]$chr8[region_start:region_end]),
                as.vector(covs[[2]]$chr8[region_start:region_end]),
                as.vector(covs[[3]]$chr8[region_start:region_end]),
                as.vector(covs[[4]]$chr8[region_start:region_end])), 
              nrow = 4, byrow = TRUE)
rownames(dat) &lt;- names(covs)
dtrack &lt;- DataTrack(data = dat,
                    start = coords[-length(coords)], end = coords[-1], chromosome = chr,
                    genome = &quot;hg38&quot;)

plotTracks(c(dtrack, eTrack),  
           type = &quot;histogram&quot;, showSampleNames = TRUE,
           shape = &quot;arrow&quot;, geneSymbols = TRUE, aggregateGroups=FALSE,
           groups = c(&quot;TDP4wOFF&quot;, &quot;TDP2wON&quot;, &quot;TDP4wONa&quot;, &quot;TDP4wONb&quot;),
           stackedBars = FALSE, fontsize=13 )</code></pre>
<p><img src="figure/TDP-07-01-STMN2_expression.Rmd/stmn2-clus12-coverage-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-stmn2-clus12-coverage-1">
Past versions of stmn2-clus12-coverage-1.png
</button>
</p>
<div id="fig-stmn2-clus12-coverage-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/5340095cabde7838184cf8f540ac0978eb007f7c/docs/figure/TDP-07-01-STMN2_expression.Rmd/stmn2-clus12-coverage-1.png" target="_blank">5340095</a>
</td>
<td>
khembach
</td>
<td>
2021-02-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/a7c4a5b9fc772bcb6014673ab5ad289bd80202a3/docs/figure/TDP-07-01-STMN2_expression.Rmd/stmn2-clus12-coverage-1.png" target="_blank">a7c4a5b</a>
</td>
<td>
khembach
</td>
<td>
2021-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## one data track per sample
dats &lt;- list(&quot;4wOFF&quot; = matrix(as.vector(covs[[1]]$chr8[region_start:region_end]),
                              nrow = 1, byrow = TRUE),
             &quot;2wON&quot; = matrix(as.vector(covs[[2]]$chr8[region_start:region_end]),
                              nrow = 1, byrow = TRUE),
             &quot;4wONa&quot; = matrix(as.vector(covs[[3]]$chr8[region_start:region_end]),
                              nrow = 1, byrow = TRUE),
             &quot;4wONb&quot; = matrix(as.vector(covs[[4]]$chr8[region_start:region_end]),
                              nrow = 1, byrow = TRUE))

# rownames(dat) &lt;- names(covs)
dtrack_4wOFF &lt;- DataTrack(data = dats[[1]],
                    start = coords[-length(coords)], end = coords[-1], chromosome = chr,
                    genome = &quot;hg38&quot;, name = &quot;4wOFF&quot;)
dtrack_2wON &lt;- DataTrack(data = dats[[2]],
                    start = coords[-length(coords)], end = coords[-1], chromosome = chr,
                    genome = &quot;hg38&quot;, name = &quot;2wON&quot;)
dtrack_4wONa &lt;- DataTrack(data = dats[[3]],
                    start = coords[-length(coords)], end = coords[-1], chromosome = chr,
                    genome = &quot;hg38&quot;, name = &quot;4wONa&quot;)
dtrack_4wONb &lt;- DataTrack(data = dats[[4]],
                    start = coords[-length(coords)], end = coords[-1], chromosome = chr,
                    genome = &quot;hg38&quot;, name = &quot;4wONb&quot;)

plotTracks(c(dtrack_4wOFF, dtrack_2wON, dtrack_4wONa, dtrack_4wONb, eTrack), 
           type = &quot;histogram&quot;, showSampleNames = TRUE,
           shape = &quot;arrow&quot;, geneSymbols = TRUE, aggregateGroups=FALSE,
           stackedBars = FALSE, fontsize=13)</code></pre>
<p><img src="figure/TDP-07-01-STMN2_expression.Rmd/stmn2-clus12-coverage-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-stmn2-clus12-coverage-2">
Past versions of stmn2-clus12-coverage-2.png
</button>
</p>
<div id="fig-stmn2-clus12-coverage-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/5340095cabde7838184cf8f540ac0978eb007f7c/docs/figure/TDP-07-01-STMN2_expression.Rmd/stmn2-clus12-coverage-2.png" target="_blank">5340095</a>
</td>
<td>
khembach
</td>
<td>
2021-02-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/a7c4a5b9fc772bcb6014673ab5ad289bd80202a3/docs/figure/TDP-07-01-STMN2_expression.Rmd/stmn2-clus12-coverage-2.png" target="_blank">a7c4a5b</a>
</td>
<td>
khembach
</td>
<td>
2021-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## zoom into intron 1 that contains the cryptic exon
chr &lt;- &quot;chr8&quot;
region_start &lt;- 79611100
region_end &lt;- 79637000

plotTracks(c(dtrack_4wOFF, dtrack_2wON, dtrack_4wONa, dtrack_4wONb, eTrack),
           type = &quot;histogram&quot;, showSampleNames = TRUE,
           chromosome = chr, from = region_start, to = region_end,
           shape = &quot;arrow&quot;, geneSymbols = TRUE, aggregateGroups=FALSE,
           stackedBars = FALSE, fontsize=13)</code></pre>
<p><img src="figure/TDP-07-01-STMN2_expression.Rmd/stmn2-clus12-coverage-3.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-stmn2-clus12-coverage-3">
Past versions of stmn2-clus12-coverage-3.png
</button>
</p>
<div id="fig-stmn2-clus12-coverage-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/5340095cabde7838184cf8f540ac0978eb007f7c/docs/figure/TDP-07-01-STMN2_expression.Rmd/stmn2-clus12-coverage-3.png" target="_blank">5340095</a>
</td>
<td>
khembach
</td>
<td>
2021-02-19
</td>
</tr>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/a7c4a5b9fc772bcb6014673ab5ad289bd80202a3/docs/figure/TDP-07-01-STMN2_expression.Rmd/stmn2-clus12-coverage-3.png" target="_blank">a7c4a5b</a>
</td>
<td>
khembach
</td>
<td>
2021-02-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## cryptic exon location: HG19 Chr8: 80,529,075-80,529,28
## in hg38: chr8:79616840-79617049
ce_start &lt;- 79616840
ce_end &lt;- 79617049
ceTrack &lt;- AnnotationTrack(start = ce_start, end = ce_end, chromosome = chr, 
                          strand = &quot;*&quot;, genome = &quot;hg38&quot;, name = &quot;CE&quot;)
plotTracks(c(dtrack_4wOFF, dtrack_2wON, dtrack_4wONa, dtrack_4wONb, 
             ceTrack, eTrack),
           type = &quot;histogram&quot;, showSampleNames = TRUE,
           chromosome = chr, from = region_start, to = region_end,
           shape = &quot;arrow&quot;, geneSymbols = TRUE, aggregateGroups=FALSE,
           stackedBars = FALSE, fontsize=13)</code></pre>
<p><img src="figure/TDP-07-01-STMN2_expression.Rmd/stmn2-clus12-coverage-4.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-stmn2-clus12-coverage-4">
Past versions of stmn2-clus12-coverage-4.png
</button>
</p>
<div id="fig-stmn2-clus12-coverage-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/a02eb1b19e76fcfb3648951cda280bde1453103f/docs/figure/TDP-07-01-STMN2_expression.Rmd/stmn2-clus12-coverage-4.png" target="_blank">a02eb1b</a>
</td>
<td>
khembach
</td>
<td>
2021-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="splice-junctions" class="section level2">
<h2>Splice junctions</h2>
<p>Are there any splice junctions in the first intron?</p>
<p>We define new functions that allow us to filter the BAM files basd on barcodes:</p>
<pre class="r"><code>####### import only reads from cells of cluster 12
## function is copid and modified from the Gviz package
.import.bam.alignments.cells &lt;- function(file, selection) {
    indNames &lt;- c(sub(&quot;\\.bam$&quot;, &quot;.bai&quot;, file), paste(file, &quot;bai&quot;, sep = &quot;.&quot;))
    index &lt;- NULL
    for (i in indNames) {
        if (file.exists(i)) {
            index &lt;- i
            break
        }
    }
    if (is.null(index)) {
          stop(
              &quot;Unable to find index for BAM file &#39;&quot;, file, &quot;&#39;. You can build an index using the following command:\n\t&quot;,
              &quot;library(Rsamtools)\n\tindexBam(\&quot;&quot;, file, &quot;\&quot;)&quot;
          )
      }
    pairedEnd &lt;- parent.env(environment())[[&quot;._isPaired&quot;]]
    if (is.null(pairedEnd)) {
          pairedEnd &lt;- TRUE
      }
    flag &lt;- parent.env(environment())[[&quot;._flag&quot;]]
    if (is.null(flag)) {
          flag &lt;- scanBamFlag(isUnmappedQuery = FALSE)
      }
    bf &lt;- BamFile(file, index = index, asMates = pairedEnd)
    cells &lt;- parent.env(environment())[[&quot;._cells&quot;]]
    if(!is.null(cells)){
      param &lt;- ScanBamParam(which = selection, what = scanBamWhat(),
                          tag = c(&quot;MD&quot;, &quot;CB&quot;), flag = flag, 
                          tagFilter = list(CB = cells))
    } else{
      param &lt;- ScanBamParam(which = selection, what = scanBamWhat(), 
                            tag = &quot;MD&quot;, flag = flag)
    }
    
    reads &lt;- if (as.character(seqnames(selection)[1]) %in% names(scanBamHeader(bf)$targets)) scanBam(bf, param = param)[[1]] else list()
    md &lt;- if (is.null(reads$tag$MD)) rep(as.character(NA), length(reads$pos)) else reads$tag$MD
    if (length(reads$pos)) {
        layed_seq &lt;- sequenceLayer(reads$seq, reads$cigar)
        region &lt;- unlist(bamWhich(param), use.names = FALSE)
        ans &lt;- stackStrings(layed_seq, start(region), end(region), shift = reads$pos - 1L, Lpadding.letter = &quot;+&quot;, Rpadding.letter = &quot;+&quot;)
        names(ans) &lt;- seq_along(reads$qname)
    } else {
        ans &lt;- DNAStringSet()
    }
    return(GRanges(
        seqnames = if (is.null(reads$rname)) character() else reads$rname,
        strand = if (is.null(reads$strand)) character() else reads$strand,
        ranges = IRanges(start = reads$pos, width = reads$qwidth),
        id = if (is.null(reads$qname)) character() else reads$qname,
        cigar = if (is.null(reads$cigar)) character() else reads$cigar,
        mapq = if (is.null(reads$mapq)) integer() else reads$mapq,
        flag = if (is.null(reads$flag)) integer() else reads$flag,
        md = md, seq = ans,
        isize = if (is.null(reads$isize)) integer() else reads$isize,
        groupid = if (pairedEnd) if (is.null(reads$groupid)) integer() else reads$groupid else seq_along(reads$pos),
        status = if (pairedEnd) {
            if (is.null(reads$mate_status)) factor(levels = c(&quot;mated&quot;, &quot;ambiguous&quot;, &quot;unmated&quot;)) else reads$mate_status
        } else {
            rep(
                factor(&quot;unmated&quot;, levels = c(&quot;mated&quot;, &quot;ambiguous&quot;, &quot;unmated&quot;)),
                length(reads$pos)
            )
        }
    ))
}


## Constructor
AlignmentsTrack &lt;- function(range = NULL, start = NULL, end = NULL, width = NULL, strand, chromosome, genome,
                            stacking = &quot;squish&quot;, id, cigar, mapq, flag = scanBamFlag(isUnmappedQuery = FALSE), isize, groupid, status, md, seqs,
                            name = &quot;AlignmentsTrack&quot;, isPaired = TRUE, importFunction, referenceSequence, cells = NULL, ...) {
    ## Some defaults
    if (missing(importFunction)) {
        importFunction &lt;- .import.bam.alignments
    }
    covars &lt;- Gviz:::.getCovars(range)
    isStream &lt;- FALSE
    if (!is.character(range)) {
        n &lt;- max(c(length(start), length(end), length(width)), nrow(covars))
        id &lt;- Gviz:::.covDefault(id, covars[[&quot;id&quot;]], paste(&quot;read&quot;, seq_len(n), sep = &quot;_&quot;))
        cigar &lt;- Gviz:::.covDefault(cigar, covars[[&quot;cigar&quot;]], paste(if (is(range, &quot;GRangesOrIRanges&quot;)) width(range) else width, &quot;M&quot;, sep = &quot;&quot;))
        mapq &lt;- Gviz:::.covDefault(mapq, covars[[&quot;mapq&quot;]], rep(as.integer(NA), n))
        flag &lt;- Gviz:::.covDefault(flag, covars[[&quot;flag&quot;]], rep(as.integer(NA), n))
        isize &lt;- Gviz:::.covDefault(isize, covars[[&quot;isize&quot;]], rep(as.integer(NA), n))
        groupid &lt;- Gviz:::.covDefault(groupid, covars[[&quot;groupid&quot;]], seq_len(n))
        md &lt;- Gviz:::.covDefault(md, covars[[&quot;md&quot;]], rep(as.character(NA), n))
        status &lt;- Gviz:::.covDefault(status, covars[[&quot;status&quot;]], ifelse(groupid %in% groupid[duplicated(groupid)], &quot;mated&quot;, &quot;unmated&quot;))
    }
    ## Build a GRanges object from the inputs
    Gviz:::.missingToNull(c(
        &quot;strand&quot;, &quot;chromosome&quot;, &quot;importFunction&quot;, &quot;genome&quot;, &quot;id&quot;, &quot;cigar&quot;, &quot;mapq&quot;, &quot;flag&quot;, &quot;isize&quot;, &quot;groupid&quot;, &quot;status&quot;,
        &quot;md&quot;, &quot;seqs&quot;, &quot;referenceSequence&quot;
    ))
    args &lt;- list(
        id = id, cigar = cigar, mapq = mapq, flag = flag, isize = isize, groupid = groupid, status = status, strand = strand, md = md,
        chromosome = chromosome, genome = genome
    )
    defs &lt;- list(
        strand = &quot;*&quot;, chromosome = &quot;chrNA&quot;, genome = NA, id = as.character(NA), cigar = as.character(NA), mapq = as.integer(NA),
        flag = as.integer(NA), isize = as.integer(NA), groupid = as.character(NA), status = as.character(NA), md = as.character(NA)
    )
    range &lt;- Gviz:::.buildRange(
        range = range, start = start, end = end, width = width,
        args = args, defaults = defs, chromosome = chromosome, trackType = &quot;AlignmentsTrack&quot;,
        importFun = importFunction, stream = TRUE, autodetect = TRUE, ...
    )
    ## This is going to be a list if we have to stream data from a file, otherwise we can compute some additional values
    if (is.list(range)) {
        isStream &lt;- TRUE
        slist &lt;- range
        range &lt;- GRanges()
        stackRanges &lt;- GRanges()
        stacks &lt;- NULL
        seqs &lt;- DNAStringSet()
    } else {
        if (is.null(seqs)) {
            seqs &lt;- DNAStringSet(vapply(width(range), function(x) paste(rep(&quot;N&quot;, x), collapse = &quot;&quot;), character(1)))
        }
        addArgs &lt;- list(...)
        if (&quot;showIndels&quot; %in% names(addArgs)) {
            showIndels &lt;- addArgs$showIndels
        } else {
            showIndels &lt;- FALSE
        }
        tmp &lt;- .computeAlignments(range, drop.D.ranges = showIndels)
        range &lt;- tmp$range
        stackRanges &lt;- tmp$stackRange
        stacks &lt;- tmp$stacks
    }
    ## If no chromosome was explicitly asked for we just take the first one in the GRanges object
    if (missing(chromosome) || is.null(chromosome)) {
        chromosome &lt;- if (length(range) &gt; 0) Gviz:::.chrName(as.character(seqnames(range)[1])) else &quot;chrNA&quot;
    }
    ## And finally the object instantiation
    genome &lt;- Gviz:::.getGenomeFromGRange(range, ifelse(is.null(genome), character(), genome[1]))
    if (!isStream) {
        return(new(&quot;AlignmentsTrack&quot;,
            chromosome = chromosome[1], range = range, stacks = stacks,
            name = name, genome = genome, stacking = stacking, stackRanges = stackRanges, sequences = seqs,
            referenceSequence = referenceSequence, ...
        ))
    } else {
        ## A bit hackish but for some functions we may want to know which track type we need but at the
        ## same time we do not want to enforce this as an additional argument
        e &lt;- new.env()
        e[[&quot;._trackType&quot;]] &lt;- &quot;AlignmentsTrack&quot;
        e[[&quot;._isPaired&quot;]] &lt;- isPaired
        e[[&quot;._flag&quot;]] &lt;- flag
        e[[&quot;._cells&quot;]] &lt;- cells
        environment(slist[[&quot;stream&quot;]]) &lt;- e
        return(new(&quot;ReferenceAlignmentsTrack&quot;,
            chromosome = chromosome[1], range = range, stackRanges = stackRanges,
            name = name, genome = genome, stacking = stacking, stream = slist[[&quot;stream&quot;]], reference = slist[[&quot;reference&quot;]],
            mapping = slist[[&quot;mapping&quot;]], args = args, defaults = defs, stacks = stacks, referenceSequence = referenceSequence, ...
        ))
    }
}</code></pre>
<pre class="r"><code>## STMN2 gene range
chr &lt;- &quot;chr8&quot;
region_start &lt;- 79611100
region_end &lt;- 79666200

# Create the alignments track
alTrack_4wOFF &lt;- AlignmentsTrack(
  range = bams[[&quot;TDP4wOFF&quot;]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == &quot;TDP4wOFF&quot;], 
  importFunction = .import.bam.alignments.cells, name = &quot;4wOFF&quot;)
alTrack_2wON &lt;- AlignmentsTrack(
  range = bams[[&quot;TDP2wON&quot;]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == &quot;TDP2wON&quot;], 
  importFunction = .import.bam.alignments.cells, name = &quot;2wON&quot;)
alTrack_4wONa &lt;- AlignmentsTrack(
  range = bams[[&quot;TDP4wONa&quot;]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == &quot;TDP4wONa&quot;], 
  importFunction = .import.bam.alignments.cells, name = &quot;4wONa&quot;)
alTrack_4wONb &lt;- AlignmentsTrack(
  range = bams[[&quot;TDP4wONb&quot;]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == &quot;TDP4wONb&quot;], 
  importFunction = .import.bam.alignments.cells, name = &quot;4wONb&quot;)

## import GTF with gene annotation
gtf &lt;- import(file.path(&quot;data&quot;, &quot;Homo_sapiens.GRCh38.98.sorted.gtf&quot;))
### import GTF, transfort to TxDb and create GeneRegionTrack
seqlevelsStyle(gtf) &lt;- &quot;UCSC&quot;
txdb &lt;- makeTxDbFromGRanges(gtf)</code></pre>
<pre><code>Warning in .get_cds_IDX(mcols0$type, mcols0$phase): The &quot;phase&quot; metadata column contains non-NA values for features of type
  stop_codon. This information was ignored.</code></pre>
<pre class="r"><code>gtftrack &lt;- GeneRegionTrack(txdb, name = &quot;annotation&quot;) 


## Whole gene
plotTracks(c(alTrack_4wOFF, alTrack_2wON, alTrack_4wONa, alTrack_4wONb,
             ceTrack, gtftrack),  
           type = c(&quot;coverage&quot;, &quot;sashimi&quot;), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100,
           fontsize=13,
           sizes = c(rep(3, 4), 1, 3),
           transcriptAnnotation = &quot;transcript&quot;)</code></pre>
<p><img src="figure/TDP-07-01-STMN2_expression.Rmd/stmn2-splice-junctions-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-stmn2-splice-junctions-1">
Past versions of stmn2-splice-junctions-1.png
</button>
</p>
<div id="fig-stmn2-splice-junctions-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/a02eb1b19e76fcfb3648951cda280bde1453103f/docs/figure/TDP-07-01-STMN2_expression.Rmd/stmn2-splice-junctions-1.png" target="_blank">a02eb1b</a>
</td>
<td>
khembach
</td>
<td>
2021-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## only first intron 
## zoom into intron 1 that contains the cryptic exon
region_start &lt;- 79611100
region_end &lt;- 79637000

plotTracks(c(alTrack_4wOFF, 
             alTrack_2wON, alTrack_4wONa, alTrack_4wONb,
             ceTrack, gtftrack),  
           type = c(&quot;coverage&quot;, &quot;sashimi&quot;), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100,
           fontsize=13,
           sizes = c(rep(3, 4), 1, 3),
           transcriptAnnotation = &quot;transcript&quot;)</code></pre>
<p><img src="figure/TDP-07-01-STMN2_expression.Rmd/stmn2-splice-junctions-2.png" width="1152" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-stmn2-splice-junctions-2">
Past versions of stmn2-splice-junctions-2.png
</button>
</p>
<div id="fig-stmn2-splice-junctions-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/a02eb1b19e76fcfb3648951cda280bde1453103f/docs/figure/TDP-07-01-STMN2_expression.Rmd/stmn2-splice-junctions-2.png" target="_blank">a02eb1b</a>
</td>
<td>
khembach
</td>
<td>
2021-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="read-coverage-in-cells-with-low-and-high-tdp-ha-expression-from-cluster-12" class="section level2">
<h2>Read coverage in cells with low and high TDP-HA expression from cluster 12</h2>
<pre class="r"><code>## LOW ##
chr &lt;- &quot;chr8&quot;
region_start &lt;- 79611100
region_end &lt;- 79666200

# Create the alignments track
alTrack_4wOFF &lt;- AlignmentsTrack(
  range = bams[[&quot;TDP4wOFF&quot;]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = low$barcode[low$sample_id == &quot;TDP4wOFF&quot;], 
  importFunction = .import.bam.alignments.cells, name = &quot;4wOFF&quot;)
alTrack_2wON &lt;- AlignmentsTrack(
  range = bams[[&quot;TDP2wON&quot;]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = low$barcode[low$sample_id == &quot;TDP2wON&quot;], 
  importFunction = .import.bam.alignments.cells, name = &quot;2wON&quot;)
alTrack_4wONa &lt;- AlignmentsTrack(
  range = bams[[&quot;TDP4wONa&quot;]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = low$barcode[low$sample_id == &quot;TDP4wONa&quot;], 
  importFunction = .import.bam.alignments.cells, name = &quot;4wONa&quot;)
alTrack_4wONb &lt;- AlignmentsTrack(
  range = bams[[&quot;TDP4wONb&quot;]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = low$barcode[low$sample_id == &quot;TDP4wONb&quot;], 
  importFunction = .import.bam.alignments.cells, name = &quot;4wONb&quot;)

plotTracks(c(alTrack_4wOFF, alTrack_2wON, alTrack_4wONa, alTrack_4wONb,
             ceTrack, gtftrack),  
           type = c(&quot;coverage&quot;, &quot;sashimi&quot;), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100,
           fontsize=13,
           sizes = c(rep(3, 4), 1, 3),
           transcriptAnnotation = &quot;transcript&quot;)</code></pre>
<p><img src="figure/TDP-07-01-STMN2_expression.Rmd/low-high-TDPHA-cells-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-low-high-TDPHA-cells-1">
Past versions of low-high-TDPHA-cells-1.png
</button>
</p>
<div id="fig-low-high-TDPHA-cells-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/a02eb1b19e76fcfb3648951cda280bde1453103f/docs/figure/TDP-07-01-STMN2_expression.Rmd/low-high-TDPHA-cells-1.png" target="_blank">a02eb1b</a>
</td>
<td>
khembach
</td>
<td>
2021-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## only first intron 
## zoom into intron 1 that contains the cryptic exon
region_start &lt;- 79611100
region_end &lt;- 79637000

plotTracks(c(alTrack_4wOFF, 
             alTrack_2wON, alTrack_4wONa, alTrack_4wONb,
             ceTrack, gtftrack),  
           type = c(&quot;coverage&quot;, &quot;sashimi&quot;), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100,
           fontsize=13,
           sizes = c(rep(3, 4), 1, 3),
           transcriptAnnotation = &quot;transcript&quot;)</code></pre>
<p><img src="figure/TDP-07-01-STMN2_expression.Rmd/low-high-TDPHA-cells-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-low-high-TDPHA-cells-2">
Past versions of low-high-TDPHA-cells-2.png
</button>
</p>
<div id="fig-low-high-TDPHA-cells-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/a02eb1b19e76fcfb3648951cda280bde1453103f/docs/figure/TDP-07-01-STMN2_expression.Rmd/low-high-TDPHA-cells-2.png" target="_blank">a02eb1b</a>
</td>
<td>
khembach
</td>
<td>
2021-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## HIGH ##
region_start &lt;- 79611100
region_end &lt;- 79666200

# Create the alignments track
alTrack_4wOFF &lt;- AlignmentsTrack(
  range = bams[[&quot;TDP4wOFF&quot;]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = high$barcode[high$sample_id == &quot;TDP4wOFF&quot;], 
  importFunction = .import.bam.alignments.cells, name = &quot;4wOFF&quot;)
alTrack_2wON &lt;- AlignmentsTrack(
  range = bams[[&quot;TDP2wON&quot;]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = high$barcode[high$sample_id == &quot;TDP2wON&quot;], 
  importFunction = .import.bam.alignments.cells, name = &quot;2wON&quot;)
alTrack_4wONa &lt;- AlignmentsTrack(
  range = bams[[&quot;TDP4wONa&quot;]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = high$barcode[high$sample_id == &quot;TDP4wONa&quot;], 
  importFunction = .import.bam.alignments.cells, name = &quot;4wONa&quot;)
alTrack_4wONb &lt;- AlignmentsTrack(
  range = bams[[&quot;TDP4wONb&quot;]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = high$barcode[high$sample_id == &quot;TDP4wONb&quot;], 
  importFunction = .import.bam.alignments.cells, name = &quot;4wONb&quot;)

plotTracks(c(alTrack_4wOFF, alTrack_2wON, alTrack_4wONa, alTrack_4wONb,
             ceTrack, gtftrack),  
           type = c(&quot;coverage&quot;, &quot;sashimi&quot;), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100, fontsize=13,
           sizes = c(rep(3, 4), 1, 2),
           transcriptAnnotation = &quot;transcript&quot;)</code></pre>
<p><img src="figure/TDP-07-01-STMN2_expression.Rmd/low-high-TDPHA-cells-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-low-high-TDPHA-cells-3">
Past versions of low-high-TDPHA-cells-3.png
</button>
</p>
<div id="fig-low-high-TDPHA-cells-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/a02eb1b19e76fcfb3648951cda280bde1453103f/docs/figure/TDP-07-01-STMN2_expression.Rmd/low-high-TDPHA-cells-3.png" target="_blank">a02eb1b</a>
</td>
<td>
khembach
</td>
<td>
2021-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>## only first intron 
## zoom into intron 1 that contains the cryptic exon
region_start &lt;- 79611100
region_end &lt;- 79637000

plotTracks(c(alTrack_4wOFF, 
             alTrack_2wON, alTrack_4wONa, alTrack_4wONb,
             ceTrack, gtftrack),  
           type = c(&quot;coverage&quot;, &quot;sashimi&quot;), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100, fontsize=13,
           sizes = c(rep(3, 4), 1, 3),
           transcriptAnnotation = &quot;transcript&quot;)</code></pre>
<p><img src="figure/TDP-07-01-STMN2_expression.Rmd/low-high-TDPHA-cells-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-low-high-TDPHA-cells-4">
Past versions of low-high-TDPHA-cells-4.png
</button>
</p>
<div id="fig-low-high-TDPHA-cells-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/a02eb1b19e76fcfb3648951cda280bde1453103f/docs/figure/TDP-07-01-STMN2_expression.Rmd/low-high-TDPHA-cells-4.png" target="_blank">a02eb1b</a>
</td>
<td>
khembach
</td>
<td>
2021-02-25
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.6 LTS

Matrix products: default
BLAS:   /usr/local/R/R-4.0.0/lib/libRblas.so
LAPACK: /usr/local/R/R-4.0.0/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] rtracklayer_1.48.0                      
 [2] GenomicAlignments_1.24.0                
 [3] Rsamtools_2.4.0                         
 [4] Biostrings_2.56.0                       
 [5] XVector_0.28.0                          
 [6] TxDb.Hsapiens.UCSC.hg38.knownGene_3.10.0
 [7] GenomicFeatures_1.40.0                  
 [8] AnnotationDbi_1.50.1                    
 [9] Gviz_1.32.0                             
[10] dplyr_1.0.2                             
[11] SingleCellExperiment_1.10.1             
[12] SummarizedExperiment_1.18.1             
[13] DelayedArray_0.14.0                     
[14] matrixStats_0.56.0                      
[15] Biobase_2.48.0                          
[16] GenomicRanges_1.40.0                    
[17] GenomeInfoDb_1.24.2                     
[18] IRanges_2.22.2                          
[19] S4Vectors_0.26.1                        
[20] BiocGenerics_0.34.0                     
[21] Seurat_3.1.5                            
[22] workflowr_1.6.2                         

loaded via a namespace (and not attached):
  [1] backports_1.1.9          Hmisc_4.4-1              BiocFileCache_1.12.0    
  [4] plyr_1.8.6               igraph_1.2.5             lazyeval_0.2.2          
  [7] splines_4.0.0            BiocParallel_1.22.0      listenv_0.8.0           
 [10] ggplot2_3.3.2            digest_0.6.25            ensembldb_2.12.1        
 [13] htmltools_0.5.0          checkmate_2.0.0          magrittr_1.5            
 [16] memoise_1.1.0            BSgenome_1.56.0          cluster_2.1.0           
 [19] ROCR_1.0-11              globals_0.12.5           askpass_1.1             
 [22] prettyunits_1.1.1        jpeg_0.1-8.1             colorspace_1.4-1        
 [25] blob_1.2.1               rappdirs_0.3.1           ggrepel_0.8.2           
 [28] xfun_0.15                crayon_1.3.4             RCurl_1.98-1.2          
 [31] jsonlite_1.7.0           VariantAnnotation_1.34.0 survival_3.2-3          
 [34] zoo_1.8-8                ape_5.4                  glue_1.4.2              
 [37] gtable_0.3.0             zlibbioc_1.34.0          leiden_0.3.3            
 [40] future.apply_1.6.0       scales_1.1.1             DBI_1.1.0               
 [43] Rcpp_1.0.5               htmlTable_2.0.1          viridisLite_0.3.0       
 [46] progress_1.2.2           reticulate_1.16          foreign_0.8-80          
 [49] bit_1.1-15.2             rsvd_1.0.3               Formula_1.2-3           
 [52] tsne_0.1-3               htmlwidgets_1.5.1        httr_1.4.1              
 [55] RColorBrewer_1.1-2       ellipsis_0.3.1           ica_1.0-2               
 [58] pkgconfig_2.0.3          XML_3.99-0.4             nnet_7.3-14             
 [61] uwot_0.1.8               dbplyr_1.4.4             tidyselect_1.1.0        
 [64] rlang_0.4.7              reshape2_1.4.4           later_1.1.0.1           
 [67] munsell_0.5.0            tools_4.0.0              generics_0.0.2          
 [70] RSQLite_2.2.0            ggridges_0.5.2           evaluate_0.14           
 [73] stringr_1.4.0            yaml_2.2.1               knitr_1.29              
 [76] bit64_0.9-7              fs_1.4.2                 fitdistrplus_1.1-1      
 [79] purrr_0.3.4              RANN_2.6.1               AnnotationFilter_1.12.0 
 [82] pbapply_1.4-2            future_1.17.0            nlme_3.1-148            
 [85] whisker_0.4              biomaRt_2.44.1           rstudioapi_0.11         
 [88] compiler_4.0.0           plotly_4.9.2.1           curl_4.3                
 [91] png_0.1-7                tibble_3.0.3             stringi_1.4.6           
 [94] lattice_0.20-41          ProtGenerics_1.20.0      Matrix_1.2-18           
 [97] vctrs_0.3.4              pillar_1.4.6             lifecycle_0.2.0         
[100] lmtest_0.9-37            RcppAnnoy_0.0.16         data.table_1.12.8       
[103] cowplot_1.0.0            bitops_1.0-6             irlba_2.3.3             
[106] httpuv_1.5.4             patchwork_1.0.1          R6_2.4.1                
[109] latticeExtra_0.6-29      promises_1.1.1           KernSmooth_2.23-17      
[112] gridExtra_2.3            codetools_0.2-16         dichromat_2.0-0         
[115] MASS_7.3-51.6            assertthat_0.2.1         openssl_1.4.2           
[118] rprojroot_1.3-2          sctransform_0.2.1        GenomeInfoDbData_1.2.3  
[121] hms_0.5.3                rpart_4.1-15             tidyr_1.1.0             
[124] rmarkdown_2.3            Rtsne_0.15               biovizBase_1.36.0       
[127] git2r_0.27.1             base64enc_0.1-3         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
