<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Katharina Hembach" />


<title>Gene expression changes in TDP-HA cultures</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">neural_scRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/khembach/neural_scRNAseq">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Gene expression changes in TDP-HA cultures</h1>
<h4 class="author">Katharina Hembach</h4>
<h4 class="date">2/25/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-02-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>neural_scRNAseq/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200522code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200522)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200522code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200522)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtreed8a5aa41f689c117c0c689d9f98287d0d5d1a2fetargetblankd8a5aa4a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/khembach/neural_scRNAseq/tree/d8a5aa41f689c117c0c689d9f98287d0d5d1a2fe" target="_blank">d8a5aa4</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomkhembachneuralscRNAseqtreed8a5aa41f689c117c0c689d9f98287d0d5d1a2fetargetblankd8a5aa4a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/khembach/neural_scRNAseq/tree/d8a5aa41f689c117c0c689d9f98287d0d5d1a2fe" target="_blank">d8a5aa4</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    ._.DS_Store
    Ignored:    ._Rplots.pdf
    Ignored:    ._Unfiltered.pdf
    Ignored:    .__workflowr.yml
    Ignored:    ._coverage.pdf
    Ignored:    ._coverage_sashimi.pdf
    Ignored:    ._coverage_sashimi.png
    Ignored:    ._neural_scRNAseq.Rproj
    Ignored:    ._sashimi.pdf
    Ignored:    ._stmn2.pdf
    Ignored:    ._tdp.pdf
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/._.DS_Store
    Ignored:    analysis/._01-preprocessing.Rmd
    Ignored:    analysis/._01-preprocessing.html
    Ignored:    analysis/._02.1-SampleQC.Rmd
    Ignored:    analysis/._03-filtering.Rmd
    Ignored:    analysis/._04-clustering.Rmd
    Ignored:    analysis/._04-clustering.knit.md
    Ignored:    analysis/._04.1-cell_cycle.Rmd
    Ignored:    analysis/._05-annotation.Rmd
    Ignored:    analysis/._Lam-0-NSC_no_integration.Rmd
    Ignored:    analysis/._Lam-01-NSC_integration.Rmd
    Ignored:    analysis/._Lam-02-NSC_annotation.Rmd
    Ignored:    analysis/._NSC-1-clustering.Rmd
    Ignored:    analysis/._NSC-2-annotation.Rmd
    Ignored:    analysis/.__site.yml
    Ignored:    analysis/._additional_filtering.Rmd
    Ignored:    analysis/._additional_filtering_clustering.Rmd
    Ignored:    analysis/._index.Rmd
    Ignored:    analysis/._organoid-01-1-qualtiy-control.Rmd
    Ignored:    analysis/._organoid-01-clustering.Rmd
    Ignored:    analysis/._organoid-02-integration.Rmd
    Ignored:    analysis/._organoid-03-cluster_analysis.Rmd
    Ignored:    analysis/._organoid-04-group_integration.Rmd
    Ignored:    analysis/._organoid-04-stage_integration.Rmd
    Ignored:    analysis/._organoid-05-group_integration_cluster_analysis.Rmd
    Ignored:    analysis/._organoid-05-stage_integration_cluster_analysis.Rmd
    Ignored:    analysis/._organoid-06-1-prepare-sce.Rmd
    Ignored:    analysis/._organoid-06-conos-analysis-Seurat.Rmd
    Ignored:    analysis/._organoid-06-conos-analysis-function.Rmd
    Ignored:    analysis/._organoid-06-conos-analysis.Rmd
    Ignored:    analysis/._organoid-06-group-integration-conos-analysis.Rmd
    Ignored:    analysis/._organoid-07-conos-visualization.Rmd
    Ignored:    analysis/._organoid-07-group-integration-conos-visualization.Rmd
    Ignored:    analysis/._organoid-08-conos-comparison.Rmd
    Ignored:    analysis/._organoid-0x-sample_integration.Rmd
    Ignored:    analysis/01-preprocessing_cache/
    Ignored:    analysis/02-1-SampleQC_cache/
    Ignored:    analysis/02-quality_control_cache/
    Ignored:    analysis/02.1-SampleQC_cache/
    Ignored:    analysis/03-filtering_cache/
    Ignored:    analysis/04-clustering_cache/
    Ignored:    analysis/04.1-cell_cycle_cache/
    Ignored:    analysis/05-annotation_cache/
    Ignored:    analysis/06-clustering-all-timepoints_cache/
    Ignored:    analysis/07-cluster-analysis-all-timepoints_cache/
    Ignored:    analysis/Lam-01-NSC_integration_cache/
    Ignored:    analysis/Lam-02-NSC_annotation_cache/
    Ignored:    analysis/NSC-1-clustering_cache/
    Ignored:    analysis/NSC-2-annotation_cache/
    Ignored:    analysis/TDP-01-preprocessing_cache/
    Ignored:    analysis/TDP-02-quality_control_cache/
    Ignored:    analysis/TDP-03-filtering_cache/
    Ignored:    analysis/TDP-04-clustering_cache/
    Ignored:    analysis/TDP-05-00-filtering-plasmid-QC_cache/
    Ignored:    analysis/TDP-05-plasmid_expression_cache/
    Ignored:    analysis/TDP-06-cluster_analysis_cache/
    Ignored:    analysis/TDP-07-01-STMN2_expression_cache/
    Ignored:    analysis/TDP-07-cluster_12_cache/
    Ignored:    analysis/TDP-08-00-clustering-HA-D96_cache/
    Ignored:    analysis/TDP-08-01-HA-D96-expression-changes_cache/
    Ignored:    analysis/TDP-08-clustering-timeline-HA_cache/
    Ignored:    analysis/additional_filtering_cache/
    Ignored:    analysis/additional_filtering_clustering_cache/
    Ignored:    analysis/organoid-01-1-qualtiy-control_cache/
    Ignored:    analysis/organoid-01-clustering_cache/
    Ignored:    analysis/organoid-02-integration_cache/
    Ignored:    analysis/organoid-03-cluster_analysis_cache/
    Ignored:    analysis/organoid-04-group_integration_cache/
    Ignored:    analysis/organoid-04-stage_integration_cache/
    Ignored:    analysis/organoid-05-group_integration_cluster_analysis_cache/
    Ignored:    analysis/organoid-05-stage_integration_cluster_analysis_cache/
    Ignored:    analysis/organoid-06-conos-analysis_cache/
    Ignored:    analysis/organoid-06-conos-analysis_test_cache/
    Ignored:    analysis/organoid-06-group-integration-conos-analysis_cache/
    Ignored:    analysis/organoid-07-conos-visualization_cache/
    Ignored:    analysis/organoid-07-group-integration-conos-visualization_cache/
    Ignored:    analysis/organoid-08-conos-comparison_cache/
    Ignored:    analysis/organoid-0x-sample_integration_cache/
    Ignored:    analysis/sample5_QC_cache/
    Ignored:    analysis/timepoints-01-organoid-integration_cache/
    Ignored:    data/.DS_Store
    Ignored:    data/._.DS_Store
    Ignored:    data/._.smbdeleteAAA17ed8b4b
    Ignored:    data/._Lam_figure2_markers.R
    Ignored:    data/._Reactive_astrocytes_markers.xlsx
    Ignored:    data/._known_NSC_markers.R
    Ignored:    data/._known_cell_type_markers.R
    Ignored:    data/._metadata.csv
    Ignored:    data/._virus_cell_tropism_markers.R
    Ignored:    data/._~$Reactive_astrocytes_markers.xlsx
    Ignored:    data/data_sushi/
    Ignored:    data/filtered_feature_matrices/
    Ignored:    output/.DS_Store
    Ignored:    output/._.DS_Store
    Ignored:    output/._NSC_cluster2_marker_genes.txt
    Ignored:    output/._TDP-06-no_integration_cluster12_marker_genes.txt
    Ignored:    output/._TDP-06-no_integration_cluster13_marker_genes.txt
    Ignored:    output/._organoid_integration_cluster1_marker_genes.txt
    Ignored:    output/Lam-01-clustering.rds
    Ignored:    output/NSC_1_clustering.rds
    Ignored:    output/NSC_cluster1_marker_genes.txt
    Ignored:    output/NSC_cluster2_marker_genes.txt
    Ignored:    output/NSC_cluster3_marker_genes.txt
    Ignored:    output/NSC_cluster4_marker_genes.txt
    Ignored:    output/NSC_cluster5_marker_genes.txt
    Ignored:    output/NSC_cluster6_marker_genes.txt
    Ignored:    output/NSC_cluster7_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster0_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster10_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster11_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster12_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster13_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster14_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster15_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster16_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster17_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster1_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster2_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster3_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster4_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster5_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster6_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster7_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster8_marker_genes.txt
    Ignored:    output/TDP-06-no_integration_cluster9_marker_genes.txt
    Ignored:    output/TDP-06_scran_markers.rds
    Ignored:    output/additional_filtering.rds
    Ignored:    output/conos/
    Ignored:    output/conos_organoid-06-conos-analysis.rds
    Ignored:    output/conos_organoid-06-group-integration-conos-analysis.rds
    Ignored:    output/figures/
    Ignored:    output/organoid_integration_cluster10_marker_genes.txt
    Ignored:    output/organoid_integration_cluster11_marker_genes.txt
    Ignored:    output/organoid_integration_cluster12_marker_genes.txt
    Ignored:    output/organoid_integration_cluster13_marker_genes.txt
    Ignored:    output/organoid_integration_cluster14_marker_genes.txt
    Ignored:    output/organoid_integration_cluster15_marker_genes.txt
    Ignored:    output/organoid_integration_cluster16_marker_genes.txt
    Ignored:    output/organoid_integration_cluster17_marker_genes.txt
    Ignored:    output/organoid_integration_cluster1_marker_genes.txt
    Ignored:    output/organoid_integration_cluster2_marker_genes.txt
    Ignored:    output/organoid_integration_cluster3_marker_genes.txt
    Ignored:    output/organoid_integration_cluster4_marker_genes.txt
    Ignored:    output/organoid_integration_cluster5_marker_genes.txt
    Ignored:    output/organoid_integration_cluster6_marker_genes.txt
    Ignored:    output/organoid_integration_cluster7_marker_genes.txt
    Ignored:    output/organoid_integration_cluster8_marker_genes.txt
    Ignored:    output/organoid_integration_cluster9_marker_genes.txt
    Ignored:    output/sce_01_preprocessing.rds
    Ignored:    output/sce_02_quality_control.rds
    Ignored:    output/sce_03_filtering.rds
    Ignored:    output/sce_03_filtering_all_genes.rds
    Ignored:    output/sce_06-1-prepare-sce.rds
    Ignored:    output/sce_TDP_01_preprocessing.rds
    Ignored:    output/sce_TDP_02_quality_control.rds
    Ignored:    output/sce_TDP_03_filtering.rds
    Ignored:    output/sce_TDP_03_filtering_all_genes.rds
    Ignored:    output/sce_organoid-01-clustering.rds
    Ignored:    output/sce_preprocessing.rds
    Ignored:    output/so_04-group_integration.rds
    Ignored:    output/so_04-stage_integration.rds
    Ignored:    output/so_04_1_cell_cycle.rds
    Ignored:    output/so_04_clustering.rds
    Ignored:    output/so_06-clustering_all_timepoints.rds
    Ignored:    output/so_08-00_clustering_HA_D96.rds
    Ignored:    output/so_08-clustering_timeline_HA.rds
    Ignored:    output/so_0x-sample_integration.rds
    Ignored:    output/so_TDP-06-cluster-analysis.rds
    Ignored:    output/so_TDP_04_clustering.rds
    Ignored:    output/so_TDP_05_plasmid_expression.rds
    Ignored:    output/so_additional_filtering_clustering.rds
    Ignored:    output/so_integrated_organoid-02-integration.rds
    Ignored:    output/so_merged_organoid-02-integration.rds
    Ignored:    output/so_organoid-01-clustering.rds
    Ignored:    output/so_sample_organoid-01-clustering.rds
    Ignored:    scripts/.DS_Store
    Ignored:    scripts/._.DS_Store
    Ignored:    scripts/._bu_Rcode.R
    Ignored:    scripts/._plasmid_expression.sh

Untracked files:
    Untracked:  Filtered.pdf
    Untracked:  Rplots.pdf
    Untracked:  Unfiltered
    Untracked:  Unfiltered.pdf
    Untracked:  analysis/Lam-0-NSC_no_integration.Rmd
    Untracked:  analysis/TDP-07-01-STMN2_expression copy.Rmd
    Untracked:  analysis/additional_filtering.Rmd
    Untracked:  analysis/additional_filtering_clustering.Rmd
    Untracked:  analysis/organoid-01-1-qualtiy-control.Rmd
    Untracked:  analysis/organoid-06-conos-analysis-Seurat.Rmd
    Untracked:  analysis/organoid-06-conos-analysis-function.Rmd
    Untracked:  analysis/organoid-07-conos-visualization.Rmd
    Untracked:  analysis/organoid-07-group-integration-conos-visualization.Rmd
    Untracked:  analysis/organoid-08-conos-comparison.Rmd
    Untracked:  analysis/organoid-0x-sample_integration.Rmd
    Untracked:  analysis/sample5_QC.Rmd
    Untracked:  ce$id <- sce$sample_id
    Untracked:  coverage.pdf
    Untracked:  coverage_sashimi.pdf
    Untracked:  coverage_sashimi.png
    Untracked:  data/Homo_sapiens.GRCh38.98.sorted.gtf
    Untracked:  data/Kanton_et_al/
    Untracked:  data/Lam_et_al/
    Untracked:  data/Sep2020/
    Untracked:  data/reference/
    Untracked:  data/virus_cell_tropism_markers.R
    Untracked:  data/~$Reactive_astrocytes_markers.xlsx
    Untracked:  sashimi.pdf
    Untracked:  scripts/bu_Rcode.R
    Untracked:  scripts/salmon-latest_linux_x86_64/
    Untracked:  stmn2.pdf
    Untracked:  tdp.pdf

Unstaged changes:
    Modified:   analysis/05-annotation.Rmd
    Modified:   analysis/Lam-02-NSC_annotation.Rmd
    Modified:   analysis/TDP-04-clustering.Rmd
    Modified:   analysis/TDP-06-cluster_analysis.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/organoid-02-integration.Rmd
    Modified:   analysis/organoid-04-group_integration.Rmd
    Modified:   analysis/organoid-06-conos-analysis.Rmd
    Modified:   analysis/timepoints-01-organoid-integration.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/TDP-08-01-HA-D96-expression-changes.Rmd</code>) and HTML (<code>docs/TDP-08-01-HA-D96-expression-changes.html</code>) files. If you've configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/khembach/neural_scRNAseq/blob/d8a5aa41f689c117c0c689d9f98287d0d5d1a2fe/analysis/TDP-08-01-HA-D96-expression-changes.Rmd" target="_blank">d8a5aa4</a>
</td>
<td>
khembach
</td>
<td>
2021-02-25
</td>
<td>
differential analysis of effect of TDP-HA expression in culture
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>library(Seurat)
library(SingleCellExperiment)
library(dplyr)
library(scran)
library(Seurat)
library(ComplexHeatmap)
library(cowplot)
library(ggplot2)
library(muscat)
library(purrr)
library(RColorBrewer)
library(viridis)
library(edgeR)</code></pre>
</div>
<div id="load-data-convert-to-sce" class="section level1">
<h1>Load data &amp; convert to SCE</h1>
<pre class="r"><code>so &lt;- readRDS(file.path(&quot;output&quot;, &quot;so_08-00_clustering_HA_D96.rds&quot;))
sce &lt;- as.SingleCellExperiment(so, assay = &quot;RNA&quot;)
colData(sce) &lt;- as.data.frame(colData(sce)) %&gt;% 
    mutate_if(is.character, as.factor) %&gt;% 
    DataFrame(row.names = colnames(sce))</code></pre>
</div>
<div id="cluster-sample-counts" class="section level1">
<h1>Cluster-sample counts</h1>
<pre class="r"><code># set cluster IDs to resolution 0.4 clustering
so &lt;- SetIdent(so, value = &quot;RNA_snn_res.0.4&quot;)
so@meta.data$cluster_id &lt;- Idents(so)
sce$cluster_id &lt;- Idents(so)
(n_cells &lt;- table(sce$cluster_id, sce$sample_id))</code></pre>
<pre><code>    
     5NC96 6NC96 TDP2wON TDP4wOFF TDP4wONa TDP4wONb
  0    770  1028    1057      803     1586      934
  1    443   631    1157     1003     1580     1362
  2    306   525     916      816     1101     1118
  3    355   289     643      582     1000      802
  4    185   311     703      605      916      750
  5    219   307     694      502      718      763
  6    150   216     419      383      559      505
  7    221   293     582      413      479      241
  8    111   170     377      347      532      434
  9     53    94     279      179      387      303
  10   465   415     118      158       70       22
  11    59    95     223      187      320      258
  12    63    98      66       63      240      145
  13   134   118     117       25       91       53
  14     3     5      48       11       44       31
  15     1     0       7        0       42        1</code></pre>
</div>
<div id="relative-cluster-abundances" class="section level1">
<h1>Relative cluster-abundances</h1>
<p>How are the samples distributed across clusters?</p>
<pre class="r"><code>fqs &lt;- prop.table(n_cells, margin = 2)
mat &lt;- as.matrix(unclass(fqs))
Heatmap(mat,
    col = rev(magma(12))[-c(1,2)],
    name = &quot;Frequency&quot;,
    cluster_rows = FALSE,
    cluster_columns = FALSE,
    row_names_side = &quot;left&quot;,
    row_title = &quot;cluster_id&quot;,
    column_title = &quot;sample_id&quot;,
    column_title_side = &quot;bottom&quot;,
    rect_gp = gpar(col = &quot;white&quot;),
    cell_fun = function(i, j, x, y, width, height, fill)
        grid.text(round(mat[j, i] * 100, 2), x = x, y = y, 
            gp = gpar(col = &quot;white&quot;, fontsize = 8)))</code></pre>
<p><img src="figure/TDP-08-01-HA-D96-expression-changes.Rmd/unnamed-chunk-2-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="dr-colored-by-cluster-id" class="section level1 tabset">
<h1>DR colored by cluster ID</h1>
<pre class="r"><code>cs &lt;- sample(colnames(so), 1e4)
.plot_dr &lt;- function(so, dr, id)
    DimPlot(so, cells = cs, group.by = id, reduction = dr, pt.size = 0.4) +
        guides(col = guide_legend(nrow = 11, 
            override.aes = list(size = 3, alpha = 1))) +
        theme_void() + theme(aspect.ratio = 1)
ids &lt;- c(&quot;group_id&quot;, &quot;sample_id&quot;, &quot;ident&quot;)
for (id in ids) {
    cat(&quot;## &quot;, id, &quot;\n&quot;)
    p1 &lt;- .plot_dr(so, &quot;tsne&quot;, id)
    lgd &lt;- get_legend(p1)
    p1 &lt;- p1 + theme(legend.position = &quot;none&quot;)
    p2 &lt;- .plot_dr(so, &quot;umap&quot;, id) + theme(legend.position = &quot;none&quot;)
    ps &lt;- plot_grid(plotlist = list(p1, p2), nrow = 1)
    p &lt;- plot_grid(ps, lgd, nrow = 1, rel_widths = c(1, 0.2))
    print(p)
    cat(&quot;\n\n&quot;)
}</code></pre>
<div id="group_id" class="section level2">
<h2>group_id</h2>
<p><img src="figure/TDP-08-01-HA-D96-expression-changes.Rmd/dim-red-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="sample_id" class="section level2">
<h2>sample_id</h2>
<p><img src="figure/TDP-08-01-HA-D96-expression-changes.Rmd/dim-red-2.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="ident" class="section level2">
<h2>ident</h2>
<p><img src="figure/TDP-08-01-HA-D96-expression-changes.Rmd/dim-red-3.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="dr-colored-by-cluster-id-per-sample" class="section level2">
<h2>DR colored by cluster ID per sample</h2>
<p>To better see how the cells from different clusters overlap, we only plot the cells from one samples at a time.</p>
<pre class="r"><code>.plot_dr &lt;- function(so, dr, id, cs) {
    DimPlot(so, cells = cs, group.by = id, reduction = dr, pt.size = 0.4, cols = ) +
        guides(col = guide_legend(nrow = 11, 
            override.aes = list(size = 3, alpha = 1))) +
        theme_void() + theme(aspect.ratio = 1) +
        theme(plot.title = element_text(hjust = 0.5))}
# ids &lt;- unique(so$sample_id)

p1 &lt;- .plot_dr(so, &quot;umap&quot;, &quot;ident&quot;, colnames(so[,so$sample_id == &quot;5NC96&quot;])) + 
  theme(legend.position = &quot;none&quot;) + ggtitle(&quot;5NC96&quot;)
p2 &lt;- .plot_dr(so, &quot;umap&quot;, &quot;ident&quot;, colnames(so[,so$sample_id == &quot;6NC96&quot;])) + 
  theme(legend.position = &quot;none&quot;) + ggtitle(&quot;6NC96&quot;)
p3 &lt;- .plot_dr(so, &quot;umap&quot;, &quot;ident&quot;, colnames(so[,so$sample_id == &quot;TDP4wOFF&quot;])) + 
  theme(legend.position = &quot;none&quot;) + ggtitle(&quot;TDP4wOFF&quot;)
p4 &lt;- .plot_dr(so, &quot;umap&quot;, &quot;ident&quot;, colnames(so[,so$sample_id == &quot;TDP2wON&quot;])) + 
  theme(legend.position = &quot;none&quot;) + ggtitle(&quot;TDP2wON&quot;)
p5 &lt;- .plot_dr(so, &quot;umap&quot;, &quot;ident&quot;, colnames(so[,so$sample_id == &quot;TDP4wONa&quot;])) + 
  theme(legend.position = &quot;none&quot;) + ggtitle(&quot;TDP4wONa&quot;)
p6 &lt;- .plot_dr(so, &quot;umap&quot;, &quot;ident&quot;, colnames(so[,so$sample_id == &quot;TDP4wONb&quot;])) + 
  theme(legend.position = &quot;none&quot;) + ggtitle(&quot;TDP4wONb&quot;)
ps &lt;- plot_grid(plotlist = list(p1, p2, p3, p4, p5, p6), nrow = 2)
lgd &lt;- get_legend(p5)
p &lt;- plot_grid(ps, lgd, nrow = 1, rel_widths = c(1, 0.2))
p</code></pre>
<p><img src="figure/TDP-08-01-HA-D96-expression-changes.Rmd/UMAP-split-by-sample-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="de-analysis" class="section level1">
<h1>DE analysis</h1>
<p>We want to test for differences in gene expression between cells from D96 samples and the TDP-HA expressing samples. We test for differences in the neuronal cells (right cell cloud in UMAP).</p>
<div id="preprocessing-with-muscat" class="section level2">
<h2>Preprocessing with muscat</h2>
<p>We filter and only keep cells from D96 and 4wON.</p>
<pre class="r"><code>## TODO subset cells

sce$id &lt;- sce$sample_id
levels(sce$id) &lt;- c(&quot;D96&quot;, &quot;D96&quot;, &quot;ON2w&quot;, &quot;OFF4w&quot;, &quot;ON4w&quot;, &quot;ON4w&quot;)
(sce &lt;- prepSCE(sce, 
    kid = &quot;cluster_id&quot;, # subpopulation assignments
    gid = &quot;id&quot;,  # group IDs (ctrl/stim)
    sid = &quot;sample_id&quot;,   # sample IDs (ctrl/stim.1234)
    drop = FALSE))  # drop all other colData columns</code></pre>
<pre><code>class: SingleCellExperiment 
dim: 14073 39003 
metadata(1): experiment_info
assays(2): counts logcounts
rownames(14073): ENSG00000187634.SAMD11 ENSG00000188976.NOC2L ...
  ENSG00000227234.SPANXB1 ENSG00000198573.SPANXC
rowData names(5): vst.mean vst.variance vst.variance.expected
  vst.variance.standardized vst.variable
colnames(39003): tdp_ha_AAACCCACATGTCTAG-1.TDP2wON
  tdp_ha_AAACCCATCACGTAGT-1.TDP2wON ... D96_TTTGTTGCACTCGATA-1.6NC96
  D96_TTTGTTGTCGTGTGAT-1.6NC96
colData names(44): cluster_id sample_id ... integrated_snn_res.2 ident
reducedDimNames(3): PCA TSNE UMAP
altExpNames(0):</code></pre>
<pre class="r"><code>nk &lt;- length(kids &lt;- levels(sce$cluster_id))
ns &lt;- length(sids &lt;- levels(sce$sample_id))
names(kids) &lt;- kids; names(sids) &lt;- sids

# nb. of cells per cluster-sample
t(table(sce$cluster_id, sce$sample_id))</code></pre>
<pre><code>          
              0    1    2    3    4    5    6    7    8    9   10   11   12
  5NC96     770  443  306  355  185  219  150  221  111   53  465   59   63
  6NC96    1028  631  525  289  311  307  216  293  170   94  415   95   98
  TDP2wON  1057 1157  916  643  703  694  419  582  377  279  118  223   66
  TDP4wOFF  803 1003  816  582  605  502  383  413  347  179  158  187   63
  TDP4wONa 1586 1580 1101 1000  916  718  559  479  532  387   70  320  240
  TDP4wONb  934 1362 1118  802  750  763  505  241  434  303   22  258  145
          
             13   14   15
  5NC96     134    3    1
  6NC96     118    5    0
  TDP2wON   117   48    7
  TDP4wOFF   25   11    0
  TDP4wONa   91   44   42
  TDP4wONb   53   31    1</code></pre>
</div>
<div id="compute-pseudobulk-counts" class="section level2">
<h2>Compute pseudobulk counts</h2>
<p>We sum the gene counts per cluster</p>
<pre class="r"><code>pb &lt;- aggregateData(sce, assay = &quot;counts&quot;, by = c(&quot;cluster_id&quot;, &quot;sample_id&quot;), 
                    fun = &quot;sum&quot;)

# one sheet per subpopulation = cluster
assayNames(pb)</code></pre>
<pre><code> [1] &quot;0&quot;  &quot;1&quot;  &quot;2&quot;  &quot;3&quot;  &quot;4&quot;  &quot;5&quot;  &quot;6&quot;  &quot;7&quot;  &quot;8&quot;  &quot;9&quot;  &quot;10&quot; &quot;11&quot; &quot;12&quot; &quot;13&quot; &quot;14&quot;
[16] &quot;15&quot;</code></pre>
<pre class="r"><code># pseudobulks for 1st cluster
t(head(assay(pb)))</code></pre>
<pre><code>         ENSG00000187634.SAMD11 ENSG00000188976.NOC2L ENSG00000187961.KLHL17
5NC96                       125                   286                     79
6NC96                       179                   400                    106
TDP2wON                      85                   475                    104
TDP4wOFF                     80                   328                     77
TDP4wONa                    105                   416                    109
TDP4wONb                     84                   362                     78
         ENSG00000188290.HES4 ENSG00000187608.ISG15 ENSG00000188157.AGRN
5NC96                    3784                   641                  985
6NC96                    6195                   897                 1442
TDP2wON                  8591                  2656                 1345
TDP4wOFF                 5736                   860                  999
TDP4wONa                12971                  2232                 1804
TDP4wONb                 8302                  1522                 1253</code></pre>
</div>
<div id="pseudobulk-mds-plot" class="section level2">
<h2>Pseudobulk MDS plot</h2>
<pre class="r"><code>(pb_mds &lt;- pbMDS(pb))</code></pre>
<p><img src="figure/TDP-08-01-HA-D96-expression-changes.Rmd/MDS-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="pseudobulk-differential-state-analysis" class="section level2">
<h2>Pseudobulk differential state analysis</h2>
<pre class="r"><code># # run DS analysis
# res &lt;- pbDS(pb, verbose = FALSE)
# # access results table for 1st comparison
# tbl &lt;- res$table[[1]]
# # one data.frame per cluster
# names(tbl)
# 
# # view results for 1st cluster
# k1 &lt;- tbl[[1]]
# head(format(k1[, -ncol(k1)], digits = 2))</code></pre>
</div>
<div id="prepare-dgelist-for-edger" class="section level2">
<h2>Prepare DGEList for edgeR</h2>
<pre class="r"><code># Creating up a DGEList object for use in edgeR:

# y &lt;- DGEList(counts(current), samples=colData(current))
# y</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.0 (2020-04-24)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 16.04.6 LTS

Matrix products: default
BLAS:   /usr/local/R/R-4.0.0/lib/libRblas.so
LAPACK: /usr/local/R/R-4.0.0/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] edgeR_3.30.3                limma_3.44.3               
 [3] viridis_0.5.1               viridisLite_0.3.0          
 [5] RColorBrewer_1.1-2          purrr_0.3.4                
 [7] muscat_1.2.1                ggplot2_3.3.2              
 [9] cowplot_1.0.0               ComplexHeatmap_2.4.2       
[11] scran_1.16.0                dplyr_1.0.2                
[13] SingleCellExperiment_1.10.1 SummarizedExperiment_1.18.1
[15] DelayedArray_0.14.0         matrixStats_0.56.0         
[17] Biobase_2.48.0              GenomicRanges_1.40.0       
[19] GenomeInfoDb_1.24.2         IRanges_2.22.2             
[21] S4Vectors_0.26.1            BiocGenerics_0.34.0        
[23] Seurat_3.1.5                workflowr_1.6.2            

loaded via a namespace (and not attached):
  [1] reticulate_1.16           tidyselect_1.1.0         
  [3] lme4_1.1-23               RSQLite_2.2.0            
  [5] AnnotationDbi_1.50.1      htmlwidgets_1.5.1        
  [7] BiocParallel_1.22.0       Rtsne_0.15               
  [9] munsell_0.5.0             codetools_0.2-16         
 [11] ica_1.0-2                 statmod_1.4.34           
 [13] future_1.17.0             withr_2.2.0              
 [15] colorspace_1.4-1          knitr_1.29               
 [17] ROCR_1.0-11               listenv_0.8.0            
 [19] labeling_0.3              git2r_0.27.1             
 [21] GenomeInfoDbData_1.2.3    farver_2.0.3             
 [23] bit64_0.9-7               glmmTMB_1.0.2.1          
 [25] rprojroot_1.3-2           vctrs_0.3.4              
 [27] generics_0.0.2            xfun_0.15                
 [29] R6_2.4.1                  doParallel_1.0.15        
 [31] ggbeeswarm_0.6.0          clue_0.3-57              
 [33] rsvd_1.0.3                locfit_1.5-9.4           
 [35] bitops_1.0-6              promises_1.1.1           
 [37] scales_1.1.1              beeswarm_0.2.3           
 [39] gtable_0.3.0              globals_0.12.5           
 [41] rlang_0.4.7               genefilter_1.70.0        
 [43] GlobalOptions_0.1.2       splines_4.0.0            
 [45] TMB_1.7.16                lazyeval_0.2.2           
 [47] yaml_2.2.1                reshape2_1.4.4           
 [49] backports_1.1.9           httpuv_1.5.4             
 [51] tools_4.0.0               ellipsis_0.3.1           
 [53] gplots_3.0.4              ggridges_0.5.2           
 [55] Rcpp_1.0.5                plyr_1.8.6               
 [57] progress_1.2.2            zlibbioc_1.34.0          
 [59] RCurl_1.98-1.2            prettyunits_1.1.1        
 [61] pbapply_1.4-2             GetoptLong_1.0.1         
 [63] zoo_1.8-8                 ggrepel_0.8.2            
 [65] cluster_2.1.0             colorRamps_2.3           
 [67] fs_1.4.2                  variancePartition_1.18.2 
 [69] magrittr_1.5              data.table_1.12.8        
 [71] lmerTest_3.1-2            circlize_0.4.10          
 [73] lmtest_0.9-37             RANN_2.6.1               
 [75] whisker_0.4               fitdistrplus_1.1-1       
 [77] hms_0.5.3                 patchwork_1.0.1          
 [79] evaluate_0.14             xtable_1.8-4             
 [81] pbkrtest_0.4-8.6          XML_3.99-0.4             
 [83] gridExtra_2.3             shape_1.4.4              
 [85] compiler_4.0.0            scater_1.16.2            
 [87] tibble_3.0.3              KernSmooth_2.23-17       
 [89] crayon_1.3.4              minqa_1.2.4              
 [91] htmltools_0.5.0           later_1.1.0.1            
 [93] tidyr_1.1.0               geneplotter_1.66.0       
 [95] DBI_1.1.0                 MASS_7.3-51.6            
 [97] rappdirs_0.3.1            boot_1.3-25              
 [99] Matrix_1.2-18             gdata_2.18.0             
[101] igraph_1.2.5              pkgconfig_2.0.3          
[103] numDeriv_2016.8-1.1       plotly_4.9.2.1           
[105] foreach_1.5.0             annotate_1.66.0          
[107] vipor_0.4.5               dqrng_0.2.1              
[109] blme_1.0-4                XVector_0.28.0           
[111] stringr_1.4.0             digest_0.6.25            
[113] sctransform_0.2.1         RcppAnnoy_0.0.16         
[115] tsne_0.1-3                rmarkdown_2.3            
[117] leiden_0.3.3              uwot_0.1.8               
[119] DelayedMatrixStats_1.10.1 gtools_3.8.2             
[121] rjson_0.2.20              nloptr_1.2.2.2           
[123] lifecycle_0.2.0           nlme_3.1-148             
[125] jsonlite_1.7.0            BiocNeighbors_1.6.0      
[127] pillar_1.4.6              lattice_0.20-41          
[129] httr_1.4.1                survival_3.2-3           
[131] glue_1.4.2                png_0.1-7                
[133] iterators_1.0.12          bit_1.1-15.2             
[135] stringi_1.4.6             blob_1.2.1               
[137] DESeq2_1.28.1             BiocSingular_1.4.0       
[139] caTools_1.18.0            memoise_1.1.0            
[141] irlba_2.3.3               future.apply_1.6.0       
[143] ape_5.4                  </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
