---
title: "Stathmin 2 expression"
author: "Katharina Hembach"
date: "21/01/2021"
output: 
  html_document:
    toc: true,
    code_folding: show
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, autodep = TRUE, cache = TRUE, dev = "png",
                      dev.args = list(png = list(type = "cairo")), 
                      message = FALSE, cache.lazy = FALSE)
```

### Load packages

```{r load-libs, message = FALSE, warning = FALSE}
library(Seurat)
library(SingleCellExperiment)
library(dplyr)
library(Gviz)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(Rsamtools)
library(GenomicAlignments)
```

# Load data & convert to SCE

```{r load-data}
# so <- readRDS(file.path("output", "so_TDP-06-cluster-analysis.rds"))
so <- readRDS(file.path("output", "so_TDP_05_plasmid_expression.rds"))
so <- SetIdent(so, value = "RNA_snn_res.0.4")
so@meta.data$cluster_id <- Idents(so)
```

# Cells from cluster 12

We want to compare the stathmin2 read coverage of cells expressing TDP-HA (from cluster 12) and other neuronal cells without TDP-HA expression. For this, we randomly select 5 cells from each group and filter the corresponding stathmin2 reads from the BAM file.

```{r select-cell-IDs}
clus12 <- subset(so, subset = cluster_id == "12")
## from which sample do the cells come from?
clus12$sample_id %>% table
## what is the range of TDP-HA expression in all cells in cluster 12?
dat_ha <- GetAssayData(object = clus12, slot = "data")["TDP43-HA",]
summary(dat_ha)
## select cells with high TDP-HA expression
high <- clus12$barcode[which(dat_ha > 3.5)]
high
## cells with low TDP-HA expression
low <- clus12$barcode[which(dat_ha < 0.5 & dat_ha > 0)]
low 
```

# Get stathmin2 reads of selected cells

We extract the reads covering the stathmin2 genes of the selected cells.

```{r read-coverage}
bams <- list(TDP4wOFF = file.path("data", "Sep2020", "CellRangerCount_50076_2020-09-22--15-40-54",
            "no1_Neural_cuture_d_96_TDP-43-HA_4w_DOXoff",
            "possorted_genome_bam.bam"),
            TDP2wON = file.path("data", "Sep2020", "CellRangerCount_50076_2020-09-22--15-40-54",
            "no2_Neural_cuture_d_96_TDP-43-HA_2w_DOXON",
            "possorted_genome_bam.bam"),
            TDP4wONa = file.path("data", "Sep2020", "CellRangerCount_50076_2020-09-22--15-40-54",
            "no3_Neural_cuture_d_96_TDP-43-HA_4w_DOXONa",
            "possorted_genome_bam.bam"),
            TDP4wONb = file.path("data", "Sep2020", "CellRangerCount_50076_2020-09-22--15-40-54",
            "no4_Neural_cuture_d_96_TDP-43-HA_4w_DOXONb",
            "possorted_genome_bam.bam"))

chr <- "chr8"
region_start <- 79611100
region_end <- 79666200
stmn2 <- GRanges(chr, IRanges(region_start, region_end), "+")


# keep all reads from cells in cluster 12
param <- ScanBamParam(which=stmn2, what = c("qname"), tag = "CB", 
                      tagFilter = list(CB = clus12$barcode))
gals <- lapply(bams, function(x) {
  readGAlignments(x, use.names = TRUE, param=param)
  })

covs <- lapply(gals, coverage)
```


# Ploting with Gviz

Plot the stathmin2 transcripts and the read coverage of all cells from cluster 12.
```{r stmn2-clus12-coverage, fig.width = 12}
## gene annotations from UCSC
options(ucscChromosomeNames = FALSE)
eTrack <- GeneRegionTrack(TxDb.Hsapiens.UCSC.hg38.knownGene, 
                          chromosome = chr, start = region_start, 
                          end = region_end, name = "annotation")

# covs[[1]]$chr8[region_start:region_end]
coords <- 79611100:79666201
dat <- matrix(c(as.vector(covs[[1]]$chr8[region_start:region_end]),
                as.vector(covs[[2]]$chr8[region_start:region_end]),
                as.vector(covs[[3]]$chr8[region_start:region_end]),
                as.vector(covs[[4]]$chr8[region_start:region_end])), 
              nrow = 4, byrow = TRUE)
rownames(dat) <- names(covs)
dtrack <- DataTrack(data = dat,
                    start = coords[-length(coords)], end = coords[-1], chromosome = chr,
                    genome = "hg38")

plotTracks(c(dtrack, eTrack),  
           # collapseTranscripts="meta", 
           type = "histogram", showSampleNames = TRUE,
           # chromosome = chr, from = region_start, to = region_end,
           shape = "arrow", geneSymbols = TRUE, aggregateGroups=FALSE,
           groups = c("TDP4wOFF", "TDP2wON", "TDP4wONa", "TDP4wONb"),
           stackedBars = FALSE, cex.legend = 4
           )

## one data track per sample
dats <- list("4wOFF" = matrix(as.vector(covs[[1]]$chr8[region_start:region_end]),
                              nrow = 1, byrow = TRUE),
             "2wON" = matrix(as.vector(covs[[2]]$chr8[region_start:region_end]),
                              nrow = 1, byrow = TRUE),
             "4wONa" = matrix(as.vector(covs[[3]]$chr8[region_start:region_end]),
                              nrow = 1, byrow = TRUE),
             "4wONb" = matrix(as.vector(covs[[4]]$chr8[region_start:region_end]),
                              nrow = 1, byrow = TRUE))
# dats <- lapply(seq_along(dats), function(x) {rownames(dats[[x]]) <- names(dats)[x]; dats[[x]]})

# rownames(dat) <- names(covs)
dtrack_4wOFF <- DataTrack(data = dats[[1]],
                    start = coords[-length(coords)], end = coords[-1], chromosome = chr,
                    genome = "hg38", name = "4wOFF")
dtrack_2wON <- DataTrack(data = dats[[2]],
                    start = coords[-length(coords)], end = coords[-1], chromosome = chr,
                    genome = "hg38", name = "2wON")
dtrack_4wONa <- DataTrack(data = dats[[3]],
                    start = coords[-length(coords)], end = coords[-1], chromosome = chr,
                    genome = "hg38", name = "4wONa")
dtrack_4wONb <- DataTrack(data = dats[[4]],
                    start = coords[-length(coords)], end = coords[-1], chromosome = chr,
                    genome = "hg38", name = "4wONb")

plotTracks(c(dtrack_4wOFF, dtrack_2wON, dtrack_4wONa, dtrack_4wONb, eTrack),  
           # collapseTranscripts="meta", 
           type = "histogram", showSampleNames = TRUE,
           # chromosome = chr, from = region_start, to = region_end,
           shape = "arrow", geneSymbols = TRUE, aggregateGroups=FALSE,
           # groups = c("TDP4wOFF", "TDP2wON", "TDP4wONa", "TDP4wONb"),
           stackedBars = FALSE, cex.legend = 4
           )

## zoom into intron 1 that contains the cryptic exon
chr <- "chr8"
region_start <- 79611100
region_end <- 79637000

plotTracks(c(dtrack_4wOFF, dtrack_2wON, dtrack_4wONa, dtrack_4wONb, eTrack),  
           # collapseTranscripts="meta", 
           type = "histogram", showSampleNames = TRUE,
           chromosome = chr, from = region_start, to = region_end,
           shape = "arrow", geneSymbols = TRUE, aggregateGroups=FALSE,
           # groups = c("TDP4wOFF", "TDP2wON", "TDP4wONa", "TDP4wONb"),
           stackedBars = FALSE, cex.legend = 4
           )
```




```{r get-reads, eval = FALSE, include = FALSE}

lim <- c(26700000, 26750000)
coords <- sort(c(lim[1], 
                 sample(seq(from = lim[1], to = lim[2]), 99), 
                 lim[2]))
dat <- runif(100, min = -10, max = 10)
dtrack <- DataTrack(data = dat, start = coords[-length(coords)],
                    end = coords[-1], chromosome = chr, genome = gen, 
                    name = "Uniform")
plotTracks(list(itrack, gtrack, atrack, grtrack, dtrack), 
           from = lim[1], to = lim[2])




## we need the read coverage in the stathmin2 genomic region for plotting
alTrack <- AlignmentsTrack(
  file.path("data", "Sep2020", "CellRangerCount_50076_2020-09-22--15-40-54",
            "no3_Neural_cuture_d_96_TDP-43-HA_4w_DOXONa",
            "possorted_genome_bam.bam"),
  chromosome = chr, from = region_start, to = region_end,
  isPaired = FALSE)

pdf()
plotTracks(list(alTrack, eTrack),  
           # collapseTranscripts="meta", 
           type = "coverage", 
           chromosome = chr, from = region_start, to = region_end,
           coverageHeight = 0.5, shape = "arrow", geneSymbols = TRUE)
dev.off()


# 
# 
# edb <- EnsDb.Hsapiens.v86
# seqlevelsStyle(edb) <- "UCSC"
# options(ensembldb.seqnameNotFound=NA)
# # seqlevels(edb)[1:30]
# 
# # options(ucscChromosomeNames = TRUE)
# eTrack <- GeneRegionTrack(edb, chromosome = "chr3", start = region_start, 
#                           end = region_end)
# chromosome(eTrack) <- "chr3" ## change to UCSC naming
# pdf()
# plotTracks(eTrack, transcriptAnnotation = "symbol", collapseTranscripts="meta")
# dev.off()



## Plot the stathmin2 gene model
## annotation track with stathmin2 gene which is on Chromosome 8: 79,611,117-79,666,158
chr <- "chr8"
region_start <- 79611100
region_end <- 79666200
## gene annotations from UCSC
options(ucscChromosomeNames = FALSE)
eTrack <- GeneRegionTrack(TxDb.Hsapiens.UCSC.hg38.knownGene, 
                          chromosome = chr, start = region_start, 
                          end = region_end)

## we need the read coverage in the stathmin2 genomic region for plotting
alTrack <- AlignmentsTrack(
  file.path("data", "Sep2020", "CellRangerCount_50076_2020-09-22--15-40-54",
            "no3_Neural_cuture_d_96_TDP-43-HA_4w_DOXONa",
            "possorted_genome_bam.bam"),
  chromosome = chr, from = region_start, to = region_end,
  isPaired = FALSE)

pdf()
plotTracks(list(alTrack, eTrack),  
           # collapseTranscripts="meta", 
           type = "coverage", 
           chromosome = chr, from = region_start, to = region_end,
           coverageHeight = 0.5, shape = "arrow", geneSymbols = TRUE)
dev.off()



## import stmn2 region from all four bam files and filter reads
## compute coverage
## plot annotation together with the 4 coverage tracks


## using biomart
library(biomaRt)
bm <- useMart(biomart = "ENSEMBL_MART_ENSEMBL", 
              dataset = "hsapiens_gene_ensembl")
biomTrack <- BiomartGeneRegionTrack(genome = "hg38", chromosome = "3", 
                                    start = region_start, end = region_end,
                                    name = "ENSEMBL", biomart = bm)


pdf()
plotTracks(list(alTrack, biomTrack),  
           # collapseTranscripts="meta", 
           type = "coverage", 
           # chromosome = "chr3", from = region_start, to = region_end,
           coverageHeight = 0.5, shape = "arrow", geneSymbols = TRUE)
dev.off()



grtrack <- GeneRegionTrack(geneModels, genome = gen,
                           chromosome = chr, name = "Gene Model")
plotTracks(list(itrack, gtrack, atrack, grtrack),)

## http://www.sthda.com/english/wiki/visualize-ngs-data-with-r-and-bioconductor
## https://www.bioconductor.org/packages/devel/bioc/vignettes/Gviz/inst/doc/Gviz.html
## rsamtools
## ggbio
## Gviz:
## write an import functions that only keeps reads from the selected cells?
myImportFun <- function(file, selection){
    ## do something here
}
DataTrack(range = bamFile, genome = "hg19", type = "l",
          name = "Coverage", window = -1, chromosome = "chr1",
          importFunction = myImportFun, stream = TRUE)
```


