---
title: "Stathmin 2 read coverage"
author: "Katharina Hembach"
date: "25/02/2021"
output: 
  html_document:
    toc: true,
    code_folding: show
---
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, autodep = TRUE, cache = TRUE, dev = "png",
                      dev.args = list(png = list(type = "cairo")), 
                      message = FALSE, cache.lazy = FALSE)
```

### Load packages

```{r load-libs, message = FALSE, warning = FALSE}
library(Seurat)
library(SingleCellExperiment)
library(dplyr)
library(Gviz)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
library(Rsamtools)
library(GenomicAlignments)
library(rtracklayer)
```

# Load data & convert to SCE

```{r load-data}
so <- readRDS(file.path("output", "so_TDP_05_plasmid_expression.rds"))
so <- SetIdent(so, value = "RNA_snn_res.0.4")
so@meta.data$cluster_id <- Idents(so)
```

# Cells from cluster 12

We want to compare the stathmin2 read coverage of cells expressing TDP-HA (from cluster 12) and other neuronal cells without TDP-HA expression. For this, we randomly select 5 cells from each group and filter the corresponding stathmin2 reads from the BAM file.

```{r select-cell-IDs}
clus12 <- subset(so, subset = cluster_id == "12")
## from which sample do the cells come from?
clus12$sample_id %>% table
## what is the range of TDP-HA expression in all cells in cluster 12?
dat_ha <- GetAssayData(object = clus12, slot = "data")["TDP43-HA",]
summary(dat_ha)
## select cells with high TDP-HA expression
high <- clus12[,which(dat_ha > 3.5)]
high$barcode %>% head
## cells with low TDP-HA expression
low <- clus12[,which(dat_ha < 0.5 & dat_ha > 0)]
low$barcode %>% head
```

# Get stathmin2 reads of selected cells

We extract the reads covering the stathmin2 genes of the selected cells.

```{r read-coverage}
bams <- list(TDP4wOFF = file.path("data", "Sep2020", "CellRangerCount_50076_2020-09-22--15-40-54",
            "no1_Neural_cuture_d_96_TDP-43-HA_4w_DOXoff",
            "possorted_genome_bam.bam"),
            TDP2wON = file.path("data", "Sep2020", "CellRangerCount_50076_2020-09-22--15-40-54",
            "no2_Neural_cuture_d_96_TDP-43-HA_2w_DOXON",
            "possorted_genome_bam.bam"),
            TDP4wONa = file.path("data", "Sep2020", "CellRangerCount_50076_2020-09-22--15-40-54",
            "no3_Neural_cuture_d_96_TDP-43-HA_4w_DOXONa",
            "possorted_genome_bam.bam"),
            TDP4wONb = file.path("data", "Sep2020", "CellRangerCount_50076_2020-09-22--15-40-54",
            "no4_Neural_cuture_d_96_TDP-43-HA_4w_DOXONb",
            "possorted_genome_bam.bam"))

chr <- "chr8"
region_start <- 79611100
region_end <- 79666200
stmn2 <- GRanges(chr, IRanges(region_start, region_end), "+")

# keep all reads from cells in cluster 12
param <- ScanBamParam(which=stmn2, what = c("qname"), tag = "CB", 
                      tagFilter = list(CB = clus12$barcode))
gals <- lapply(bams, function(x) {
  readGAlignments(x, use.names = TRUE, param=param)
  })

covs <- lapply(gals, coverage)
```


# Ploting with Gviz

Plot the stathmin2 transcripts and the read coverage of all cells from cluster 12.
```{r stmn2-clus12-coverage, fig.width = 12}
## gene annotations from UCSC
options(ucscChromosomeNames = FALSE)
eTrack <- GeneRegionTrack(TxDb.Hsapiens.UCSC.hg38.knownGene, 
                          chromosome = chr, start = region_start, 
                          end = region_end, name = "annotation")

coords <- 79611100:79666201
dat <- matrix(c(as.vector(covs[[1]]$chr8[region_start:region_end]),
                as.vector(covs[[2]]$chr8[region_start:region_end]),
                as.vector(covs[[3]]$chr8[region_start:region_end]),
                as.vector(covs[[4]]$chr8[region_start:region_end])), 
              nrow = 4, byrow = TRUE)
rownames(dat) <- names(covs)
dtrack <- DataTrack(data = dat,
                    start = coords[-length(coords)], end = coords[-1], chromosome = chr,
                    genome = "hg38")

plotTracks(c(dtrack, eTrack),  
           type = "histogram", showSampleNames = TRUE,
           shape = "arrow", geneSymbols = TRUE, aggregateGroups=FALSE,
           groups = c("TDP4wOFF", "TDP2wON", "TDP4wONa", "TDP4wONb"),
           stackedBars = FALSE, fontsize=13 )

## one data track per sample
dats <- list("4wOFF" = matrix(as.vector(covs[[1]]$chr8[region_start:region_end]),
                              nrow = 1, byrow = TRUE),
             "2wON" = matrix(as.vector(covs[[2]]$chr8[region_start:region_end]),
                              nrow = 1, byrow = TRUE),
             "4wONa" = matrix(as.vector(covs[[3]]$chr8[region_start:region_end]),
                              nrow = 1, byrow = TRUE),
             "4wONb" = matrix(as.vector(covs[[4]]$chr8[region_start:region_end]),
                              nrow = 1, byrow = TRUE))

# rownames(dat) <- names(covs)
dtrack_4wOFF <- DataTrack(data = dats[[1]],
                    start = coords[-length(coords)], end = coords[-1], chromosome = chr,
                    genome = "hg38", name = "4wOFF")
dtrack_2wON <- DataTrack(data = dats[[2]],
                    start = coords[-length(coords)], end = coords[-1], chromosome = chr,
                    genome = "hg38", name = "2wON")
dtrack_4wONa <- DataTrack(data = dats[[3]],
                    start = coords[-length(coords)], end = coords[-1], chromosome = chr,
                    genome = "hg38", name = "4wONa")
dtrack_4wONb <- DataTrack(data = dats[[4]],
                    start = coords[-length(coords)], end = coords[-1], chromosome = chr,
                    genome = "hg38", name = "4wONb")

plotTracks(c(dtrack_4wOFF, dtrack_2wON, dtrack_4wONa, dtrack_4wONb, eTrack), 
           type = "histogram", showSampleNames = TRUE,
           shape = "arrow", geneSymbols = TRUE, aggregateGroups=FALSE,
           stackedBars = FALSE, fontsize=13)

## zoom into intron 1 that contains the cryptic exon
chr <- "chr8"
region_start <- 79611100
region_end <- 79637000

plotTracks(c(dtrack_4wOFF, dtrack_2wON, dtrack_4wONa, dtrack_4wONb, eTrack),
           type = "histogram", showSampleNames = TRUE,
           chromosome = chr, from = region_start, to = region_end,
           shape = "arrow", geneSymbols = TRUE, aggregateGroups=FALSE,
           stackedBars = FALSE, fontsize=13)


## cryptic exon location: HG19 Chr8: 80,529,075-80,529,28
## in hg38: chr8:79616840-79617049
ce_start <- 79616840
ce_end <- 79617049
ceTrack <- AnnotationTrack(start = ce_start, end = ce_end, chromosome = chr, 
                          strand = "*", genome = "hg38", name = "CE")
plotTracks(c(dtrack_4wOFF, dtrack_2wON, dtrack_4wONa, dtrack_4wONb, 
             ceTrack, eTrack),
           type = "histogram", showSampleNames = TRUE,
           chromosome = chr, from = region_start, to = region_end,
           shape = "arrow", geneSymbols = TRUE, aggregateGroups=FALSE,
           stackedBars = FALSE, fontsize=13)
```

## Splice junctions

Are there any splice junctions in the first intron?

We define new functions that allow us to filter the BAM files basd on barcodes:
```{r Gviz-read-filtering}
####### import only reads from cells of cluster 12
## function is copid and modified from the Gviz package
.import.bam.alignments.cells <- function(file, selection) {
    indNames <- c(sub("\\.bam$", ".bai", file), paste(file, "bai", sep = "."))
    index <- NULL
    for (i in indNames) {
        if (file.exists(i)) {
            index <- i
            break
        }
    }
    if (is.null(index)) {
          stop(
              "Unable to find index for BAM file '", file, "'. You can build an index using the following command:\n\t",
              "library(Rsamtools)\n\tindexBam(\"", file, "\")"
          )
      }
    pairedEnd <- parent.env(environment())[["._isPaired"]]
    if (is.null(pairedEnd)) {
          pairedEnd <- TRUE
      }
    flag <- parent.env(environment())[["._flag"]]
    if (is.null(flag)) {
          flag <- scanBamFlag(isUnmappedQuery = FALSE)
      }
    bf <- BamFile(file, index = index, asMates = pairedEnd)
    cells <- parent.env(environment())[["._cells"]]
    if(!is.null(cells)){
      param <- ScanBamParam(which = selection, what = scanBamWhat(),
                          tag = c("MD", "CB"), flag = flag, 
                          tagFilter = list(CB = cells))
    } else{
      param <- ScanBamParam(which = selection, what = scanBamWhat(), 
                            tag = "MD", flag = flag)
    }
    
    reads <- if (as.character(seqnames(selection)[1]) %in% names(scanBamHeader(bf)$targets)) scanBam(bf, param = param)[[1]] else list()
    md <- if (is.null(reads$tag$MD)) rep(as.character(NA), length(reads$pos)) else reads$tag$MD
    if (length(reads$pos)) {
        layed_seq <- sequenceLayer(reads$seq, reads$cigar)
        region <- unlist(bamWhich(param), use.names = FALSE)
        ans <- stackStrings(layed_seq, start(region), end(region), shift = reads$pos - 1L, Lpadding.letter = "+", Rpadding.letter = "+")
        names(ans) <- seq_along(reads$qname)
    } else {
        ans <- DNAStringSet()
    }
    return(GRanges(
        seqnames = if (is.null(reads$rname)) character() else reads$rname,
        strand = if (is.null(reads$strand)) character() else reads$strand,
        ranges = IRanges(start = reads$pos, width = reads$qwidth),
        id = if (is.null(reads$qname)) character() else reads$qname,
        cigar = if (is.null(reads$cigar)) character() else reads$cigar,
        mapq = if (is.null(reads$mapq)) integer() else reads$mapq,
        flag = if (is.null(reads$flag)) integer() else reads$flag,
        md = md, seq = ans,
        isize = if (is.null(reads$isize)) integer() else reads$isize,
        groupid = if (pairedEnd) if (is.null(reads$groupid)) integer() else reads$groupid else seq_along(reads$pos),
        status = if (pairedEnd) {
            if (is.null(reads$mate_status)) factor(levels = c("mated", "ambiguous", "unmated")) else reads$mate_status
        } else {
            rep(
                factor("unmated", levels = c("mated", "ambiguous", "unmated")),
                length(reads$pos)
            )
        }
    ))
}


## Constructor
AlignmentsTrack <- function(range = NULL, start = NULL, end = NULL, width = NULL, strand, chromosome, genome,
                            stacking = "squish", id, cigar, mapq, flag = scanBamFlag(isUnmappedQuery = FALSE), isize, groupid, status, md, seqs,
                            name = "AlignmentsTrack", isPaired = TRUE, importFunction, referenceSequence, cells = NULL, ...) {
    ## Some defaults
    if (missing(importFunction)) {
        importFunction <- Gviz:::.import.bam.alignments
    }
    covars <- Gviz:::.getCovars(range)
    isStream <- FALSE
    if (!is.character(range)) {
        n <- max(c(length(start), length(end), length(width)), nrow(covars))
        id <- Gviz:::.covDefault(id, covars[["id"]], paste("read", seq_len(n), sep = "_"))
        cigar <- Gviz:::.covDefault(cigar, covars[["cigar"]], paste(if (is(range, "GRangesOrIRanges")) width(range) else width, "M", sep = ""))
        mapq <- Gviz:::.covDefault(mapq, covars[["mapq"]], rep(as.integer(NA), n))
        flag <- Gviz:::.covDefault(flag, covars[["flag"]], rep(as.integer(NA), n))
        isize <- Gviz:::.covDefault(isize, covars[["isize"]], rep(as.integer(NA), n))
        groupid <- Gviz:::.covDefault(groupid, covars[["groupid"]], seq_len(n))
        md <- Gviz:::.covDefault(md, covars[["md"]], rep(as.character(NA), n))
        status <- Gviz:::.covDefault(status, covars[["status"]], ifelse(groupid %in% groupid[duplicated(groupid)], "mated", "unmated"))
    }
    ## Build a GRanges object from the inputs
    Gviz:::.missingToNull(c(
        "strand", "chromosome", "importFunction", "genome", "id", "cigar", "mapq", "flag", "isize", "groupid", "status",
        "md", "seqs", "referenceSequence"
    ))
    args <- list(
        id = id, cigar = cigar, mapq = mapq, flag = flag, isize = isize, groupid = groupid, status = status, strand = strand, md = md,
        chromosome = chromosome, genome = genome
    )
    defs <- list(
        strand = "*", chromosome = "chrNA", genome = NA, id = as.character(NA), cigar = as.character(NA), mapq = as.integer(NA),
        flag = as.integer(NA), isize = as.integer(NA), groupid = as.character(NA), status = as.character(NA), md = as.character(NA)
    )
    range <- Gviz:::.buildRange(
        range = range, start = start, end = end, width = width,
        args = args, defaults = defs, chromosome = chromosome, trackType = "AlignmentsTrack",
        importFun = importFunction, stream = TRUE, autodetect = TRUE, ...
    )
    ## This is going to be a list if we have to stream data from a file, otherwise we can compute some additional values
    if (is.list(range)) {
        isStream <- TRUE
        slist <- range
        range <- GRanges()
        stackRanges <- GRanges()
        stacks <- NULL
        seqs <- DNAStringSet()
    } else {
        if (is.null(seqs)) {
            seqs <- DNAStringSet(vapply(width(range), function(x) paste(rep("N", x), collapse = ""), character(1)))
        }
        addArgs <- list(...)
        if ("showIndels" %in% names(addArgs)) {
            showIndels <- addArgs$showIndels
        } else {
            showIndels <- FALSE
        }
        tmp <- .computeAlignments(range, drop.D.ranges = showIndels)
        range <- tmp$range
        stackRanges <- tmp$stackRange
        stacks <- tmp$stacks
    }
    ## If no chromosome was explicitly asked for we just take the first one in the GRanges object
    if (missing(chromosome) || is.null(chromosome)) {
        chromosome <- if (length(range) > 0) Gviz:::.chrName(as.character(seqnames(range)[1])) else "chrNA"
    }
    ## And finally the object instantiation
    genome <- Gviz:::.getGenomeFromGRange(range, ifelse(is.null(genome), character(), genome[1]))
    if (!isStream) {
        return(new("AlignmentsTrack",
            chromosome = chromosome[1], range = range, stacks = stacks,
            name = name, genome = genome, stacking = stacking, stackRanges = stackRanges, sequences = seqs,
            referenceSequence = referenceSequence, ...
        ))
    } else {
        ## A bit hackish but for some functions we may want to know which track type we need but at the
        ## same time we do not want to enforce this as an additional argument
        e <- new.env()
        e[["._trackType"]] <- "AlignmentsTrack"
        e[["._isPaired"]] <- isPaired
        e[["._flag"]] <- flag
        e[["._cells"]] <- cells
        environment(slist[["stream"]]) <- e
        return(new("ReferenceAlignmentsTrack",
            chromosome = chromosome[1], range = range, stackRanges = stackRanges,
            name = name, genome = genome, stacking = stacking, stream = slist[["stream"]], reference = slist[["reference"]],
            mapping = slist[["mapping"]], args = args, defaults = defs, stacks = stacks, referenceSequence = referenceSequence, ...
        ))
    }
}
```


```{r stmn2-splice-junctions, fig.width = 12}
## STMN2 gene range
chr <- "chr8"
region_start <- 79611100
region_end <- 79666200

# Create the alignments track
alTrack_4wOFF <- AlignmentsTrack(
  range = bams[["TDP4wOFF"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == "TDP4wOFF"], 
  importFunction = .import.bam.alignments.cells, name = "4wOFF")
alTrack_2wON <- AlignmentsTrack(
  range = bams[["TDP2wON"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == "TDP2wON"], 
  importFunction = .import.bam.alignments.cells, name = "2wON")
alTrack_4wONa <- AlignmentsTrack(
  range = bams[["TDP4wONa"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == "TDP4wONa"], 
  importFunction = .import.bam.alignments.cells, name = "4wONa")
alTrack_4wONb <- AlignmentsTrack(
  range = bams[["TDP4wONb"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == "TDP4wONb"], 
  importFunction = .import.bam.alignments.cells, name = "4wONb")

## import GTF with gene annotation
gtf <- import(file.path("data", "Homo_sapiens.GRCh38.98.sorted.gtf"))
### import GTF, transfort to TxDb and create GeneRegionTrack
seqlevelsStyle(gtf) <- "UCSC"
txdb <- makeTxDbFromGRanges(gtf)
gtftrack <- GeneRegionTrack(txdb, name = "annotation") 


## Whole gene
plotTracks(c(alTrack_4wOFF, alTrack_2wON, alTrack_4wONa, alTrack_4wONb,
             ceTrack, gtftrack),  
           type = c("coverage", "sashimi"), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100,
           fontsize=13,
           sizes = c(rep(3, 4), 1, 3),
           transcriptAnnotation = "transcript")

## only first intron 
## zoom into intron 1 that contains the cryptic exon
region_start <- 79611100
region_end <- 79637000

plotTracks(c(alTrack_4wOFF, 
             alTrack_2wON, alTrack_4wONa, alTrack_4wONb,
             ceTrack, gtftrack),  
           type = c("coverage", "sashimi"), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100,
           fontsize=13,
           sizes = c(rep(3, 4), 1, 3),
           transcriptAnnotation = "transcript")
```


## Read coverage in cells with low and high TDP-HA expression from cluster 12

```{r low-high-TDPHA-cells}
## LOW ##
chr <- "chr8"
region_start <- 79611100
region_end <- 79666200

# Create the alignments track
low$sample_id %>% table
alTrack_4wOFF <- AlignmentsTrack(
  range = bams[["TDP4wOFF"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = low$barcode[low$sample_id == "TDP4wOFF"], 
  importFunction = .import.bam.alignments.cells, name = "4wOFF")
alTrack_2wON <- AlignmentsTrack(
  range = bams[["TDP2wON"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = low$barcode[low$sample_id == "TDP2wON"], 
  importFunction = .import.bam.alignments.cells, name = "2wON")
alTrack_4wONa <- AlignmentsTrack(
  range = bams[["TDP4wONa"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = low$barcode[low$sample_id == "TDP4wONa"], 
  importFunction = .import.bam.alignments.cells, name = "4wONa")

plotTracks(c(alTrack_4wOFF, alTrack_2wON, alTrack_4wONa,
             ceTrack, gtftrack),  
           type = c("coverage", "sashimi"), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100,
           fontsize=13,
           sizes = c(rep(3, 3), 1, 3),
           transcriptAnnotation = "transcript")

## only first intron 
## zoom into intron 1 that contains the cryptic exon
region_start <- 79611100
region_end <- 79637000

plotTracks(c(alTrack_4wOFF, alTrack_2wON, alTrack_4wONa, 
             ceTrack, gtftrack),  
           type = c("coverage", "sashimi"), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100,
           fontsize=13,
           sizes = c(rep(3, 3), 1, 3),
           transcriptAnnotation = "transcript")

## HIGH ##
region_start <- 79611100
region_end <- 79666200

# Create the alignments track
high$sample_id %>% table
alTrack_2wON <- AlignmentsTrack(
  range = bams[["TDP2wON"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = high$barcode[high$sample_id == "TDP2wON"], 
  importFunction = .import.bam.alignments.cells, name = "2wON")
alTrack_4wONa <- AlignmentsTrack(
  range = bams[["TDP4wONa"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = high$barcode[high$sample_id == "TDP4wONa"], 
  importFunction = .import.bam.alignments.cells, name = "4wONa")

plotTracks(c(alTrack_2wON, alTrack_4wONa, ceTrack, gtftrack),  
           type = c("coverage", "sashimi"), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100, fontsize=13,
           sizes = c(rep(4, 2), 1, 2),
           transcriptAnnotation = "transcript")


## only first intron 
## zoom into intron 1 that contains the cryptic exon
region_start <- 79611100
region_end <- 79637000

plotTracks(c(alTrack_2wON, alTrack_4wONa, ceTrack, gtftrack),  
           type = c("coverage", "sashimi"), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100, fontsize=13,
           sizes = c(rep(4, 2), 1, 3),
           transcriptAnnotation = "transcript")
```


# APOE genotype
We want to know the APOE genotype of our cells. The three APOE isoforms differ in only two positions in exon 4 (rs429358 = chr19:44908684 T>C and rs7412 = chr19:44908822 C>T). For this, we look at exon 4 of APOE and visualize mismatched bases. APOE-ε2 (rs7412-T, rs429358-T), APOE-ε3 (rs7412-C, rs429358-T), and APOE-ε4 (rs7412-C, rs429358-C)



```{r apoe-variants, fig.width=12}
## start of exon 4 44908520
## end of exon 4 44909400 
## region on chr 19 including the two variants: 44908690 - 44908820
region_start <- 44908680
region_end <- 44908830
chr <- "chr19"

## sequence track with the reference genome
library(BSgenome.Hsapiens.UCSC.hg38)
sTrack <- SequenceTrack(Hsapiens, chromosome = chr)

# genome <- BSgenome.Hsapiens.UCSC.hg38
# seq <- DNAStringSet(genome$chr19)
# names(seq) <- chr
# sTrack <- SequenceTrack(seq)

alTrack_4wOFF <- AlignmentsTrack(
  range = bams[["TDP4wOFF"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  name = "4wOFF")

# highlight variant regions
ht <- HighlightTrack(trackList = list(alTrack_4wOFF, sTrack),
                     start = c(44908684, 44908822), width = 0,
                     chromosome = chr)
plotTracks(ht, chromosome = chr,
           from = region_start, to = region_end, cex = 0.5, min.height = 2)

alTrack_2wON <- AlignmentsTrack(
  range = bams[["TDP2wON"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  name = "2wON")
ht <- HighlightTrack(trackList = list(alTrack_2wON, sTrack),
                     start = c(44908684, 44908822), width = 0,
                     chromosome = chr)
plotTracks(ht, chromosome = chr,
           from = region_start, to = region_end, cex = 0.5, min.height = 2)


alTrack_4wONa <- AlignmentsTrack(
  range = bams[["TDP4wONa"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  name = "4wONa")
ht <- HighlightTrack(trackList = list(alTrack_4wONa, sTrack),
                     start = c(44908684, 44908822), width = 0,
                     chromosome = chr)
plotTracks(ht, chromosome = chr,
           from = region_start, to = region_end, cex = 0.5, min.height = 2)


alTrack_4wONb <- AlignmentsTrack(
  range = bams[["TDP4wONb"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  name = "4wONb")
ht <- HighlightTrack(trackList = list(alTrack_4wONb, sTrack),
                     start = c(44908684, 44908822), width = 0,
                     chromosome = chr)
plotTracks(ht, chromosome = chr,
           from = region_start, to = region_end, cex = 0.5, min.height = 2)
```

# UNC13A read coverage

Is the cryptic exon in UNC13A expressed in our TDP-43-HA cells from cluster 12?

```{r UNC13A-splice-junctions, fig.width = 12}
## UNC13A gene range
chr <- "chr19"
region_start <- 17601328
region_end <- 17688199

# Create the alignments track
alTrack_4wOFF <- AlignmentsTrack(
  range = bams[["TDP4wOFF"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == "TDP4wOFF"], 
  importFunction = .import.bam.alignments.cells, name = "4wOFF")
alTrack_2wON <- AlignmentsTrack(
  range = bams[["TDP2wON"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == "TDP2wON"], 
  importFunction = .import.bam.alignments.cells, name = "2wON")
alTrack_4wONa <- AlignmentsTrack(
  range = bams[["TDP4wONa"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == "TDP4wONa"], 
  importFunction = .import.bam.alignments.cells, name = "4wONa")
alTrack_4wONb <- AlignmentsTrack(
  range = bams[["TDP4wONb"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == "TDP4wONb"], 
  importFunction = .import.bam.alignments.cells, name = "4wONb")

## cryptic exon between exon 20 and 21:
## exon 20: chr19:17642845-17642960
## exon 21: chr19:17641393-17641556
## cryptic exon location: chr19: 17642414-17642541
ce_start <- 17642414
ce_end <- 17642541
ceTrack <- AnnotationTrack(start = ce_start, end = ce_end, chromosome = chr, 
                          strand = "*", genome = "hg38", name = "CE")

## Whole gene
plotTracks(c(alTrack_4wOFF, alTrack_2wON, alTrack_4wONa, alTrack_4wONb,
             ceTrack, gtftrack),  
           type = c("coverage", "sashimi"), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100,
           fontsize=13,
           sizes = c(rep(3, 4), 1, 3),
           transcriptAnnotation = "transcript")

## only region around exon 20 and 21
region_start <- 17641393
region_end <- 17642960
plotTracks(c(alTrack_4wOFF, alTrack_2wON, alTrack_4wONa, alTrack_4wONb,
             ceTrack, gtftrack),  
           type = c("coverage", "sashimi"), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100,
           fontsize=13,
           sizes = c(rep(3, 4), 1, 3),
           transcriptAnnotation = "transcript")
```

# NPTX2 read coverage

We want to compare the read coverage in cluster 12 cells and cells from other neuronal clusters.

```{r NPTX2-read-coverage, fig.width = 12}
## UNC13A gene range
chr <- "chr7"
region_start <- 98617285
region_end <- 98629869

# Create the alignments track
alTrack_4wOFF <- AlignmentsTrack(
  range = bams[["TDP4wOFF"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == "TDP4wOFF"], 
  importFunction = .import.bam.alignments.cells, name = "4wOFF")
alTrack_2wON <- AlignmentsTrack(
  range = bams[["TDP2wON"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == "TDP2wON"], 
  importFunction = .import.bam.alignments.cells, name = "2wON")
alTrack_4wONa <- AlignmentsTrack(
  range = bams[["TDP4wONa"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == "TDP4wONa"], 
  importFunction = .import.bam.alignments.cells, name = "4wONa")
alTrack_4wONb <- AlignmentsTrack(
  range = bams[["TDP4wONb"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus12$barcode[clus12$sample_id == "TDP4wONb"], 
  importFunction = .import.bam.alignments.cells, name = "4wONb")

## Whole gene
plotTracks(c(alTrack_4wOFF, alTrack_2wON, alTrack_4wONa, alTrack_4wONb,gtftrack),  
           type = c("coverage", "sashimi"), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100,
           fontsize=13,
           sizes = c(rep(3, 4), 3),
           transcriptAnnotation = "transcript")


## other neuronal clusters
## cluster 8 are maturing inhibitory neurons
## cluster 7 are maturing excitatory neurons
clus7 <- subset(so, subset = cluster_id == "7")
clus8 <- subset(so, subset = cluster_id == "8")
clus12$sample_id %>% table
clus7$sample_id %>% table
clus8$sample_id %>% table

## excitatory
alTrack_4wOFF <- AlignmentsTrack(
  range = bams[["TDP4wOFF"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus7$barcode[clus7$sample_id == "TDP4wOFF"], 
  importFunction = .import.bam.alignments.cells, name = "4wOFF")
alTrack_2wON <- AlignmentsTrack(
  range = bams[["TDP2wON"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus7$barcode[clus7$sample_id == "TDP2wON"], 
  importFunction = .import.bam.alignments.cells, name = "2wON")
alTrack_4wONa <- AlignmentsTrack(
  range = bams[["TDP4wONa"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus7$barcode[clus7$sample_id == "TDP4wONa"], 
  importFunction = .import.bam.alignments.cells, name = "4wONa")
alTrack_4wONb <- AlignmentsTrack(
  range = bams[["TDP4wONb"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus7$barcode[clus7$sample_id == "TDP4wONb"], 
  importFunction = .import.bam.alignments.cells, name = "4wONb")

plotTracks(c(alTrack_4wOFF, alTrack_2wON, alTrack_4wONa, alTrack_4wONb,gtftrack),  
           type = c("coverage", "sashimi"), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100,
           fontsize=13,
           sizes = c(rep(3, 4), 3),
           transcriptAnnotation = "transcript")

## inhibitory
alTrack_4wOFF <- AlignmentsTrack(
  range = bams[["TDP4wOFF"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus8$barcode[clus8$sample_id == "TDP4wOFF"], 
  importFunction = .import.bam.alignments.cells, name = "4wOFF")
alTrack_2wON <- AlignmentsTrack(
  range = bams[["TDP2wON"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus8$barcode[clus8$sample_id == "TDP2wON"], 
  importFunction = .import.bam.alignments.cells, name = "2wON")
alTrack_4wONa <- AlignmentsTrack(
  range = bams[["TDP4wONa"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus8$barcode[clus8$sample_id == "TDP4wONa"], 
  importFunction = .import.bam.alignments.cells, name = "4wONa")
alTrack_4wONb <- AlignmentsTrack(
  range = bams[["TDP4wONb"]],
  isPaired = FALSE, chromosome = chr, from = region_start, to = region_end,
  cells = clus8$barcode[clus8$sample_id == "TDP4wONb"], 
  importFunction = .import.bam.alignments.cells, name = "4wONb")

plotTracks(c(alTrack_4wOFF, alTrack_2wON, alTrack_4wONa, alTrack_4wONb,gtftrack),  
           type = c("coverage", "sashimi"), 
           chromosome = chr, from = region_start, to = region_end,
           extend.left = 500, extend.right = 100,
           fontsize=13,
           sizes = c(rep(3, 4), 3),
           transcriptAnnotation = "transcript")
```

