---
title: "XL sites in cluster 12 markers (neuronal scRNA-seq)"
author: "Katharina Hembach"
date: "13.07.2021"
output: 
  html_document:
    toc: true,
    code_folding: show
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, autodep = TRUE, cache = TRUE, dev = "png",
                      dev.args = list(png = list(type = "cairo")), 
                      message = FALSE, cache.lazy = FALSE)
```

### Load packages
```{r, message = FALSE}
library(rtracklayer)
library(dplyr)
library(BSgenome.Hsapiens.UCSC.hg38)
library(GenomicFeatures)
library(ggrepel)
```

# Load data

```{r load-data}
dat_dir <- file.path("/home", "kathi", "tollervey_iCLIP")
gtf_file <- file.path("data", "Homo_sapiens.GRCh38.98.sorted.gtf")
gtf <- import(gtf_file)
genes <- gtf[gtf$type =="gene"]

genome <- BSgenome.Hsapiens.UCSC.hg38

xls <- list(control = import(file.path(dat_dir, "analysis", "merged_repl", 
                                       "xls_merged_control.bed")),
            FTLD = import(file.path(dat_dir, "analysis", "merged_repl", 
                                    "xls_merged_FTLD.bed")))
```

Marker lists:
```{r def-markers}
markers <- list(up = c("NPTX2", "FGF18", "PCED1A", "MEF2A", "DYNC2H1", 
                   "APOE", "GADD45A", "BCAM", "DDIT3"),
           down = c("VGF", "SCG2", "GAP43", "C4orf48", "HINT1", "LY6H", 
                     "TUBA1A", "TMSB4X", "TUBB2B", "STMN2", "UNC13A"))
```


# XL sites in cluster markers

What is the difference in the number of XL sites in cluster 12 marker genes?

```{r number-xls-markers, warnings = FALSE}
df <- data.frame(gene = c(markers[["up"]], markers[["down"]]),
                 direction = c(rep("up", length(markers[["up"]])),
                               rep("down", length(markers[["down"]]))))

olaps <- sapply(df$gene, function(x) {
    g <- genes[genes$gene_name == x]  
    res <- c(countOverlaps(g ,xls[["control"]]), countOverlaps(g, xls[["FTLD"]]))
    names(res) <- c("control", "FTLD")
    res
})

df$control <- olaps["control", ]
df$FTLD <- olaps["FTLD", ]
```

```{r plot-nr-xls, fig.width = 5, fig.height = 4}
ggplot(df, aes(x = control + 1, y = FTLD + 1, color = direction, label = gene)) + 
  geom_abline(intercept = 0, slope = 1, color = "grey") +
  geom_point(size = 3, alpha = 0.7) +
  geom_text_repel(color = "black", force = 2, size = 2.5) + 
  theme_bw() + 
  scale_x_log10(breaks=c(1, 2, 3, 6, 11, 21, 41, 61, 101),
                labels=c("0", "1", "2", "5", "10", "20", "40", "60", "100"),
                limits = c(1, max(df$control, df$FTLD)+1)) + 
  scale_y_log10(breaks=c(1, 2, 3, 6, 11, 21, 41, 61, 101),
                labels=c("0", "1", "2", "5", "10", "20", "40", "60", "100"),
                limits = c(1, max(df$control, df$FTLD)+1)) + 
  theme(panel.grid.minor = element_blank()) + 
  coord_fixed() + 
  scale_color_manual(values = c("steelblue", "orange2")) + 
  xlab("number of XLs control") + 
  ylab("number of XLs FTLD") +
  guides(colour = guide_legend(override.aes = list(alpha = 1)))
```

# UG repeats in NPTX2

Where are the UG repeats located in NPTX2?
```{r UG-repeats-NPTX2}
target <- genes[genes$gene_name == "NPTX2"]
seqlevelsStyle(target) <- "UCSC"

seq <- getSeq(genome, target)[[1]]
reps <- matchPattern("TGTG", seq)
reps <- GRanges(seqnames(target), 
                range = IRanges(start(reps) + start(target) -1, 
                                end(reps) + start(target) -1),
                strand = strand(target))
reps <- reduce(reps)
export(reps, file.path(dat_dir,"analysis", "cluster_12_markers", "UGUG_NPTX2.bed"))

reps <- matchPattern("TG", seq)
reps <- GRanges(seqnames(target), 
                range = IRanges(start(reps) + start(target) -1, 
                                end(reps) + start(target) -1),
                strand = strand(target))
reps <- reduce(reps)

export(reps, file.path(dat_dir, "analysis", "cluster_12_markers", "UG_NPTX2.bed"))
```

We also need the full 3'UTR region so we can highlight the UGs in the final figure.

```{r 3utr-seq}
utr <- gtf[gtf$gene_name == "NPTX2" & gtf$type == "three_prime_utr"]
seqlevelsStyle(utr) <- "UCSC"
export(getSeq(genome, utr), file.path(dat_dir, "analysis", 
                                      "cluster_12_markers", "NPTX2_3UTR.fasta"))
```

